// @nylas/.web-elements-GF2lXkj6/dist/collection/components/scheduler/nylas-confirmed-event-card/nylas-confirmed-event-card.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
        return Reflect.metadata(k, v);
};
import { RegisterComponent } from "../../../common/register-component";
import { h, Host } from "@stencil/core";
import { debug } from "../../../utils/utils";
import i18next from "../../../utils/i18n";
export class NylasConfirmedEventCard {
    constructor() {
        this.handleCloseClicked = () => {
            this.isClosing = true;
            const errorHandler = (error) => {
                this.confirmedEventCardError.emit(error);
                this.isClosing = false;
            };
            this.closeConfirmEventCardClicked.emit({ errorHandler });
        };
        this.confirmedEventInfo = undefined;
        this.isClosing = false;
        this.configSettings = undefined;
    }
    connectedCallback() {
        debug(`[nylas-confirmed-event-card] Component connected`);
    }
    async componentWillLoad() {
        debug(`[nylas-confirmed-event-card] Component will load`);
    }
    async componentDidLoad() {
        debug(`[nylas-confirmed-event-card] Component did load`);
        if (!this.confirmedEventInfo) {
            console.warn(`[nylas-confirmed-event-card] No confirmed event info provided, "confirmedEventInfo" prop is required.`);
        }
    }
    disconnectedCallback() {
        debug(`[nylas-confirmed-event-card] Component disconnected`);
    }
    render() {
        return (h(Host, { key: 'f78755849f1e51daa16349080efbd1154ee5421f', part: "ncec" }, h("div", { key: '3bef7ce7a2365171566b29b096b1625d0724a322', class: "nylas-confirmed-event-card", part: "ncec__card" }, h("div", { key: '1a246865b9400017d1945eb079691ac8f94baf1b', class: "nylas-confirmed-event-card__calendar-icon", part: "ncec__icon" }, h("calendar-cancel-icon", { key: '6af3a586336dc9d89f9e20bd9fbb2d64641c883f' })), h("h3", { key: 'ccb933bf503bd9f316c04647dd99652ac4422c78', class: "nylas-confirmed-event-card__title", part: "ncec__title" }, i18next.t('confirmedEventCardTitle'), "!"), h("div", { key: '21d9fa9069fc947e05b1bfe5f9c5e9a6aa4bb091', class: "nylas-confirmed-event-card__description", part: "ncec__description" }, i18next.t('confirmedEventCardDescription'), " ", this.configSettings?.organizer?.name || 'the participants', "."), h("div", { key: '93ee8ddcb48b9ec5f30df4cce75ca1a13a38bfff', class: "nylas-confirmed-event-card__cta" }, h("button-component", { key: 'ca0960f860c3701821b492e318a47a7cf58d6c40', variant: "basic", disabled: this.isClosing, part: "ncec__button-outline", onClick: this.handleCloseClicked }, this.isClosing ? `${i18next.t('closingButton')}...` : `${i18next.t('closeButton')}`)))));
    }
    static get is() { return "nylas-confirmed-event-card"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["nylas-confirmed-event-card.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["nylas-confirmed-event-card.css"]
        };
    }
    static get properties() {
        return {
            "confirmedEventInfo": {
                "type": "unknown",
                "mutable": false,
                "complexType": {
                    "original": "Partial<NylasEvent>",
                    "resolved": "{ booking_id?: string | undefined; organizer?: { email: string; name: string; is_organizer?: boolean | undefined; } | undefined; title?: string | undefined; description?: string | undefined; status?: string | undefined; }",
                    "references": {
                        "Partial": {
                            "location": "global",
                            "id": "global::Partial"
                        },
                        "NylasEvent": {
                            "location": "import",
                            "path": "../../..",
                            "id": "src/index.ts::NylasEvent"
                        }
                    }
                },
                "required": true,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "The participant's name who booked the event / is logged in."
                }
            },
            "configSettings": {
                "type": "unknown",
                "mutable": false,
                "complexType": {
                    "original": "ConfigSettings",
                    "resolved": "undefined | { scheduler: { available_days_in_future: number; min_cancellation_notice: number; min_booking_notice: number; rescheduling_url?: string | undefined; cancellation_url?: string | undefined; cancellation_policy?: string | undefined; hide_additional_guests?: boolean | undefined; hide_cancellation_options?: boolean | undefined; hide_rescheduling_options?: boolean | undefined; additional_fields?: Record<string, AdditionalFields> | undefined; confirmation_redirect_url?: string | undefined; organizer_confirmation_url?: string | undefined; }; organizer: { name: string; email: string; }; slug: string; appearance: Appearance; booking_type: string; name: string; }",
                    "references": {
                        "ConfigSettings": {
                            "location": "import",
                            "path": "@/stores/scheduler-store",
                            "id": "src/stores/scheduler-store.ts::ConfigSettings"
                        }
                    }
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": "The config settings for the scheduler."
                }
            }
        };
    }
    static get states() {
        return {
            "isClosing": {}
        };
    }
    static get events() {
        return [{
                "method": "closeConfirmEventCardClicked",
                "name": "closeConfirmEventCardClicked",
                "bubbles": true,
                "cancelable": true,
                "composed": true,
                "docs": {
                    "tags": [],
                    "text": "This event is fired when the close button is clicked on the confirmed event card."
                },
                "complexType": {
                    "original": "{ errorHandler?: (error: NylasSchedulerErrorResponse) => void }",
                    "resolved": "{ errorHandler?: ((error: NylasSchedulerErrorResponse) => void) | undefined; }",
                    "references": {
                        "NylasSchedulerErrorResponse": {
                            "location": "import",
                            "path": "../../..",
                            "id": "src/index.ts::NylasSchedulerErrorResponse"
                        }
                    }
                }
            }, {
                "method": "confirmedEventCardError",
                "name": "confirmedEventCardError",
                "bubbles": true,
                "cancelable": true,
                "composed": true,
                "docs": {
                    "tags": [],
                    "text": "This event is fired when an error occurs."
                },
                "complexType": {
                    "original": "NylasSchedulerErrorResponse",
                    "resolved": "NylasSchedulerErrorResponse",
                    "references": {
                        "NylasSchedulerErrorResponse": {
                            "location": "import",
                            "path": "../../..",
                            "id": "src/index.ts::NylasSchedulerErrorResponse"
                        }
                    }
                }
            }];
    }
}
__decorate([
    RegisterComponent({
        name: 'nylas-confirmed-event-card',
        stateToProps: new Map([['scheduler.configSettings', 'configSettings']]),
        eventToProps: {
            closeConfirmEventCardClicked: async (_, nylasschedulerconnector) => {
                const result = await nylasschedulerconnector.scheduler.resetConfirm();
                const { errorHandler } = _.detail;
                if (errorHandler && (!result || 'error' in result)) {
                    errorHandler(result);
                }
            },
        },
        fireRegisterEvent: true,
    }),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", []),
    __metadata("design:returntype", void 0)
], NylasConfirmedEventCard.prototype, "render", null);
