// @nylas/.web-elements-GF2lXkj6/dist/cdn/calendar-check/calendar-check.es.js
const b = {
  allRenderFn: !1,
  cmpDidLoad: !0,
  cmpDidUnload: !1,
  cmpDidUpdate: !0,
  cmpDidRender: !0,
  cmpWillLoad: !0,
  cmpWillUpdate: !0,
  cmpWillRender: !0,
  connectedCallback: !0,
  disconnectedCallback: !0,
  element: !0,
  event: !0,
  hasRenderFn: !0,
  lifecycle: !0,
  hostListener: !0,
  hostListenerTargetWindow: !0,
  hostListenerTargetDocument: !0,
  hostListenerTargetBody: !0,
  hostListenerTargetParent: !1,
  hostListenerTarget: !0,
  member: !0,
  method: !0,
  mode: !0,
  observeAttribute: !0,
  prop: !0,
  propMutable: !0,
  reflect: !0,
  scoped: !0,
  shadowDom: !0,
  slot: !0,
  cssAnnotations: !0,
  state: !0,
  style: !0,
  formAssociated: !1,
  svg: !0,
  updatable: !0,
  vdomAttribute: !0,
  vdomXlink: !0,
  vdomClass: !0,
  vdomFunctional: !0,
  vdomKey: !0,
  vdomListener: !0,
  vdomRef: !0,
  vdomPropOrAttr: !0,
  vdomRender: !0,
  vdomStyle: !0,
  vdomText: !0,
  watchCallback: !0,
  taskQueue: !0,
  hotModuleReplacement: !1,
  isDebug: !1,
  isDev: !1,
  isTesting: !1,
  hydrateServerSide: !1,
  hydrateClientSide: !1,
  lifecycleDOMEvents: !1,
  lazyLoad: !1,
  profile: !1,
  slotRelocation: !0,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  appendChildSlotFix: !1,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  cloneNodeFix: !1,
  hydratedAttribute: !1,
  hydratedClass: !0,
  scriptDataOpts: !1,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  scopedSlotTextContentFix: !1,
  // TODO(STENCIL-854): Remove code related to legacy shadowDomShim field
  shadowDomShim: !1,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  slotChildNodesFix: !1,
  invisiblePrehydration: !0,
  propBoolean: !0,
  propNumber: !0,
  propString: !0,
  constructableCSS: !0,
  cmpShouldUpdate: !0,
  devTools: !1,
  shadowDelegatesFocus: !0,
  initializeNextTick: !1,
  asyncLoading: !1,
  asyncQueue: !1,
  transformTagName: !1,
  attachStyles: !0,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  experimentalSlotFixes: !1
};
let x, B, _, le = !1, F = !1, W = !1, v = !1, I = null, D = !1;
const L = (e, t = "") => () => {
}, ve = "slot-fb{display:contents}slot-fb[hidden]{display:none}", Q = "http://www.w3.org/1999/xlink", X = {}, Se = "http://www.w3.org/2000/svg", ke = "http://www.w3.org/1999/xhtml", me = (e) => e != null, z = (e) => (e = typeof e, e === "object" || e === "function");
function be(e) {
  var t, n, s;
  return (s = (n = (t = e.head) === null || t === void 0 ? void 0 : t.querySelector('meta[name="csp-nonce"]')) === null || n === void 0 ? void 0 : n.getAttribute("content")) !== null && s !== void 0 ? s : void 0;
}
const U = (e, t, ...n) => {
  let s = null, l = null, c = null, o = !1, r = !1;
  const i = [], u = (a) => {
    for (let d = 0; d < a.length; d++)
      s = a[d], Array.isArray(s) ? u(s) : s != null && typeof s != "boolean" && ((o = typeof e != "function" && !z(s)) && (s = String(s)), o && r ? i[i.length - 1].$text$ += s : i.push(o ? P(null, s) : s), r = o);
  };
  if (u(n), t) {
    t.key && (l = t.key), t.name && (c = t.name);
    {
      const a = t.className || t.class;
      a && (t.class = typeof a != "object" ? a : Object.keys(a).filter((d) => a[d]).join(" "));
    }
  }
  if (typeof e == "function")
    return e(t === null ? {} : t, i, xe);
  const f = P(e, null);
  return f.$attrs$ = t, i.length > 0 && (f.$children$ = i), f.$key$ = l, f.$name$ = c, f;
}, P = (e, t) => {
  const n = {
    $flags$: 0,
    $tag$: e,
    $text$: t,
    $elm$: null,
    $children$: null
  };
  return n.$attrs$ = null, n.$key$ = null, n.$name$ = null, n;
}, Te = {}, Le = (e) => e && e.$tag$ === Te, xe = {
  forEach: (e, t) => e.map(Z).forEach(t),
  map: (e, t) => e.map(Z).map(t).map(Ce)
}, Z = (e) => ({
  vattrs: e.$attrs$,
  vchildren: e.$children$,
  vkey: e.$key$,
  vname: e.$name$,
  vtag: e.$tag$,
  vtext: e.$text$
}), Ce = (e) => {
  if (typeof e.vtag == "function") {
    const n = Object.assign({}, e.vattrs);
    return e.vkey && (n.key = e.vkey), e.vname && (n.name = e.vname), U(e.vtag, n, ...e.vchildren || []);
  }
  const t = P(e.vtag, e.vtext);
  return t.$attrs$ = e.vattrs, t.$children$ = e.vchildren, t.$key$ = e.vkey, t.$name$ = e.vname, t;
}, Ee = (e) => et.map((t) => t(e)).find((t) => !!t), _e = (e, t) => e != null && !z(e) ? t & 4 ? e === "false" ? !1 : e === "" || !!e : t & 2 ? parseFloat(e) : t & 1 ? String(e) : e : e, G = /* @__PURE__ */ new WeakMap(), Ae = (e, t, n) => {
  let s = M.get(e);
  lt && n ? (s = s || new CSSStyleSheet(), typeof s == "string" ? s = t : s.replaceSync(t)) : s = t, M.set(e, s);
}, Re = (e, t, n) => {
  var s;
  const l = oe(t, n), c = M.get(l);
  if (e = e.nodeType === 11 ? e : S, c)
    if (typeof c == "string") {
      e = e.head || e;
      let o = G.get(e), r;
      if (o || G.set(e, o = /* @__PURE__ */ new Set()), !o.has(l)) {
        {
          r = S.createElement("style"), r.innerHTML = c;
          const i = (s = p.$nonce$) !== null && s !== void 0 ? s : be(S);
          i != null && r.setAttribute("nonce", i), e.insertBefore(r, e.querySelector("link"));
        }
        t.$flags$ & 4 && (r.innerHTML += ve), o && o.add(l);
      }
    } else e.adoptedStyleSheets.includes(c) || (e.adoptedStyleSheets = [...e.adoptedStyleSheets, c]);
  return l;
}, Oe = (e) => {
  const t = e.$cmpMeta$, n = e.$hostElement$, s = t.$flags$, l = L("attachStyles", t.$tagName$), c = Re(n.shadowRoot ? n.shadowRoot : n.getRootNode(), t, e.$modeName$);
  s & 10 && (n["s-sc"] = c, n.classList.add(c + "-h"), s & 2 && n.classList.add(c + "-s")), l();
}, oe = (e, t) => "sc-" + (t && e.$flags$ & 32 ? e.$tagName$ + "-" + t : e.$tagName$), K = (e, t, n, s, l, c) => {
  if (n !== s) {
    let o = te(e, t), r = t.toLowerCase();
    if (t === "class") {
      const i = e.classList, u = J(n), f = J(s);
      i.remove(...u.filter((a) => a && !f.includes(a))), i.add(...f.filter((a) => a && !u.includes(a)));
    } else if (t === "style") {
      for (const i in n)
        (!s || s[i] == null) && (i.includes("-") ? e.style.removeProperty(i) : e.style[i] = "");
      for (const i in s)
        (!n || s[i] !== n[i]) && (i.includes("-") ? e.style.setProperty(i, s[i]) : e.style[i] = s[i]);
    } else if (t !== "key")
      if (t === "ref")
        s && s(e);
      else if (!e.__lookupSetter__(t) && t[0] === "o" && t[1] === "n") {
        if (t[2] === "-" ? t = t.slice(3) : te(w, r) ? t = r.slice(2) : t = r[2] + t.slice(3), n || s) {
          const i = t.endsWith(ie);
          t = t.replace(Fe, ""), n && p.rel(e, t, n, i), s && p.ael(e, t, s, i);
        }
      } else {
        const i = z(s);
        if ((o || i && s !== null) && !l)
          try {
            if (e.tagName.includes("-"))
              e[t] = s;
            else {
              const f = s ?? "";
              t === "list" ? o = !1 : (n == null || e[t] != f) && (e[t] = f);
            }
          } catch {
          }
        let u = !1;
        r !== (r = r.replace(/^xlink\:?/, "")) && (t = r, u = !0), s == null || s === !1 ? (s !== !1 || e.getAttribute(t) === "") && (u ? e.removeAttributeNS(Q, t) : e.removeAttribute(t)) : (!o || c & 4 || l) && !i && (s = s === !0 ? "" : s, u ? e.setAttributeNS(Q, t, s) : e.setAttribute(t, s));
      }
  }
}, He = /\s/, J = (e) => e ? e.split(He) : [], ie = "Capture", Fe = new RegExp(ie + "$"), ce = (e, t, n, s) => {
  const l = t.$elm$.nodeType === 11 && t.$elm$.host ? t.$elm$.host : t.$elm$, c = e && e.$attrs$ || X, o = t.$attrs$ || X;
  for (s in c)
    s in o || K(l, s, c[s], void 0, n, t.$flags$);
  for (s in o)
    K(l, s, c[s], o[s], n, t.$flags$);
}, j = (e, t, n, s) => {
  const l = t.$children$[n];
  let c = 0, o, r, i;
  if (le || (W = !0, l.$tag$ === "slot" && (x && s.classList.add(x + "-s"), l.$flags$ |= l.$children$ ? (
    // slot element has fallback content
    2
  ) : (
    // slot element does not have fallback content
    1
  ))), l.$text$ !== null)
    o = l.$elm$ = S.createTextNode(l.$text$);
  else if (l.$flags$ & 1)
    o = l.$elm$ = S.createTextNode("");
  else {
    if (v || (v = l.$tag$ === "svg"), o = l.$elm$ = S.createElementNS(v ? Se : ke, l.$flags$ & 2 ? "slot-fb" : l.$tag$), v && l.$tag$ === "foreignObject" && (v = !1), ce(null, l, v), me(x) && o["s-si"] !== x && o.classList.add(o["s-si"] = x), l.$children$)
      for (c = 0; c < l.$children$.length; ++c)
        r = j(e, l, c, o), r && o.appendChild(r);
    l.$tag$ === "svg" ? v = !1 : o.tagName === "foreignObject" && (v = !0);
  }
  return o["s-hn"] = _, l.$flags$ & 3 && (o["s-sr"] = !0, o["s-cr"] = B, o["s-sn"] = l.$name$ || "", i = e && e.$children$ && e.$children$[n], i && i.$tag$ === l.$tag$ && e.$elm$ && A(e.$elm$, !1)), o;
}, A = (e, t) => {
  p.$flags$ |= 1;
  const n = e.childNodes;
  for (let s = n.length - 1; s >= 0; s--) {
    const l = n[s];
    l["s-hn"] !== _ && l["s-ol"] && ($e(l).insertBefore(l, q(l)), l["s-ol"].remove(), l["s-ol"] = void 0, l["s-sh"] = void 0, W = !0), t && A(l, t);
  }
  p.$flags$ &= -2;
}, re = (e, t, n, s, l, c) => {
  let o = e["s-cr"] && e["s-cr"].parentNode || e, r;
  for (o.shadowRoot && o.tagName === _ && (o = o.shadowRoot); l <= c; ++l)
    s[l] && (r = j(null, n, l, e), r && (s[l].$elm$ = r, o.insertBefore(r, q(t))));
}, ae = (e, t, n) => {
  for (let s = t; s <= n; ++s) {
    const l = e[s];
    if (l) {
      const c = l.$elm$;
      ue(l), c && (F = !0, c["s-ol"] ? c["s-ol"].remove() : A(c, !0), c.remove());
    }
  }
}, Ue = (e, t, n, s, l = !1) => {
  let c = 0, o = 0, r = 0, i = 0, u = t.length - 1, f = t[0], a = t[u], d = s.length - 1, $ = s[0], h = s[d], y, g;
  for (; c <= u && o <= d; )
    if (f == null)
      f = t[++c];
    else if (a == null)
      a = t[--u];
    else if ($ == null)
      $ = s[++o];
    else if (h == null)
      h = s[--d];
    else if (H(f, $, l))
      C(f, $, l), f = t[++c], $ = s[++o];
    else if (H(a, h, l))
      C(a, h, l), a = t[--u], h = s[--d];
    else if (H(f, h, l))
      (f.$tag$ === "slot" || h.$tag$ === "slot") && A(f.$elm$.parentNode, !1), C(f, h, l), e.insertBefore(f.$elm$, a.$elm$.nextSibling), f = t[++c], h = s[--d];
    else if (H(a, $, l))
      (f.$tag$ === "slot" || h.$tag$ === "slot") && A(a.$elm$.parentNode, !1), C(a, $, l), e.insertBefore(a.$elm$, f.$elm$), a = t[--u], $ = s[++o];
    else {
      for (r = -1, i = c; i <= u; ++i)
        if (t[i] && t[i].$key$ !== null && t[i].$key$ === $.$key$) {
          r = i;
          break;
        }
      r >= 0 ? (g = t[r], g.$tag$ !== $.$tag$ ? y = j(t && t[o], n, r, e) : (C(g, $, l), t[r] = void 0, y = g.$elm$), $ = s[++o]) : (y = j(t && t[o], n, o, e), $ = s[++o]), y && $e(f.$elm$).insertBefore(y, q(f.$elm$));
    }
  c > u ? re(e, s[d + 1] == null ? null : s[d + 1].$elm$, n, s, o, d) : o > d && ae(t, c, u);
}, H = (e, t, n = !1) => e.$tag$ === t.$tag$ ? e.$tag$ === "slot" ? e.$name$ === t.$name$ : n ? !0 : e.$key$ === t.$key$ : !1, q = (e) => e && e["s-ol"] || e, $e = (e) => (e["s-ol"] ? e["s-ol"] : e).parentNode, C = (e, t, n = !1) => {
  const s = t.$elm$ = e.$elm$, l = e.$children$, c = t.$children$, o = t.$tag$, r = t.$text$;
  let i;
  r === null ? (v = o === "svg" ? !0 : o === "foreignObject" ? !1 : v, o === "slot" || ce(e, t, v), l !== null && c !== null ? Ue(s, l, t, c, n) : c !== null ? (e.$text$ !== null && (s.textContent = ""), re(s, null, t, c, 0, c.length - 1)) : l !== null && ae(l, 0, l.length - 1), v && o === "svg" && (v = !1)) : (i = s["s-cr"]) ? i.parentNode.textContent = r : e.$text$ !== r && (s.data = r);
}, fe = (e) => {
  const t = e.childNodes;
  for (const n of t)
    if (n.nodeType === 1) {
      if (n["s-sr"]) {
        const s = n["s-sn"];
        n.hidden = !1;
        for (const l of t)
          if (l !== n) {
            if (l["s-hn"] !== n["s-hn"] || s !== "") {
              if (l.nodeType === 1 && (s === l.getAttribute("slot") || s === l["s-sn"])) {
                n.hidden = !0;
                break;
              }
            } else if (l.nodeType === 1 || l.nodeType === 3 && l.textContent.trim() !== "") {
              n.hidden = !0;
              break;
            }
          }
      }
      fe(n);
    }
}, k = [], de = (e) => {
  let t, n, s;
  for (const l of e.childNodes) {
    if (l["s-sr"] && (t = l["s-cr"]) && t.parentNode) {
      n = t.parentNode.childNodes;
      const c = l["s-sn"];
      for (s = n.length - 1; s >= 0; s--)
        if (t = n[s], !t["s-cn"] && !t["s-nr"] && t["s-hn"] !== l["s-hn"] && !b.experimentalSlotFixes)
          if (Y(t, c)) {
            let o = k.find((r) => r.$nodeToRelocate$ === t);
            F = !0, t["s-sn"] = t["s-sn"] || c, o ? (o.$nodeToRelocate$["s-sh"] = l["s-hn"], o.$slotRefNode$ = l) : (t["s-sh"] = l["s-hn"], k.push({
              $slotRefNode$: l,
              $nodeToRelocate$: t
            })), t["s-sr"] && k.map((r) => {
              Y(r.$nodeToRelocate$, t["s-sn"]) && (o = k.find((i) => i.$nodeToRelocate$ === t), o && !r.$slotRefNode$ && (r.$slotRefNode$ = o.$slotRefNode$));
            });
          } else k.some((o) => o.$nodeToRelocate$ === t) || k.push({
            $nodeToRelocate$: t
          });
    }
    l.nodeType === 1 && de(l);
  }
}, Y = (e, t) => e.nodeType === 1 ? e.getAttribute("slot") === null && t === "" || e.getAttribute("slot") === t : e["s-sn"] === t ? !0 : t === "", ue = (e) => {
  e.$attrs$ && e.$attrs$.ref && e.$attrs$.ref(null), e.$children$ && e.$children$.map(ue);
}, Pe = (e, t, n = !1) => {
  var s, l, c, o, r;
  const i = e.$hostElement$, u = e.$cmpMeta$, f = e.$vnode$ || P(null, null), a = Le(t) ? t : U(null, null, t);
  if (_ = i.tagName, u.$attrsToReflect$ && (a.$attrs$ = a.$attrs$ || {}, u.$attrsToReflect$.map(([d, $]) => a.$attrs$[$] = i[d])), n && a.$attrs$)
    for (const d of Object.keys(a.$attrs$))
      i.hasAttribute(d) && !["key", "ref", "style", "class"].includes(d) && (a.$attrs$[d] = i[d]);
  a.$tag$ = null, a.$flags$ |= 4, e.$vnode$ = a, a.$elm$ = f.$elm$ = i.shadowRoot || i, x = i["s-sc"], B = i["s-cr"], le = (u.$flags$ & 1) !== 0, F = !1, C(f, a, n);
  {
    if (p.$flags$ |= 1, W) {
      de(a.$elm$);
      for (const d of k) {
        const $ = d.$nodeToRelocate$;
        if (!$["s-ol"]) {
          const h = S.createTextNode("");
          h["s-nr"] = $, $.parentNode.insertBefore($["s-ol"] = h, $);
        }
      }
      for (const d of k) {
        const $ = d.$nodeToRelocate$, h = d.$slotRefNode$;
        if (h) {
          const y = h.parentNode;
          let g = h.nextSibling;
          {
            let T = (s = $["s-ol"]) === null || s === void 0 ? void 0 : s.previousSibling;
            for (; T; ) {
              let m = (l = T["s-nr"]) !== null && l !== void 0 ? l : null;
              if (m && m["s-sn"] === $["s-sn"] && y === m.parentNode && (m = m.nextSibling, !m || !m["s-nr"])) {
                g = m;
                break;
              }
              T = T.previousSibling;
            }
          }
          (!g && y !== $.parentNode || $.nextSibling !== g) && $ !== g && (!$["s-hn"] && $["s-ol"] && ($["s-hn"] = $["s-ol"].parentNode.nodeName), y.insertBefore($, g), $.nodeType === 1 && ($.hidden = (c = $["s-ih"]) !== null && c !== void 0 ? c : !1));
        } else
          $.nodeType === 1 && (n && ($["s-ih"] = (o = $.hidden) !== null && o !== void 0 ? o : !1), $.hidden = !0);
      }
    }
    F && fe(a.$elm$), p.$flags$ &= -2, k.length = 0;
  }
  if (b.experimentalScopedSlotChanges && u.$flags$ & 2)
    for (const d of a.$elm$.childNodes)
      d["s-hn"] !== _ && !d["s-sh"] && (n && d["s-ih"] == null && (d["s-ih"] = (r = d.hidden) !== null && r !== void 0 ? r : !1), d.hidden = !0);
  B = void 0;
}, je = (e, t) => {
}, he = (e, t) => (e.$flags$ |= 16, je(e, e.$ancestorComponent$), ct(() => Me(e, t))), Me = (e, t) => {
  const n = e.$hostElement$, s = L("scheduleUpdate", e.$cmpMeta$.$tagName$), l = n;
  let c;
  return t ? c = E(l, "componentWillLoad") : c = E(l, "componentWillUpdate"), c = V(c, () => E(l, "componentWillRender")), s(), V(c, () => Be(e, l, t));
}, V = (e, t) => we(e) ? e.then(t) : t(), we = (e) => e instanceof Promise || e && e.then && typeof e.then == "function", Be = async (e, t, n) => {
  const s = e.$hostElement$, l = L("update", e.$cmpMeta$.$tagName$);
  s["s-rc"], n && Oe(e);
  const c = L("render", e.$cmpMeta$.$tagName$);
  De(e, t, s, n), c(), l(), Ne(e);
}, De = (e, t, n, s) => {
  try {
    I = t, t = t.render && t.render(), e.$flags$ &= -17, e.$flags$ |= 2, (b.hasRenderFn || b.reflect) && (b.vdomRender || b.reflect) && (b.hydrateServerSide || Pe(e, t, s));
  } catch (i) {
    O(i, e.$hostElement$);
  }
  return I = null, null;
}, Ne = (e) => {
  const t = e.$cmpMeta$.$tagName$, n = e.$hostElement$, s = L("postUpdate", t), l = n;
  e.$ancestorComponent$, E(l, "componentDidRender"), e.$flags$ & 64 ? (E(l, "componentDidUpdate"), s()) : (e.$flags$ |= 64, E(l, "componentDidLoad"), s());
}, E = (e, t, n) => {
  if (e && e[t])
    try {
      return e[t](n);
    } catch (s) {
      O(s);
    }
}, We = (e, t) => R(e).$instanceValues$.get(t), ze = (e, t, n, s) => {
  const l = R(e), c = e, o = l.$instanceValues$.get(t), r = l.$flags$, i = c;
  n = _e(n, s.$members$[t][0]);
  const u = Number.isNaN(o) && Number.isNaN(n);
  if (n !== o && !u) {
    l.$instanceValues$.set(t, n);
    {
      if (s.$watchers$ && r & 128) {
        const a = s.$watchers$[t];
        a && a.map((d) => {
          try {
            i[d](n, o, t);
          } catch ($) {
            O($, c);
          }
        });
      }
      if ((r & 18) === 2) {
        if (i.componentShouldUpdate && i.componentShouldUpdate(n, o, t) === !1)
          return;
        he(l, !1);
      }
    }
  }
}, qe = (e, t, n) => {
  var s;
  const l = e.prototype;
  if (t.$members$) {
    e.watchers && (t.$watchers$ = e.watchers);
    const c = Object.entries(t.$members$);
    c.map(([o, [r]]) => {
      (r & 31 || r & 32) && Object.defineProperty(l, o, {
        get() {
          return We(this, o);
        },
        set(i) {
          ze(this, o, i, t);
        },
        configurable: !0,
        enumerable: !0
      });
    });
    {
      const o = /* @__PURE__ */ new Map();
      l.attributeChangedCallback = function(r, i, u) {
        p.jmp(() => {
          var f;
          const a = o.get(r);
          if (this.hasOwnProperty(a))
            u = this[a], delete this[a];
          else {
            if (l.hasOwnProperty(a) && typeof this[a] == "number" && this[a] == u)
              return;
            if (a == null) {
              const d = R(this), $ = d == null ? void 0 : d.$flags$;
              if ($ && !($ & 8) && $ & 128 && u !== i) {
                const y = this, g = (f = t.$watchers$) === null || f === void 0 ? void 0 : f[r];
                g == null || g.forEach((T) => {
                  y[T] != null && y[T].call(y, u, i, r);
                });
              }
              return;
            }
          }
          this[a] = u === null && typeof this[a] == "boolean" ? !1 : u;
        });
      }, e.observedAttributes = Array.from(/* @__PURE__ */ new Set([
        ...Object.keys((s = t.$watchers$) !== null && s !== void 0 ? s : {}),
        ...c.filter(
          ([r, i]) => i[0] & 15
          /* MEMBER_FLAGS.HasAttribute */
        ).map(([r, i]) => {
          var u;
          const f = i[1] || r;
          return o.set(f, r), i[0] & 512 && ((u = t.$attrsToReflect$) === null || u === void 0 || u.push([r, f])), f;
        })
      ]));
    }
  }
  return e;
}, Ie = async (e, t, n, s) => {
  let l;
  if (!(t.$flags$ & 32) && (t.$flags$ |= 32, l = e.constructor, customElements.whenDefined(n.$tagName$).then(() => t.$flags$ |= 128), l.style)) {
    let o = l.style;
    typeof o != "string" && (o = o[t.$modeName$ = Ee(e)]);
    const r = oe(n, t.$modeName$);
    if (!M.has(r)) {
      const i = L("registerStyles", n.$tagName$);
      Ae(r, o, !!(n.$flags$ & 1)), i();
    }
  }
  t.$ancestorComponent$, he(t, !0);
}, ee = (e) => {
}, Qe = (e) => {
  if (!(p.$flags$ & 1)) {
    const t = R(e), n = t.$cmpMeta$, s = L("connectedCallback", n.$tagName$);
    t.$flags$ & 1 ? (pe(e, t, n.$listeners$), t != null && t.$lazyInstance$ ? ee(t.$lazyInstance$) : t != null && t.$onReadyPromise$ && t.$onReadyPromise$.then(() => ee(t.$lazyInstance$))) : (t.$flags$ |= 1, // TODO(STENCIL-854): Remove code related to legacy shadowDomShim field
    n.$flags$ & 12 && Xe(e), n.$members$ && Object.entries(n.$members$).map(([l, [c]]) => {
      if (c & 31 && e.hasOwnProperty(l)) {
        const o = e[l];
        delete e[l], e[l] = o;
      }
    }), Ie(e, t, n)), s();
  }
}, Xe = (e) => {
  const t = e["s-cr"] = S.createComment("");
  t["s-cn"] = !0, e.insertBefore(t, e.firstChild);
}, Ze = async (e) => {
  if (!(p.$flags$ & 1)) {
    const t = R(e);
    t.$rmListeners$ && (t.$rmListeners$.map((n) => n()), t.$rmListeners$ = void 0);
  }
}, Ge = (e, t) => {
  const n = {
    $flags$: t[0],
    $tagName$: t[1]
  };
  n.$members$ = t[2], n.$listeners$ = t[3], n.$watchers$ = e.$watchers$, n.$attrsToReflect$ = [];
  const s = e.prototype.connectedCallback, l = e.prototype.disconnectedCallback;
  return Object.assign(e.prototype, {
    __registerHost() {
      Ve(this, n);
    },
    connectedCallback() {
      Qe(this), s && s.call(this);
    },
    disconnectedCallback() {
      Ze(this), l && l.call(this);
    },
    __attachShadow() {
      this.attachShadow({
        mode: "open",
        delegatesFocus: !!(n.$flags$ & 16)
      });
    }
  }), e.is = n.$tagName$, qe(e, n);
}, pe = (e, t, n, s) => {
  n && n.map(([l, c, o]) => {
    const r = Je(e, l), i = Ke(t, o), u = Ye(l);
    p.ael(r, c, i, u), (t.$rmListeners$ = t.$rmListeners$ || []).push(() => p.rel(r, c, i, u));
  });
}, Ke = (e, t) => (n) => {
  try {
    b.lazyLoad || e.$hostElement$[t](n);
  } catch (s) {
    O(s);
  }
}, Je = (e, t) => t & 4 ? S : t & 8 ? w : t & 16 ? S.body : e, Ye = (e) => st ? {
  passive: (e & 1) !== 0,
  capture: (e & 2) !== 0
} : (e & 2) !== 0, ge = /* @__PURE__ */ new WeakMap(), R = (e) => ge.get(e), Ve = (e, t) => {
  const n = {
    $flags$: 0,
    $hostElement$: e,
    $cmpMeta$: t,
    $instanceValues$: /* @__PURE__ */ new Map()
  };
  return pe(e, n, t.$listeners$), ge.set(e, n);
}, te = (e, t) => t in e, O = (e, t) => (0, console.error)(e, t), M = /* @__PURE__ */ new Map(), et = [], w = typeof window < "u" ? window : {}, S = w.document || { head: {} }, tt = w.HTMLElement || class {
}, p = {
  $flags$: 0,
  $resourcesUrl$: "",
  jmp: (e) => e(),
  raf: (e) => requestAnimationFrame(e),
  ael: (e, t, n, s) => e.addEventListener(t, n, s),
  rel: (e, t, n, s) => e.removeEventListener(t, n, s),
  ce: (e, t) => new CustomEvent(e, t)
}, st = /* @__PURE__ */ (() => {
  let e = !1;
  try {
    S.addEventListener("e", null, Object.defineProperty({}, "passive", {
      get() {
        e = !0;
      }
    }));
  } catch {
  }
  return e;
})(), nt = (e) => Promise.resolve(e), lt = /* @__PURE__ */ (() => {
  try {
    return new CSSStyleSheet(), typeof new CSSStyleSheet().replaceSync == "function";
  } catch {
  }
  return !1;
})(), se = [], ye = [], ot = (e, t) => (n) => {
  e.push(n), D || (D = !0, t && p.$flags$ & 4 ? it(N) : p.raf(N));
}, ne = (e) => {
  for (let t = 0; t < e.length; t++)
    try {
      e[t](performance.now());
    } catch (n) {
      O(n);
    }
  e.length = 0;
}, N = () => {
  ne(se), ne(ye), (D = se.length > 0) && p.raf(N);
}, it = (e) => nt().then(e), ct = /* @__PURE__ */ ot(ye, !0), rt = ".sc-calendar-check-icon-h{display:flex}", at = rt, $t = Ge(class extends tt {
  constructor() {
    super(), this.__registerHost(), this.width = "20", this.height = "20";
  }
  render() {
    return U("svg", { key: "a1c54b1279fece33095b459f3e94555b27f703b7", width: this.width, height: this.height, fill: "none", xmlns: "http://www.w3.org/2000/svg" }, U("path", { key: "cc74a6a2aaf0961c56af2243361bbf9bcda9d999", fill: "currentColor", d: "M14.75 0C16.5449 0 18 1.45507 18 3.25V9.02182C17.5368 8.72526 17.0335 8.48584 16.5 8.3135V5.5H1.5V14.75C1.5 15.7165 2.2835 16.5 3.25 16.5H8.3135C8.48584 17.0335 8.72526 17.5368 9.02182 18H3.25C1.45507 18 0 16.5449 0 14.75V3.25C0 1.45507 1.45507 0 3.25 0H14.75ZM14.75 1.5H3.25C2.2835 1.5 1.5 2.2835 1.5 3.25V4H16.5V3.25C16.5 2.2835 15.7165 1.5 14.75 1.5ZM20 14.5C20 17.5376 17.5376 20 14.5 20C11.4624 20 9 17.5376 9 14.5C9 11.4624 11.4624 9 14.5 9C17.5376 9 20 11.4624 20 14.5ZM17.8536 12.1464C17.6583 11.9512 17.3417 11.9512 17.1464 12.1464L13.5 15.7929L11.8536 14.1464C11.6583 13.9512 11.3417 13.9512 11.1464 14.1464C10.9512 14.3417 10.9512 14.6583 11.1464 14.8536L13.1464 16.8536C13.3417 17.0488 13.6583 17.0488 13.8536 16.8536L17.8536 12.8536C18.0488 12.6583 18.0488 12.3417 17.8536 12.1464Z" }));
  }
  static get style() {
    return at;
  }
}, [2, "calendar-check-icon", {
  width: [1],
  height: [1]
}]);
function dt() {
  if (typeof customElements > "u")
    return;
  ["calendar-check-icon"].forEach((t) => {
    switch (t) {
      case "calendar-check-icon":
        customElements.get(t) || customElements.define(t, $t);
        break;
    }
  });
}
export {
  $t as C,
  dt as d
};
