// @nylas/.web-elements-GF2lXkj6/dist/cdn/sent/sent.es.js
const b = {
  allRenderFn: !1,
  cmpDidLoad: !0,
  cmpDidUnload: !1,
  cmpDidUpdate: !0,
  cmpDidRender: !0,
  cmpWillLoad: !0,
  cmpWillUpdate: !0,
  cmpWillRender: !0,
  connectedCallback: !0,
  disconnectedCallback: !0,
  element: !0,
  event: !0,
  hasRenderFn: !0,
  lifecycle: !0,
  hostListener: !0,
  hostListenerTargetWindow: !0,
  hostListenerTargetDocument: !0,
  hostListenerTargetBody: !0,
  hostListenerTargetParent: !1,
  hostListenerTarget: !0,
  member: !0,
  method: !0,
  mode: !0,
  observeAttribute: !0,
  prop: !0,
  propMutable: !0,
  reflect: !0,
  scoped: !0,
  shadowDom: !0,
  slot: !0,
  cssAnnotations: !0,
  state: !0,
  style: !0,
  formAssociated: !1,
  svg: !0,
  updatable: !0,
  vdomAttribute: !0,
  vdomXlink: !0,
  vdomClass: !0,
  vdomFunctional: !0,
  vdomKey: !0,
  vdomListener: !0,
  vdomRef: !0,
  vdomPropOrAttr: !0,
  vdomRender: !0,
  vdomStyle: !0,
  vdomText: !0,
  watchCallback: !0,
  taskQueue: !0,
  hotModuleReplacement: !1,
  isDebug: !1,
  isDev: !1,
  isTesting: !1,
  hydrateServerSide: !1,
  hydrateClientSide: !1,
  lifecycleDOMEvents: !1,
  lazyLoad: !1,
  profile: !1,
  slotRelocation: !0,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  appendChildSlotFix: !1,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  cloneNodeFix: !1,
  hydratedAttribute: !1,
  hydratedClass: !0,
  scriptDataOpts: !1,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  scopedSlotTextContentFix: !1,
  // TODO(STENCIL-854): Remove code related to legacy shadowDomShim field
  shadowDomShim: !1,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  slotChildNodesFix: !1,
  invisiblePrehydration: !0,
  propBoolean: !0,
  propNumber: !0,
  propString: !0,
  constructableCSS: !0,
  cmpShouldUpdate: !0,
  devTools: !1,
  shadowDelegatesFocus: !0,
  initializeNextTick: !1,
  asyncLoading: !1,
  asyncQueue: !1,
  transformTagName: !1,
  attachStyles: !0,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  experimentalSlotFixes: !1
};
let x, M, R, oe = !1, U = !1, W = !1, v = !1, I = null, H = !1;
const L = (e, t = "") => () => {
}, ve = "slot-fb{display:contents}slot-fb[hidden]{display:none}", Q = "http://www.w3.org/1999/xlink", X = {}, Se = "http://www.w3.org/2000/svg", ke = "http://www.w3.org/1999/xhtml", me = (e) => e != null, z = (e) => (e = typeof e, e === "object" || e === "function");
function be(e) {
  var t, n, s;
  return (s = (n = (t = e.head) === null || t === void 0 ? void 0 : t.querySelector('meta[name="csp-nonce"]')) === null || n === void 0 ? void 0 : n.getAttribute("content")) !== null && s !== void 0 ? s : void 0;
}
const _ = (e, t, ...n) => {
  let s = null, o = null, r = null, l = !1, c = !1;
  const i = [], u = (a) => {
    for (let d = 0; d < a.length; d++)
      s = a[d], Array.isArray(s) ? u(s) : s != null && typeof s != "boolean" && ((l = typeof e != "function" && !z(s)) && (s = String(s)), l && c ? i[i.length - 1].$text$ += s : i.push(l ? P(null, s) : s), c = l);
  };
  if (u(n), t) {
    t.key && (o = t.key), t.name && (r = t.name);
    {
      const a = t.className || t.class;
      a && (t.class = typeof a != "object" ? a : Object.keys(a).filter((d) => a[d]).join(" "));
    }
  }
  if (typeof e == "function")
    return e(t === null ? {} : t, i, xe);
  const f = P(e, null);
  return f.$attrs$ = t, i.length > 0 && (f.$children$ = i), f.$key$ = o, f.$name$ = r, f;
}, P = (e, t) => {
  const n = {
    $flags$: 0,
    $tag$: e,
    $text$: t,
    $elm$: null,
    $children$: null
  };
  return n.$attrs$ = null, n.$key$ = null, n.$name$ = null, n;
}, Te = {}, Le = (e) => e && e.$tag$ === Te, xe = {
  forEach: (e, t) => e.map(G).forEach(t),
  map: (e, t) => e.map(G).map(t).map(Ee)
}, G = (e) => ({
  vattrs: e.$attrs$,
  vchildren: e.$children$,
  vkey: e.$key$,
  vname: e.$name$,
  vtag: e.$tag$,
  vtext: e.$text$
}), Ee = (e) => {
  if (typeof e.vtag == "function") {
    const n = Object.assign({}, e.vattrs);
    return e.vkey && (n.key = e.vkey), e.vname && (n.name = e.vname), _(e.vtag, n, ...e.vchildren || []);
  }
  const t = P(e.vtag, e.vtext);
  return t.$attrs$ = e.vattrs, t.$children$ = e.vchildren, t.$key$ = e.vkey, t.$name$ = e.vname, t;
}, _e = (e) => et.map((t) => t(e)).find((t) => !!t), Ae = (e, t) => e != null && !z(e) ? t & 4 ? e === "false" ? !1 : e === "" || !!e : t & 2 ? parseFloat(e) : t & 1 ? String(e) : e : e, K = /* @__PURE__ */ new WeakMap(), Re = (e, t, n) => {
  let s = B.get(e);
  ot && n ? (s = s || new CSSStyleSheet(), typeof s == "string" ? s = t : s.replaceSync(t)) : s = t, B.set(e, s);
}, Oe = (e, t, n) => {
  var s;
  const o = le(t, n), r = B.get(o);
  if (e = e.nodeType === 11 ? e : S, r)
    if (typeof r == "string") {
      e = e.head || e;
      let l = K.get(e), c;
      if (l || K.set(e, l = /* @__PURE__ */ new Set()), !l.has(o)) {
        {
          c = S.createElement("style"), c.innerHTML = r;
          const i = (s = p.$nonce$) !== null && s !== void 0 ? s : be(S);
          i != null && c.setAttribute("nonce", i), e.insertBefore(c, e.querySelector("link"));
        }
        t.$flags$ & 4 && (c.innerHTML += ve), l && l.add(o);
      }
    } else e.adoptedStyleSheets.includes(r) || (e.adoptedStyleSheets = [...e.adoptedStyleSheets, r]);
  return o;
}, we = (e) => {
  const t = e.$cmpMeta$, n = e.$hostElement$, s = t.$flags$, o = L("attachStyles", t.$tagName$), r = Oe(n.shadowRoot ? n.shadowRoot : n.getRootNode(), t, e.$modeName$);
  s & 10 && (n["s-sc"] = r, n.classList.add(r + "-h"), s & 2 && n.classList.add(r + "-s")), o();
}, le = (e, t) => "sc-" + (t && e.$flags$ & 32 ? e.$tagName$ + "-" + t : e.$tagName$), J = (e, t, n, s, o, r) => {
  if (n !== s) {
    let l = te(e, t), c = t.toLowerCase();
    if (t === "class") {
      const i = e.classList, u = Y(n), f = Y(s);
      i.remove(...u.filter((a) => a && !f.includes(a))), i.add(...f.filter((a) => a && !u.includes(a)));
    } else if (t === "style") {
      for (const i in n)
        (!s || s[i] == null) && (i.includes("-") ? e.style.removeProperty(i) : e.style[i] = "");
      for (const i in s)
        (!n || s[i] !== n[i]) && (i.includes("-") ? e.style.setProperty(i, s[i]) : e.style[i] = s[i]);
    } else if (t !== "key")
      if (t === "ref")
        s && s(e);
      else if (!e.__lookupSetter__(t) && t[0] === "o" && t[1] === "n") {
        if (t[2] === "-" ? t = t.slice(3) : te(D, c) ? t = c.slice(2) : t = c[2] + t.slice(3), n || s) {
          const i = t.endsWith(ie);
          t = t.replace(Fe, ""), n && p.rel(e, t, n, i), s && p.ael(e, t, s, i);
        }
      } else {
        const i = z(s);
        if ((l || i && s !== null) && !o)
          try {
            if (e.tagName.includes("-"))
              e[t] = s;
            else {
              const f = s ?? "";
              t === "list" ? l = !1 : (n == null || e[t] != f) && (e[t] = f);
            }
          } catch {
          }
        let u = !1;
        c !== (c = c.replace(/^xlink\:?/, "")) && (t = c, u = !0), s == null || s === !1 ? (s !== !1 || e.getAttribute(t) === "") && (u ? e.removeAttributeNS(Q, t) : e.removeAttribute(t)) : (!l || r & 4 || o) && !i && (s = s === !0 ? "" : s, u ? e.setAttributeNS(Q, t, s) : e.setAttribute(t, s));
      }
  }
}, Ce = /\s/, Y = (e) => e ? e.split(Ce) : [], ie = "Capture", Fe = new RegExp(ie + "$"), re = (e, t, n, s) => {
  const o = t.$elm$.nodeType === 11 && t.$elm$.host ? t.$elm$.host : t.$elm$, r = e && e.$attrs$ || X, l = t.$attrs$ || X;
  for (s in r)
    s in l || J(o, s, r[s], void 0, n, t.$flags$);
  for (s in l)
    J(o, s, r[s], l[s], n, t.$flags$);
}, j = (e, t, n, s) => {
  const o = t.$children$[n];
  let r = 0, l, c, i;
  if (oe || (W = !0, o.$tag$ === "slot" && (x && s.classList.add(x + "-s"), o.$flags$ |= o.$children$ ? (
    // slot element has fallback content
    2
  ) : (
    // slot element does not have fallback content
    1
  ))), o.$text$ !== null)
    l = o.$elm$ = S.createTextNode(o.$text$);
  else if (o.$flags$ & 1)
    l = o.$elm$ = S.createTextNode("");
  else {
    if (v || (v = o.$tag$ === "svg"), l = o.$elm$ = S.createElementNS(v ? Se : ke, o.$flags$ & 2 ? "slot-fb" : o.$tag$), v && o.$tag$ === "foreignObject" && (v = !1), re(null, o, v), me(x) && l["s-si"] !== x && l.classList.add(l["s-si"] = x), o.$children$)
      for (r = 0; r < o.$children$.length; ++r)
        c = j(e, o, r, l), c && l.appendChild(c);
    o.$tag$ === "svg" ? v = !1 : l.tagName === "foreignObject" && (v = !0);
  }
  return l["s-hn"] = R, o.$flags$ & 3 && (l["s-sr"] = !0, l["s-cr"] = M, l["s-sn"] = o.$name$ || "", i = e && e.$children$ && e.$children$[n], i && i.$tag$ === o.$tag$ && e.$elm$ && O(e.$elm$, !1)), l;
}, O = (e, t) => {
  p.$flags$ |= 1;
  const n = e.childNodes;
  for (let s = n.length - 1; s >= 0; s--) {
    const o = n[s];
    o["s-hn"] !== R && o["s-ol"] && ($e(o).insertBefore(o, q(o)), o["s-ol"].remove(), o["s-ol"] = void 0, o["s-sh"] = void 0, W = !0), t && O(o, t);
  }
  p.$flags$ &= -2;
}, ce = (e, t, n, s, o, r) => {
  let l = e["s-cr"] && e["s-cr"].parentNode || e, c;
  for (l.shadowRoot && l.tagName === R && (l = l.shadowRoot); o <= r; ++o)
    s[o] && (c = j(null, n, o, e), c && (s[o].$elm$ = c, l.insertBefore(c, q(t))));
}, ae = (e, t, n) => {
  for (let s = t; s <= n; ++s) {
    const o = e[s];
    if (o) {
      const r = o.$elm$;
      ue(o), r && (U = !0, r["s-ol"] ? r["s-ol"].remove() : O(r, !0), r.remove());
    }
  }
}, Ue = (e, t, n, s, o = !1) => {
  let r = 0, l = 0, c = 0, i = 0, u = t.length - 1, f = t[0], a = t[u], d = s.length - 1, $ = s[0], h = s[d], y, g;
  for (; r <= u && l <= d; )
    if (f == null)
      f = t[++r];
    else if (a == null)
      a = t[--u];
    else if ($ == null)
      $ = s[++l];
    else if (h == null)
      h = s[--d];
    else if (F(f, $, o))
      E(f, $, o), f = t[++r], $ = s[++l];
    else if (F(a, h, o))
      E(a, h, o), a = t[--u], h = s[--d];
    else if (F(f, h, o))
      (f.$tag$ === "slot" || h.$tag$ === "slot") && O(f.$elm$.parentNode, !1), E(f, h, o), e.insertBefore(f.$elm$, a.$elm$.nextSibling), f = t[++r], h = s[--d];
    else if (F(a, $, o))
      (f.$tag$ === "slot" || h.$tag$ === "slot") && O(a.$elm$.parentNode, !1), E(a, $, o), e.insertBefore(a.$elm$, f.$elm$), a = t[--u], $ = s[++l];
    else {
      for (c = -1, i = r; i <= u; ++i)
        if (t[i] && t[i].$key$ !== null && t[i].$key$ === $.$key$) {
          c = i;
          break;
        }
      c >= 0 ? (g = t[c], g.$tag$ !== $.$tag$ ? y = j(t && t[l], n, c, e) : (E(g, $, o), t[c] = void 0, y = g.$elm$), $ = s[++l]) : (y = j(t && t[l], n, l, e), $ = s[++l]), y && $e(f.$elm$).insertBefore(y, q(f.$elm$));
    }
  r > u ? ce(e, s[d + 1] == null ? null : s[d + 1].$elm$, n, s, l, d) : l > d && ae(t, r, u);
}, F = (e, t, n = !1) => e.$tag$ === t.$tag$ ? e.$tag$ === "slot" ? e.$name$ === t.$name$ : n ? !0 : e.$key$ === t.$key$ : !1, q = (e) => e && e["s-ol"] || e, $e = (e) => (e["s-ol"] ? e["s-ol"] : e).parentNode, E = (e, t, n = !1) => {
  const s = t.$elm$ = e.$elm$, o = e.$children$, r = t.$children$, l = t.$tag$, c = t.$text$;
  let i;
  c === null ? (v = l === "svg" ? !0 : l === "foreignObject" ? !1 : v, l === "slot" || re(e, t, v), o !== null && r !== null ? Ue(s, o, t, r, n) : r !== null ? (e.$text$ !== null && (s.textContent = ""), ce(s, null, t, r, 0, r.length - 1)) : o !== null && ae(o, 0, o.length - 1), v && l === "svg" && (v = !1)) : (i = s["s-cr"]) ? i.parentNode.textContent = c : e.$text$ !== c && (s.data = c);
}, fe = (e) => {
  const t = e.childNodes;
  for (const n of t)
    if (n.nodeType === 1) {
      if (n["s-sr"]) {
        const s = n["s-sn"];
        n.hidden = !1;
        for (const o of t)
          if (o !== n) {
            if (o["s-hn"] !== n["s-hn"] || s !== "") {
              if (o.nodeType === 1 && (s === o.getAttribute("slot") || s === o["s-sn"])) {
                n.hidden = !0;
                break;
              }
            } else if (o.nodeType === 1 || o.nodeType === 3 && o.textContent.trim() !== "") {
              n.hidden = !0;
              break;
            }
          }
      }
      fe(n);
    }
}, k = [], de = (e) => {
  let t, n, s;
  for (const o of e.childNodes) {
    if (o["s-sr"] && (t = o["s-cr"]) && t.parentNode) {
      n = t.parentNode.childNodes;
      const r = o["s-sn"];
      for (s = n.length - 1; s >= 0; s--)
        if (t = n[s], !t["s-cn"] && !t["s-nr"] && t["s-hn"] !== o["s-hn"] && !b.experimentalSlotFixes)
          if (Z(t, r)) {
            let l = k.find((c) => c.$nodeToRelocate$ === t);
            U = !0, t["s-sn"] = t["s-sn"] || r, l ? (l.$nodeToRelocate$["s-sh"] = o["s-hn"], l.$slotRefNode$ = o) : (t["s-sh"] = o["s-hn"], k.push({
              $slotRefNode$: o,
              $nodeToRelocate$: t
            })), t["s-sr"] && k.map((c) => {
              Z(c.$nodeToRelocate$, t["s-sn"]) && (l = k.find((i) => i.$nodeToRelocate$ === t), l && !c.$slotRefNode$ && (c.$slotRefNode$ = l.$slotRefNode$));
            });
          } else k.some((l) => l.$nodeToRelocate$ === t) || k.push({
            $nodeToRelocate$: t
          });
    }
    o.nodeType === 1 && de(o);
  }
}, Z = (e, t) => e.nodeType === 1 ? e.getAttribute("slot") === null && t === "" || e.getAttribute("slot") === t : e["s-sn"] === t ? !0 : t === "", ue = (e) => {
  e.$attrs$ && e.$attrs$.ref && e.$attrs$.ref(null), e.$children$ && e.$children$.map(ue);
}, Pe = (e, t, n = !1) => {
  var s, o, r, l, c;
  const i = e.$hostElement$, u = e.$cmpMeta$, f = e.$vnode$ || P(null, null), a = Le(t) ? t : _(null, null, t);
  if (R = i.tagName, u.$attrsToReflect$ && (a.$attrs$ = a.$attrs$ || {}, u.$attrsToReflect$.map(([d, $]) => a.$attrs$[$] = i[d])), n && a.$attrs$)
    for (const d of Object.keys(a.$attrs$))
      i.hasAttribute(d) && !["key", "ref", "style", "class"].includes(d) && (a.$attrs$[d] = i[d]);
  a.$tag$ = null, a.$flags$ |= 4, e.$vnode$ = a, a.$elm$ = f.$elm$ = i.shadowRoot || i, x = i["s-sc"], M = i["s-cr"], oe = (u.$flags$ & 1) !== 0, U = !1, E(f, a, n);
  {
    if (p.$flags$ |= 1, W) {
      de(a.$elm$);
      for (const d of k) {
        const $ = d.$nodeToRelocate$;
        if (!$["s-ol"]) {
          const h = S.createTextNode("");
          h["s-nr"] = $, $.parentNode.insertBefore($["s-ol"] = h, $);
        }
      }
      for (const d of k) {
        const $ = d.$nodeToRelocate$, h = d.$slotRefNode$;
        if (h) {
          const y = h.parentNode;
          let g = h.nextSibling;
          {
            let T = (s = $["s-ol"]) === null || s === void 0 ? void 0 : s.previousSibling;
            for (; T; ) {
              let m = (o = T["s-nr"]) !== null && o !== void 0 ? o : null;
              if (m && m["s-sn"] === $["s-sn"] && y === m.parentNode && (m = m.nextSibling, !m || !m["s-nr"])) {
                g = m;
                break;
              }
              T = T.previousSibling;
            }
          }
          (!g && y !== $.parentNode || $.nextSibling !== g) && $ !== g && (!$["s-hn"] && $["s-ol"] && ($["s-hn"] = $["s-ol"].parentNode.nodeName), y.insertBefore($, g), $.nodeType === 1 && ($.hidden = (r = $["s-ih"]) !== null && r !== void 0 ? r : !1));
        } else
          $.nodeType === 1 && (n && ($["s-ih"] = (l = $.hidden) !== null && l !== void 0 ? l : !1), $.hidden = !0);
      }
    }
    U && fe(a.$elm$), p.$flags$ &= -2, k.length = 0;
  }
  if (b.experimentalScopedSlotChanges && u.$flags$ & 2)
    for (const d of a.$elm$.childNodes)
      d["s-hn"] !== R && !d["s-sh"] && (n && d["s-ih"] == null && (d["s-ih"] = (c = d.hidden) !== null && c !== void 0 ? c : !1), d.hidden = !0);
  M = void 0;
}, je = (e, t) => {
}, he = (e, t) => (e.$flags$ |= 16, je(e, e.$ancestorComponent$), rt(() => Be(e, t))), Be = (e, t) => {
  const n = e.$hostElement$, s = L("scheduleUpdate", e.$cmpMeta$.$tagName$), o = n;
  let r;
  return t ? r = A(o, "componentWillLoad") : r = A(o, "componentWillUpdate"), r = V(r, () => A(o, "componentWillRender")), s(), V(r, () => Me(e, o, t));
}, V = (e, t) => De(e) ? e.then(t) : t(), De = (e) => e instanceof Promise || e && e.then && typeof e.then == "function", Me = async (e, t, n) => {
  const s = e.$hostElement$, o = L("update", e.$cmpMeta$.$tagName$);
  s["s-rc"], n && we(e);
  const r = L("render", e.$cmpMeta$.$tagName$);
  He(e, t, s, n), r(), o(), Ne(e);
}, He = (e, t, n, s) => {
  try {
    I = t, t = t.render && t.render(), e.$flags$ &= -17, e.$flags$ |= 2, (b.hasRenderFn || b.reflect) && (b.vdomRender || b.reflect) && (b.hydrateServerSide || Pe(e, t, s));
  } catch (i) {
    C(i, e.$hostElement$);
  }
  return I = null, null;
}, Ne = (e) => {
  const t = e.$cmpMeta$.$tagName$, n = e.$hostElement$, s = L("postUpdate", t), o = n;
  e.$ancestorComponent$, A(o, "componentDidRender"), e.$flags$ & 64 ? (A(o, "componentDidUpdate"), s()) : (e.$flags$ |= 64, A(o, "componentDidLoad"), s());
}, A = (e, t, n) => {
  if (e && e[t])
    try {
      return e[t](n);
    } catch (s) {
      C(s);
    }
}, We = (e, t) => w(e).$instanceValues$.get(t), ze = (e, t, n, s) => {
  const o = w(e), r = e, l = o.$instanceValues$.get(t), c = o.$flags$, i = r;
  n = Ae(n, s.$members$[t][0]);
  const u = Number.isNaN(l) && Number.isNaN(n);
  if (n !== l && !u) {
    o.$instanceValues$.set(t, n);
    {
      if (s.$watchers$ && c & 128) {
        const a = s.$watchers$[t];
        a && a.map((d) => {
          try {
            i[d](n, l, t);
          } catch ($) {
            C($, r);
          }
        });
      }
      if ((c & 18) === 2) {
        if (i.componentShouldUpdate && i.componentShouldUpdate(n, l, t) === !1)
          return;
        he(o, !1);
      }
    }
  }
}, qe = (e, t, n) => {
  var s;
  const o = e.prototype;
  if (t.$members$) {
    e.watchers && (t.$watchers$ = e.watchers);
    const r = Object.entries(t.$members$);
    r.map(([l, [c]]) => {
      (c & 31 || c & 32) && Object.defineProperty(o, l, {
        get() {
          return We(this, l);
        },
        set(i) {
          ze(this, l, i, t);
        },
        configurable: !0,
        enumerable: !0
      });
    });
    {
      const l = /* @__PURE__ */ new Map();
      o.attributeChangedCallback = function(c, i, u) {
        p.jmp(() => {
          var f;
          const a = l.get(c);
          if (this.hasOwnProperty(a))
            u = this[a], delete this[a];
          else {
            if (o.hasOwnProperty(a) && typeof this[a] == "number" && this[a] == u)
              return;
            if (a == null) {
              const d = w(this), $ = d == null ? void 0 : d.$flags$;
              if ($ && !($ & 8) && $ & 128 && u !== i) {
                const y = this, g = (f = t.$watchers$) === null || f === void 0 ? void 0 : f[c];
                g == null || g.forEach((T) => {
                  y[T] != null && y[T].call(y, u, i, c);
                });
              }
              return;
            }
          }
          this[a] = u === null && typeof this[a] == "boolean" ? !1 : u;
        });
      }, e.observedAttributes = Array.from(/* @__PURE__ */ new Set([
        ...Object.keys((s = t.$watchers$) !== null && s !== void 0 ? s : {}),
        ...r.filter(
          ([c, i]) => i[0] & 15
          /* MEMBER_FLAGS.HasAttribute */
        ).map(([c, i]) => {
          var u;
          const f = i[1] || c;
          return l.set(f, c), i[0] & 512 && ((u = t.$attrsToReflect$) === null || u === void 0 || u.push([c, f])), f;
        })
      ]));
    }
  }
  return e;
}, Ie = async (e, t, n, s) => {
  let o;
  if (!(t.$flags$ & 32) && (t.$flags$ |= 32, o = e.constructor, customElements.whenDefined(n.$tagName$).then(() => t.$flags$ |= 128), o.style)) {
    let l = o.style;
    typeof l != "string" && (l = l[t.$modeName$ = _e(e)]);
    const c = le(n, t.$modeName$);
    if (!B.has(c)) {
      const i = L("registerStyles", n.$tagName$);
      Re(c, l, !!(n.$flags$ & 1)), i();
    }
  }
  t.$ancestorComponent$, he(t, !0);
}, ee = (e) => {
}, Qe = (e) => {
  if (!(p.$flags$ & 1)) {
    const t = w(e), n = t.$cmpMeta$, s = L("connectedCallback", n.$tagName$);
    t.$flags$ & 1 ? (pe(e, t, n.$listeners$), t != null && t.$lazyInstance$ ? ee(t.$lazyInstance$) : t != null && t.$onReadyPromise$ && t.$onReadyPromise$.then(() => ee(t.$lazyInstance$))) : (t.$flags$ |= 1, // TODO(STENCIL-854): Remove code related to legacy shadowDomShim field
    n.$flags$ & 12 && Xe(e), n.$members$ && Object.entries(n.$members$).map(([o, [r]]) => {
      if (r & 31 && e.hasOwnProperty(o)) {
        const l = e[o];
        delete e[o], e[o] = l;
      }
    }), Ie(e, t, n)), s();
  }
}, Xe = (e) => {
  const t = e["s-cr"] = S.createComment("");
  t["s-cn"] = !0, e.insertBefore(t, e.firstChild);
}, Ge = async (e) => {
  if (!(p.$flags$ & 1)) {
    const t = w(e);
    t.$rmListeners$ && (t.$rmListeners$.map((n) => n()), t.$rmListeners$ = void 0);
  }
}, Ke = (e, t) => {
  const n = {
    $flags$: t[0],
    $tagName$: t[1]
  };
  n.$members$ = t[2], n.$listeners$ = t[3], n.$watchers$ = e.$watchers$, n.$attrsToReflect$ = [];
  const s = e.prototype.connectedCallback, o = e.prototype.disconnectedCallback;
  return Object.assign(e.prototype, {
    __registerHost() {
      Ve(this, n);
    },
    connectedCallback() {
      Qe(this), s && s.call(this);
    },
    disconnectedCallback() {
      Ge(this), o && o.call(this);
    },
    __attachShadow() {
      this.attachShadow({
        mode: "open",
        delegatesFocus: !!(n.$flags$ & 16)
      });
    }
  }), e.is = n.$tagName$, qe(e, n);
}, pe = (e, t, n, s) => {
  n && n.map(([o, r, l]) => {
    const c = Ye(e, o), i = Je(t, l), u = Ze(o);
    p.ael(c, r, i, u), (t.$rmListeners$ = t.$rmListeners$ || []).push(() => p.rel(c, r, i, u));
  });
}, Je = (e, t) => (n) => {
  try {
    b.lazyLoad || e.$hostElement$[t](n);
  } catch (s) {
    C(s);
  }
}, Ye = (e, t) => t & 4 ? S : t & 8 ? D : t & 16 ? S.body : e, Ze = (e) => st ? {
  passive: (e & 1) !== 0,
  capture: (e & 2) !== 0
} : (e & 2) !== 0, ge = /* @__PURE__ */ new WeakMap(), w = (e) => ge.get(e), Ve = (e, t) => {
  const n = {
    $flags$: 0,
    $hostElement$: e,
    $cmpMeta$: t,
    $instanceValues$: /* @__PURE__ */ new Map()
  };
  return pe(e, n, t.$listeners$), ge.set(e, n);
}, te = (e, t) => t in e, C = (e, t) => (0, console.error)(e, t), B = /* @__PURE__ */ new Map(), et = [], D = typeof window < "u" ? window : {}, S = D.document || { head: {} }, tt = D.HTMLElement || class {
}, p = {
  $flags$: 0,
  $resourcesUrl$: "",
  jmp: (e) => e(),
  raf: (e) => requestAnimationFrame(e),
  ael: (e, t, n, s) => e.addEventListener(t, n, s),
  rel: (e, t, n, s) => e.removeEventListener(t, n, s),
  ce: (e, t) => new CustomEvent(e, t)
}, st = /* @__PURE__ */ (() => {
  let e = !1;
  try {
    S.addEventListener("e", null, Object.defineProperty({}, "passive", {
      get() {
        e = !0;
      }
    }));
  } catch {
  }
  return e;
})(), nt = (e) => Promise.resolve(e), ot = /* @__PURE__ */ (() => {
  try {
    return new CSSStyleSheet(), typeof new CSSStyleSheet().replaceSync == "function";
  } catch {
  }
  return !1;
})(), se = [], ye = [], lt = (e, t) => (n) => {
  e.push(n), H || (H = !0, t && p.$flags$ & 4 ? it(N) : p.raf(N));
}, ne = (e) => {
  for (let t = 0; t < e.length; t++)
    try {
      e[t](performance.now());
    } catch (n) {
      C(n);
    }
  e.length = 0;
}, N = () => {
  ne(se), ne(ye), (H = se.length > 0) && p.raf(N);
}, it = (e) => nt().then(e), rt = /* @__PURE__ */ lt(ye, !0), ct = ".sc-sent-icon-h{display:flex}", at = ct, $t = Ke(class extends tt {
  constructor() {
    super(), this.__registerHost(), this.width = "15", this.height = "15";
  }
  render() {
    return _("svg", { key: "098aa96cca21e440ea4fe8d8363528fe9b1f277f", width: this.width, height: this.height, viewBox: "0 0 24 24", fill: "auto", xmlns: "http://www.w3.org/2000/svg" }, _("path", { key: "b981887697dbb3c37cb32993c88324bea80cc098", d: "M21.5 12L3.5 3.5L3.5 20.5L21.5 12Z", stroke: "auto", "stroke-width": "1.5" }), _("path", { key: "e7985ddf523274e49faee8da881e857a3aaca1ed", d: "M3.5 3.5L12 10.5", stroke: "auto", "stroke-width": "1.5" }), _("path", { key: "42a0ba5483a0cc9d79c1215bc5e0f31a406d256d", d: "M3.5 20.5L12 13.5", stroke: "auto", "stroke-width": "1.5" }));
  }
  static get style() {
    return at;
  }
}, [2, "sent-icon", {
  width: [1],
  height: [1]
}]);
function dt() {
  if (typeof customElements > "u")
    return;
  ["sent-icon"].forEach((t) => {
    switch (t) {
      case "sent-icon":
        customElements.get(t) || customElements.define(t, $t);
        break;
    }
  });
}
export {
  $t as S,
  dt as d
};
