// @nylas/.web-elements-GF2lXkj6/dist/collection/stores/scheduler-store.js
import { debug } from "../utils/utils";
import { createStore } from "@stencil/store";
export function CreateNylasSchedulerStore(defaultState = {}) {
    const defaultNylasStoreState = {
        selectedDate: null,
        selectedLanguage: navigator.language,
        selectedTimezone: Intl.DateTimeFormat().resolvedOptions().timeZone,
        selectedTimeslot: null,
        showBookingForm: false,
        availabilityOrderEmails: [],
        selectableDates: null,
        availability: [],
        eventDuration: 0,
        state: 'ready',
        eventInfo: null,
        cancelledEventInfo: null,
        isLoading: false,
        nylasBranding: true,
        ...defaultState,
    };
    debug(`[defaultNylasStoreState]: `, defaultNylasStoreState);
    const store = createStore(defaultNylasStoreState);
    store.onChange('availability', availability => {
        debug(`[availability]: `, availability);
        const selectableDates = availability.map(timeslot => timeslot.start_time);
        debug(`[selectableDates]: `, selectableDates);
        store.set('selectableDates', selectableDates);
        const timeslot = availability[0];
        if (!timeslot)
            return;
        const durationMinutes = Math.floor((timeslot.end_time.getTime() - timeslot.start_time.getTime()) / 60000);
        debug(`[durationMinutes]: `, durationMinutes);
        store.set('eventDuration', durationMinutes);
    });
    store.reset = () => {
        for (const key in defaultNylasStoreState) {
            const value = defaultNylasStoreState[key];
            store.set(key, value);
        }
    };
    return store;
}
