// @nylas/.web-elements-GF2lXkj6/dist/collection/components/scheduler-editor/nylas-event-info/nylas-event-info.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
        return Reflect.metadata(k, v);
};
import { RegisterComponent } from "../../../common/register-component";
import { checkForMissingSlots, debug } from "../../../utils/utils";
import { Host, h } from "@stencil/core";
export class NylasEventInfo {
    connectedCallback() {
        debug('nylas-event-info', 'connectedCallback');
    }
    componentWillLoad() {
        debug('nylas-event-info', 'componentWillLoad');
    }
    componentDidLoad() {
        debug('nylas-event-info', 'componentDidLoad');
        checkForMissingSlots(['inputs'], this.el);
    }
    disconnectedCallback() {
        debug('nylas-event-info', 'disconnectedCallback');
    }
    render() {
        return (h(Host, { key: '99e646983326861e86df8fc8aaa731a22122e4ee' }, h("nylas-form-card", { key: 'd55c0c68e265458e464a077487cfcc4a34139467', exportparts: "nfc__header: nei__header" }, h("h1", { key: '2c934774492d57e6aa1e3a686a5712b23fa6ba6d', slot: "header-title", class: "nylas-event-info__title", part: "nei__title" }, "Event information"), h("div", { key: '89f21683e0ea7624eb8b9bcaa769cf4afd8e3337', slot: "content", class: "nylas-event-info__content" }, h("slot", { key: '8963ff3ffbb8309fb497ada00ddab86821726481', name: "inputs" })))));
    }
    static get is() { return "nylas-event-info"; }
    static get encapsulation() { return "shadow"; }
    static get formAssociated() { return true; }
    static get originalStyleUrls() {
        return {
            "$": ["nylas-event-info.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["nylas-event-info.css"]
        };
    }
    static get elementRef() { return "el"; }
    static get attachInternalsMemberName() { return "internals"; }
}
__decorate([
    RegisterComponent({
        name: 'nylas-event-info',
        eventToProps: {},
        fireRegisterEvent: true,
    }),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", []),
    __metadata("design:returntype", void 0)
], NylasEventInfo.prototype, "render", null);
