// @nylas/.web-elements-GF2lXkj6/dist/cdn/refresh/refresh.es.js
const T = {
  allRenderFn: !1,
  cmpDidLoad: !0,
  cmpDidUnload: !1,
  cmpDidUpdate: !0,
  cmpDidRender: !0,
  cmpWillLoad: !0,
  cmpWillUpdate: !0,
  cmpWillRender: !0,
  connectedCallback: !0,
  disconnectedCallback: !0,
  element: !0,
  event: !0,
  hasRenderFn: !0,
  lifecycle: !0,
  hostListener: !0,
  hostListenerTargetWindow: !0,
  hostListenerTargetDocument: !0,
  hostListenerTargetBody: !0,
  hostListenerTargetParent: !1,
  hostListenerTarget: !0,
  member: !0,
  method: !0,
  mode: !0,
  observeAttribute: !0,
  prop: !0,
  propMutable: !0,
  reflect: !0,
  scoped: !0,
  shadowDom: !0,
  slot: !0,
  cssAnnotations: !0,
  state: !0,
  style: !0,
  formAssociated: !1,
  svg: !0,
  updatable: !0,
  vdomAttribute: !0,
  vdomXlink: !0,
  vdomClass: !0,
  vdomFunctional: !0,
  vdomKey: !0,
  vdomListener: !0,
  vdomRef: !0,
  vdomPropOrAttr: !0,
  vdomRender: !0,
  vdomStyle: !0,
  vdomText: !0,
  watchCallback: !0,
  taskQueue: !0,
  hotModuleReplacement: !1,
  isDebug: !1,
  isDev: !1,
  isTesting: !1,
  hydrateServerSide: !1,
  hydrateClientSide: !1,
  lifecycleDOMEvents: !1,
  lazyLoad: !1,
  profile: !1,
  slotRelocation: !0,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  appendChildSlotFix: !1,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  cloneNodeFix: !1,
  hydratedAttribute: !1,
  hydratedClass: !0,
  scriptDataOpts: !1,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  scopedSlotTextContentFix: !1,
  // TODO(STENCIL-854): Remove code related to legacy shadowDomShim field
  shadowDomShim: !1,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  slotChildNodesFix: !1,
  invisiblePrehydration: !0,
  propBoolean: !0,
  propNumber: !0,
  propString: !0,
  constructableCSS: !0,
  cmpShouldUpdate: !0,
  devTools: !1,
  shadowDelegatesFocus: !0,
  initializeNextTick: !1,
  asyncLoading: !1,
  asyncQueue: !1,
  transformTagName: !1,
  attachStyles: !0,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  experimentalSlotFixes: !1
};
let L, M, _, le = !1, U = !1, N = !1, v = !1, I = null, H = !1;
const x = (e, t = "") => () => {
}, ve = "slot-fb{display:contents}slot-fb[hidden]{display:none}", Q = "http://www.w3.org/1999/xlink", X = {}, Se = "http://www.w3.org/2000/svg", me = "http://www.w3.org/1999/xhtml", ke = (e) => e != null, W = (e) => (e = typeof e, e === "object" || e === "function");
function Te(e) {
  var t, n, s;
  return (s = (n = (t = e.head) === null || t === void 0 ? void 0 : t.querySelector('meta[name="csp-nonce"]')) === null || n === void 0 ? void 0 : n.getAttribute("content")) !== null && s !== void 0 ? s : void 0;
}
const P = (e, t, ...n) => {
  let s = null, l = null, c = null, o = !1, r = !1;
  const i = [], u = ($) => {
    for (let d = 0; d < $.length; d++)
      s = $[d], Array.isArray(s) ? u(s) : s != null && typeof s != "boolean" && ((o = typeof e != "function" && !W(s)) && (s = String(s)), o && r ? i[i.length - 1].$text$ += s : i.push(o ? j(null, s) : s), r = o);
  };
  if (u(n), t) {
    t.key && (l = t.key), t.name && (c = t.name);
    {
      const $ = t.className || t.class;
      $ && (t.class = typeof $ != "object" ? $ : Object.keys($).filter((d) => $[d]).join(" "));
    }
  }
  if (typeof e == "function")
    return e(t === null ? {} : t, i, Le);
  const f = j(e, null);
  return f.$attrs$ = t, i.length > 0 && (f.$children$ = i), f.$key$ = l, f.$name$ = c, f;
}, j = (e, t) => {
  const n = {
    $flags$: 0,
    $tag$: e,
    $text$: t,
    $elm$: null,
    $children$: null
  };
  return n.$attrs$ = null, n.$key$ = null, n.$name$ = null, n;
}, be = {}, xe = (e) => e && e.$tag$ === be, Le = {
  forEach: (e, t) => e.map(G).forEach(t),
  map: (e, t) => e.map(G).map(t).map(Ee)
}, G = (e) => ({
  vattrs: e.$attrs$,
  vchildren: e.$children$,
  vkey: e.$key$,
  vname: e.$name$,
  vtag: e.$tag$,
  vtext: e.$text$
}), Ee = (e) => {
  if (typeof e.vtag == "function") {
    const n = Object.assign({}, e.vattrs);
    return e.vkey && (n.key = e.vkey), e.vname && (n.name = e.vname), P(e.vtag, n, ...e.vchildren || []);
  }
  const t = j(e.vtag, e.vtext);
  return t.$attrs$ = e.vattrs, t.$children$ = e.vchildren, t.$key$ = e.vkey, t.$name$ = e.vname, t;
}, Re = (e) => et.map((t) => t(e)).find((t) => !!t), _e = (e, t) => e != null && !W(e) ? t & 4 ? e === "false" ? !1 : e === "" || !!e : t & 2 ? parseFloat(e) : t & 1 ? String(e) : e : e, K = /* @__PURE__ */ new WeakMap(), Ae = (e, t, n) => {
  let s = B.get(e);
  lt && n ? (s = s || new CSSStyleSheet(), typeof s == "string" ? s = t : s.replaceSync(t)) : s = t, B.set(e, s);
}, Oe = (e, t, n) => {
  var s;
  const l = oe(t, n), c = B.get(l);
  if (e = e.nodeType === 11 ? e : S, c)
    if (typeof c == "string") {
      e = e.head || e;
      let o = K.get(e), r;
      if (o || K.set(e, o = /* @__PURE__ */ new Set()), !o.has(l)) {
        {
          r = S.createElement("style"), r.innerHTML = c;
          const i = (s = p.$nonce$) !== null && s !== void 0 ? s : Te(S);
          i != null && r.setAttribute("nonce", i), e.insertBefore(r, e.querySelector("link"));
        }
        t.$flags$ & 4 && (r.innerHTML += ve), o && o.add(l);
      }
    } else e.adoptedStyleSheets.includes(c) || (e.adoptedStyleSheets = [...e.adoptedStyleSheets, c]);
  return l;
}, Ce = (e) => {
  const t = e.$cmpMeta$, n = e.$hostElement$, s = t.$flags$, l = x("attachStyles", t.$tagName$), c = Oe(n.shadowRoot ? n.shadowRoot : n.getRootNode(), t, e.$modeName$);
  s & 10 && (n["s-sc"] = c, n.classList.add(c + "-h"), s & 2 && n.classList.add(c + "-s")), l();
}, oe = (e, t) => "sc-" + (t && e.$flags$ & 32 ? e.$tagName$ + "-" + t : e.$tagName$), J = (e, t, n, s, l, c) => {
  if (n !== s) {
    let o = te(e, t), r = t.toLowerCase();
    if (t === "class") {
      const i = e.classList, u = Y(n), f = Y(s);
      i.remove(...u.filter(($) => $ && !f.includes($))), i.add(...f.filter(($) => $ && !u.includes($)));
    } else if (t === "style") {
      for (const i in n)
        (!s || s[i] == null) && (i.includes("-") ? e.style.removeProperty(i) : e.style[i] = "");
      for (const i in s)
        (!n || s[i] !== n[i]) && (i.includes("-") ? e.style.setProperty(i, s[i]) : e.style[i] = s[i]);
    } else if (t !== "key")
      if (t === "ref")
        s && s(e);
      else if (!e.__lookupSetter__(t) && t[0] === "o" && t[1] === "n") {
        if (t[2] === "-" ? t = t.slice(3) : te(D, r) ? t = r.slice(2) : t = r[2] + t.slice(3), n || s) {
          const i = t.endsWith(ie);
          t = t.replace(Ue, ""), n && p.rel(e, t, n, i), s && p.ael(e, t, s, i);
        }
      } else {
        const i = W(s);
        if ((o || i && s !== null) && !l)
          try {
            if (e.tagName.includes("-"))
              e[t] = s;
            else {
              const f = s ?? "";
              t === "list" ? o = !1 : (n == null || e[t] != f) && (e[t] = f);
            }
          } catch {
          }
        let u = !1;
        r !== (r = r.replace(/^xlink\:?/, "")) && (t = r, u = !0), s == null || s === !1 ? (s !== !1 || e.getAttribute(t) === "") && (u ? e.removeAttributeNS(Q, t) : e.removeAttribute(t)) : (!o || c & 4 || l) && !i && (s = s === !0 ? "" : s, u ? e.setAttributeNS(Q, t, s) : e.setAttribute(t, s));
      }
  }
}, Fe = /\s/, Y = (e) => e ? e.split(Fe) : [], ie = "Capture", Ue = new RegExp(ie + "$"), ce = (e, t, n, s) => {
  const l = t.$elm$.nodeType === 11 && t.$elm$.host ? t.$elm$.host : t.$elm$, c = e && e.$attrs$ || X, o = t.$attrs$ || X;
  for (s in c)
    s in o || J(l, s, c[s], void 0, n, t.$flags$);
  for (s in o)
    J(l, s, c[s], o[s], n, t.$flags$);
}, w = (e, t, n, s) => {
  const l = t.$children$[n];
  let c = 0, o, r, i;
  if (le || (N = !0, l.$tag$ === "slot" && (L && s.classList.add(L + "-s"), l.$flags$ |= l.$children$ ? (
    // slot element has fallback content
    2
  ) : (
    // slot element does not have fallback content
    1
  ))), l.$text$ !== null)
    o = l.$elm$ = S.createTextNode(l.$text$);
  else if (l.$flags$ & 1)
    o = l.$elm$ = S.createTextNode("");
  else {
    if (v || (v = l.$tag$ === "svg"), o = l.$elm$ = S.createElementNS(v ? Se : me, l.$flags$ & 2 ? "slot-fb" : l.$tag$), v && l.$tag$ === "foreignObject" && (v = !1), ce(null, l, v), ke(L) && o["s-si"] !== L && o.classList.add(o["s-si"] = L), l.$children$)
      for (c = 0; c < l.$children$.length; ++c)
        r = w(e, l, c, o), r && o.appendChild(r);
    l.$tag$ === "svg" ? v = !1 : o.tagName === "foreignObject" && (v = !0);
  }
  return o["s-hn"] = _, l.$flags$ & 3 && (o["s-sr"] = !0, o["s-cr"] = M, o["s-sn"] = l.$name$ || "", i = e && e.$children$ && e.$children$[n], i && i.$tag$ === l.$tag$ && e.$elm$ && A(e.$elm$, !1)), o;
}, A = (e, t) => {
  p.$flags$ |= 1;
  const n = e.childNodes;
  for (let s = n.length - 1; s >= 0; s--) {
    const l = n[s];
    l["s-hn"] !== _ && l["s-ol"] && (ae(l).insertBefore(l, q(l)), l["s-ol"].remove(), l["s-ol"] = void 0, l["s-sh"] = void 0, N = !0), t && A(l, t);
  }
  p.$flags$ &= -2;
}, re = (e, t, n, s, l, c) => {
  let o = e["s-cr"] && e["s-cr"].parentNode || e, r;
  for (o.shadowRoot && o.tagName === _ && (o = o.shadowRoot); l <= c; ++l)
    s[l] && (r = w(null, n, l, e), r && (s[l].$elm$ = r, o.insertBefore(r, q(t))));
}, $e = (e, t, n) => {
  for (let s = t; s <= n; ++s) {
    const l = e[s];
    if (l) {
      const c = l.$elm$;
      ue(l), c && (U = !0, c["s-ol"] ? c["s-ol"].remove() : A(c, !0), c.remove());
    }
  }
}, Pe = (e, t, n, s, l = !1) => {
  let c = 0, o = 0, r = 0, i = 0, u = t.length - 1, f = t[0], $ = t[u], d = s.length - 1, a = s[0], h = s[d], y, g;
  for (; c <= u && o <= d; )
    if (f == null)
      f = t[++c];
    else if ($ == null)
      $ = t[--u];
    else if (a == null)
      a = s[++o];
    else if (h == null)
      h = s[--d];
    else if (F(f, a, l))
      E(f, a, l), f = t[++c], a = s[++o];
    else if (F($, h, l))
      E($, h, l), $ = t[--u], h = s[--d];
    else if (F(f, h, l))
      (f.$tag$ === "slot" || h.$tag$ === "slot") && A(f.$elm$.parentNode, !1), E(f, h, l), e.insertBefore(f.$elm$, $.$elm$.nextSibling), f = t[++c], h = s[--d];
    else if (F($, a, l))
      (f.$tag$ === "slot" || h.$tag$ === "slot") && A($.$elm$.parentNode, !1), E($, a, l), e.insertBefore($.$elm$, f.$elm$), $ = t[--u], a = s[++o];
    else {
      for (r = -1, i = c; i <= u; ++i)
        if (t[i] && t[i].$key$ !== null && t[i].$key$ === a.$key$) {
          r = i;
          break;
        }
      r >= 0 ? (g = t[r], g.$tag$ !== a.$tag$ ? y = w(t && t[o], n, r, e) : (E(g, a, l), t[r] = void 0, y = g.$elm$), a = s[++o]) : (y = w(t && t[o], n, o, e), a = s[++o]), y && ae(f.$elm$).insertBefore(y, q(f.$elm$));
    }
  c > u ? re(e, s[d + 1] == null ? null : s[d + 1].$elm$, n, s, o, d) : o > d && $e(t, c, u);
}, F = (e, t, n = !1) => e.$tag$ === t.$tag$ ? e.$tag$ === "slot" ? e.$name$ === t.$name$ : n ? !0 : e.$key$ === t.$key$ : !1, q = (e) => e && e["s-ol"] || e, ae = (e) => (e["s-ol"] ? e["s-ol"] : e).parentNode, E = (e, t, n = !1) => {
  const s = t.$elm$ = e.$elm$, l = e.$children$, c = t.$children$, o = t.$tag$, r = t.$text$;
  let i;
  r === null ? (v = o === "svg" ? !0 : o === "foreignObject" ? !1 : v, o === "slot" || ce(e, t, v), l !== null && c !== null ? Pe(s, l, t, c, n) : c !== null ? (e.$text$ !== null && (s.textContent = ""), re(s, null, t, c, 0, c.length - 1)) : l !== null && $e(l, 0, l.length - 1), v && o === "svg" && (v = !1)) : (i = s["s-cr"]) ? i.parentNode.textContent = r : e.$text$ !== r && (s.data = r);
}, fe = (e) => {
  const t = e.childNodes;
  for (const n of t)
    if (n.nodeType === 1) {
      if (n["s-sr"]) {
        const s = n["s-sn"];
        n.hidden = !1;
        for (const l of t)
          if (l !== n) {
            if (l["s-hn"] !== n["s-hn"] || s !== "") {
              if (l.nodeType === 1 && (s === l.getAttribute("slot") || s === l["s-sn"])) {
                n.hidden = !0;
                break;
              }
            } else if (l.nodeType === 1 || l.nodeType === 3 && l.textContent.trim() !== "") {
              n.hidden = !0;
              break;
            }
          }
      }
      fe(n);
    }
}, m = [], de = (e) => {
  let t, n, s;
  for (const l of e.childNodes) {
    if (l["s-sr"] && (t = l["s-cr"]) && t.parentNode) {
      n = t.parentNode.childNodes;
      const c = l["s-sn"];
      for (s = n.length - 1; s >= 0; s--)
        if (t = n[s], !t["s-cn"] && !t["s-nr"] && t["s-hn"] !== l["s-hn"] && !T.experimentalSlotFixes)
          if (Z(t, c)) {
            let o = m.find((r) => r.$nodeToRelocate$ === t);
            U = !0, t["s-sn"] = t["s-sn"] || c, o ? (o.$nodeToRelocate$["s-sh"] = l["s-hn"], o.$slotRefNode$ = l) : (t["s-sh"] = l["s-hn"], m.push({
              $slotRefNode$: l,
              $nodeToRelocate$: t
            })), t["s-sr"] && m.map((r) => {
              Z(r.$nodeToRelocate$, t["s-sn"]) && (o = m.find((i) => i.$nodeToRelocate$ === t), o && !r.$slotRefNode$ && (r.$slotRefNode$ = o.$slotRefNode$));
            });
          } else m.some((o) => o.$nodeToRelocate$ === t) || m.push({
            $nodeToRelocate$: t
          });
    }
    l.nodeType === 1 && de(l);
  }
}, Z = (e, t) => e.nodeType === 1 ? e.getAttribute("slot") === null && t === "" || e.getAttribute("slot") === t : e["s-sn"] === t ? !0 : t === "", ue = (e) => {
  e.$attrs$ && e.$attrs$.ref && e.$attrs$.ref(null), e.$children$ && e.$children$.map(ue);
}, je = (e, t, n = !1) => {
  var s, l, c, o, r;
  const i = e.$hostElement$, u = e.$cmpMeta$, f = e.$vnode$ || j(null, null), $ = xe(t) ? t : P(null, null, t);
  if (_ = i.tagName, u.$attrsToReflect$ && ($.$attrs$ = $.$attrs$ || {}, u.$attrsToReflect$.map(([d, a]) => $.$attrs$[a] = i[d])), n && $.$attrs$)
    for (const d of Object.keys($.$attrs$))
      i.hasAttribute(d) && !["key", "ref", "style", "class"].includes(d) && ($.$attrs$[d] = i[d]);
  $.$tag$ = null, $.$flags$ |= 4, e.$vnode$ = $, $.$elm$ = f.$elm$ = i.shadowRoot || i, L = i["s-sc"], M = i["s-cr"], le = (u.$flags$ & 1) !== 0, U = !1, E(f, $, n);
  {
    if (p.$flags$ |= 1, N) {
      de($.$elm$);
      for (const d of m) {
        const a = d.$nodeToRelocate$;
        if (!a["s-ol"]) {
          const h = S.createTextNode("");
          h["s-nr"] = a, a.parentNode.insertBefore(a["s-ol"] = h, a);
        }
      }
      for (const d of m) {
        const a = d.$nodeToRelocate$, h = d.$slotRefNode$;
        if (h) {
          const y = h.parentNode;
          let g = h.nextSibling;
          {
            let b = (s = a["s-ol"]) === null || s === void 0 ? void 0 : s.previousSibling;
            for (; b; ) {
              let k = (l = b["s-nr"]) !== null && l !== void 0 ? l : null;
              if (k && k["s-sn"] === a["s-sn"] && y === k.parentNode && (k = k.nextSibling, !k || !k["s-nr"])) {
                g = k;
                break;
              }
              b = b.previousSibling;
            }
          }
          (!g && y !== a.parentNode || a.nextSibling !== g) && a !== g && (!a["s-hn"] && a["s-ol"] && (a["s-hn"] = a["s-ol"].parentNode.nodeName), y.insertBefore(a, g), a.nodeType === 1 && (a.hidden = (c = a["s-ih"]) !== null && c !== void 0 ? c : !1));
        } else
          a.nodeType === 1 && (n && (a["s-ih"] = (o = a.hidden) !== null && o !== void 0 ? o : !1), a.hidden = !0);
      }
    }
    U && fe($.$elm$), p.$flags$ &= -2, m.length = 0;
  }
  if (T.experimentalScopedSlotChanges && u.$flags$ & 2)
    for (const d of $.$elm$.childNodes)
      d["s-hn"] !== _ && !d["s-sh"] && (n && d["s-ih"] == null && (d["s-ih"] = (r = d.hidden) !== null && r !== void 0 ? r : !1), d.hidden = !0);
  M = void 0;
}, we = (e, t) => {
}, he = (e, t) => (e.$flags$ |= 16, we(e, e.$ancestorComponent$), ct(() => Be(e, t))), Be = (e, t) => {
  const n = e.$hostElement$, s = x("scheduleUpdate", e.$cmpMeta$.$tagName$), l = n;
  let c;
  return t ? c = R(l, "componentWillLoad") : c = R(l, "componentWillUpdate"), c = V(c, () => R(l, "componentWillRender")), s(), V(c, () => Me(e, l, t));
}, V = (e, t) => De(e) ? e.then(t) : t(), De = (e) => e instanceof Promise || e && e.then && typeof e.then == "function", Me = async (e, t, n) => {
  const s = e.$hostElement$, l = x("update", e.$cmpMeta$.$tagName$);
  s["s-rc"], n && Ce(e);
  const c = x("render", e.$cmpMeta$.$tagName$);
  He(e, t, s, n), c(), l(), ze(e);
}, He = (e, t, n, s) => {
  try {
    I = t, t = t.render && t.render(), e.$flags$ &= -17, e.$flags$ |= 2, (T.hasRenderFn || T.reflect) && (T.vdomRender || T.reflect) && (T.hydrateServerSide || je(e, t, s));
  } catch (i) {
    C(i, e.$hostElement$);
  }
  return I = null, null;
}, ze = (e) => {
  const t = e.$cmpMeta$.$tagName$, n = e.$hostElement$, s = x("postUpdate", t), l = n;
  e.$ancestorComponent$, R(l, "componentDidRender"), e.$flags$ & 64 ? (R(l, "componentDidUpdate"), s()) : (e.$flags$ |= 64, R(l, "componentDidLoad"), s());
}, R = (e, t, n) => {
  if (e && e[t])
    try {
      return e[t](n);
    } catch (s) {
      C(s);
    }
}, Ne = (e, t) => O(e).$instanceValues$.get(t), We = (e, t, n, s) => {
  const l = O(e), c = e, o = l.$instanceValues$.get(t), r = l.$flags$, i = c;
  n = _e(n, s.$members$[t][0]);
  const u = Number.isNaN(o) && Number.isNaN(n);
  if (n !== o && !u) {
    l.$instanceValues$.set(t, n);
    {
      if (s.$watchers$ && r & 128) {
        const $ = s.$watchers$[t];
        $ && $.map((d) => {
          try {
            i[d](n, o, t);
          } catch (a) {
            C(a, c);
          }
        });
      }
      if ((r & 18) === 2) {
        if (i.componentShouldUpdate && i.componentShouldUpdate(n, o, t) === !1)
          return;
        he(l, !1);
      }
    }
  }
}, qe = (e, t, n) => {
  var s;
  const l = e.prototype;
  if (t.$members$) {
    e.watchers && (t.$watchers$ = e.watchers);
    const c = Object.entries(t.$members$);
    c.map(([o, [r]]) => {
      (r & 31 || r & 32) && Object.defineProperty(l, o, {
        get() {
          return Ne(this, o);
        },
        set(i) {
          We(this, o, i, t);
        },
        configurable: !0,
        enumerable: !0
      });
    });
    {
      const o = /* @__PURE__ */ new Map();
      l.attributeChangedCallback = function(r, i, u) {
        p.jmp(() => {
          var f;
          const $ = o.get(r);
          if (this.hasOwnProperty($))
            u = this[$], delete this[$];
          else {
            if (l.hasOwnProperty($) && typeof this[$] == "number" && this[$] == u)
              return;
            if ($ == null) {
              const d = O(this), a = d == null ? void 0 : d.$flags$;
              if (a && !(a & 8) && a & 128 && u !== i) {
                const y = this, g = (f = t.$watchers$) === null || f === void 0 ? void 0 : f[r];
                g == null || g.forEach((b) => {
                  y[b] != null && y[b].call(y, u, i, r);
                });
              }
              return;
            }
          }
          this[$] = u === null && typeof this[$] == "boolean" ? !1 : u;
        });
      }, e.observedAttributes = Array.from(/* @__PURE__ */ new Set([
        ...Object.keys((s = t.$watchers$) !== null && s !== void 0 ? s : {}),
        ...c.filter(
          ([r, i]) => i[0] & 15
          /* MEMBER_FLAGS.HasAttribute */
        ).map(([r, i]) => {
          var u;
          const f = i[1] || r;
          return o.set(f, r), i[0] & 512 && ((u = t.$attrsToReflect$) === null || u === void 0 || u.push([r, f])), f;
        })
      ]));
    }
  }
  return e;
}, Ie = async (e, t, n, s) => {
  let l;
  if (!(t.$flags$ & 32) && (t.$flags$ |= 32, l = e.constructor, customElements.whenDefined(n.$tagName$).then(() => t.$flags$ |= 128), l.style)) {
    let o = l.style;
    typeof o != "string" && (o = o[t.$modeName$ = Re(e)]);
    const r = oe(n, t.$modeName$);
    if (!B.has(r)) {
      const i = x("registerStyles", n.$tagName$);
      Ae(r, o, !!(n.$flags$ & 1)), i();
    }
  }
  t.$ancestorComponent$, he(t, !0);
}, ee = (e) => {
}, Qe = (e) => {
  if (!(p.$flags$ & 1)) {
    const t = O(e), n = t.$cmpMeta$, s = x("connectedCallback", n.$tagName$);
    t.$flags$ & 1 ? (pe(e, t, n.$listeners$), t != null && t.$lazyInstance$ ? ee(t.$lazyInstance$) : t != null && t.$onReadyPromise$ && t.$onReadyPromise$.then(() => ee(t.$lazyInstance$))) : (t.$flags$ |= 1, // TODO(STENCIL-854): Remove code related to legacy shadowDomShim field
    n.$flags$ & 12 && Xe(e), n.$members$ && Object.entries(n.$members$).map(([l, [c]]) => {
      if (c & 31 && e.hasOwnProperty(l)) {
        const o = e[l];
        delete e[l], e[l] = o;
      }
    }), Ie(e, t, n)), s();
  }
}, Xe = (e) => {
  const t = e["s-cr"] = S.createComment("");
  t["s-cn"] = !0, e.insertBefore(t, e.firstChild);
}, Ge = async (e) => {
  if (!(p.$flags$ & 1)) {
    const t = O(e);
    t.$rmListeners$ && (t.$rmListeners$.map((n) => n()), t.$rmListeners$ = void 0);
  }
}, Ke = (e, t) => {
  const n = {
    $flags$: t[0],
    $tagName$: t[1]
  };
  n.$members$ = t[2], n.$listeners$ = t[3], n.$watchers$ = e.$watchers$, n.$attrsToReflect$ = [];
  const s = e.prototype.connectedCallback, l = e.prototype.disconnectedCallback;
  return Object.assign(e.prototype, {
    __registerHost() {
      Ve(this, n);
    },
    connectedCallback() {
      Qe(this), s && s.call(this);
    },
    disconnectedCallback() {
      Ge(this), l && l.call(this);
    },
    __attachShadow() {
      this.attachShadow({
        mode: "open",
        delegatesFocus: !!(n.$flags$ & 16)
      });
    }
  }), e.is = n.$tagName$, qe(e, n);
}, pe = (e, t, n, s) => {
  n && n.map(([l, c, o]) => {
    const r = Ye(e, l), i = Je(t, o), u = Ze(l);
    p.ael(r, c, i, u), (t.$rmListeners$ = t.$rmListeners$ || []).push(() => p.rel(r, c, i, u));
  });
}, Je = (e, t) => (n) => {
  try {
    T.lazyLoad || e.$hostElement$[t](n);
  } catch (s) {
    C(s);
  }
}, Ye = (e, t) => t & 4 ? S : t & 8 ? D : t & 16 ? S.body : e, Ze = (e) => st ? {
  passive: (e & 1) !== 0,
  capture: (e & 2) !== 0
} : (e & 2) !== 0, ge = /* @__PURE__ */ new WeakMap(), O = (e) => ge.get(e), Ve = (e, t) => {
  const n = {
    $flags$: 0,
    $hostElement$: e,
    $cmpMeta$: t,
    $instanceValues$: /* @__PURE__ */ new Map()
  };
  return pe(e, n, t.$listeners$), ge.set(e, n);
}, te = (e, t) => t in e, C = (e, t) => (0, console.error)(e, t), B = /* @__PURE__ */ new Map(), et = [], D = typeof window < "u" ? window : {}, S = D.document || { head: {} }, tt = D.HTMLElement || class {
}, p = {
  $flags$: 0,
  $resourcesUrl$: "",
  jmp: (e) => e(),
  raf: (e) => requestAnimationFrame(e),
  ael: (e, t, n, s) => e.addEventListener(t, n, s),
  rel: (e, t, n, s) => e.removeEventListener(t, n, s),
  ce: (e, t) => new CustomEvent(e, t)
}, st = /* @__PURE__ */ (() => {
  let e = !1;
  try {
    S.addEventListener("e", null, Object.defineProperty({}, "passive", {
      get() {
        e = !0;
      }
    }));
  } catch {
  }
  return e;
})(), nt = (e) => Promise.resolve(e), lt = /* @__PURE__ */ (() => {
  try {
    return new CSSStyleSheet(), typeof new CSSStyleSheet().replaceSync == "function";
  } catch {
  }
  return !1;
})(), se = [], ye = [], ot = (e, t) => (n) => {
  e.push(n), H || (H = !0, t && p.$flags$ & 4 ? it(z) : p.raf(z));
}, ne = (e) => {
  for (let t = 0; t < e.length; t++)
    try {
      e[t](performance.now());
    } catch (n) {
      C(n);
    }
  e.length = 0;
}, z = () => {
  ne(se), ne(ye), (H = se.length > 0) && p.raf(z);
}, it = (e) => nt().then(e), ct = /* @__PURE__ */ ot(ye, !0), rt = ".sc-refresh-icon-h{display:flex}", $t = rt, at = Ke(class extends tt {
  constructor() {
    super(), this.__registerHost(), this.width = "15", this.height = "15";
  }
  render() {
    return P("svg", { key: "6b97426e4b19084109dd6615f5bdc654a0020fb9", fill: "auto", width: this.width, height: this.height, viewBox: "0 0 32 32", version: "1.1", xmlns: "http://www.w3.org/2000/svg" }, P("path", { key: "fc8867cd8e42697f0d4a3dac3dc0973cd0c29d65", d: "M15.544 28.080c-0.002-0.006-0.008-0.010-0.012-0.015l-2.873-4.489c-0.204-0.335-0.646-0.454-0.992-0.266l-0.256 0.157c-0.344 0.188-0.337 0.673-0.133 1.008l1.841 2.857c-0.157-0.035-0.316-0.063-0.471-0.103-3.224-0.843-5.953-3.026-7.485-5.988-1.448-2.797-1.72-5.99-0.766-8.992s3.118-5.452 5.915-6.899c0.476-0.247 0.662-0.832 0.416-1.308-0.246-0.477-0.832-0.663-1.308-0.416-3.258 1.686-5.763 4.54-6.874 8.036s-0.794 7.215 0.892 10.473c1.786 3.448 4.963 5.989 8.72 6.973 0.043 0.011 0.087 0.017 0.13 0.028l-2.541 1.288c-0.344 0.189-0.458 0.613-0.254 0.948l0.098 0.256c0.205 0.335 0.557 0.454 0.9 0.266l4.651-2.381c0.006-0.004 0.012-0.003 0.018-0.007l0.312-0.171c0.172-0.095 0.287-0.249 0.332-0.422 0.047-0.172 0.025-0.364-0.076-0.531zM28.559 10.025c-1.783-3.447-4.862-5.988-8.618-6.972-0.267-0.070-0.541-0.124-0.814-0.179l2.494-1.265c0.344-0.189 0.549-0.614 0.345-0.949l-0.099-0.255c-0.205-0.336-0.648-0.454-0.991-0.267l-4.651 2.381c-0.006 0.003-0.012 0.002-0.018 0.006l-0.312 0.171c-0.173 0.095-0.287 0.249-0.332 0.422-0.047 0.172-0.025 0.364 0.077 0.53l0.185 0.304c0.003 0.006 0.008 0.010 0.012 0.016l2.873 4.489c0.203 0.335 0.646 0.454 0.991 0.266l0.226-0.157c0.344-0.188 0.366-0.673 0.163-1.008l-1.85-2.87c0.407 0.063 0.811 0.138 1.207 0.242 3.226 0.845 5.856 3.027 7.387 5.986 1.448 2.797 1.72 5.99 0.765 8.991s-3.020 5.451-5.818 6.901c-0.476 0.247-0.662 0.831-0.415 1.308 0.172 0.332 0.511 0.524 0.863 0.524 0.15 0 0.302-0.035 0.446-0.109 3.259-1.686 5.664-4.54 6.776-8.035 1.111-3.497 0.794-7.217-0.893-10.473z" }));
  }
  static get style() {
    return $t;
  }
}, [2, "refresh-icon", {
  width: [1],
  height: [1]
}]);
function dt() {
  if (typeof customElements > "u")
    return;
  ["refresh-icon"].forEach((t) => {
    switch (t) {
      case "refresh-icon":
        customElements.get(t) || customElements.define(t, at);
        break;
    }
  });
}
export {
  at as R,
  dt as d
};
