// @nylas/.web-elements-GF2lXkj6/dist/cdn/button-component/button-component.es.js
const x = {
  allRenderFn: !1,
  cmpDidLoad: !0,
  cmpDidUnload: !1,
  cmpDidUpdate: !0,
  cmpDidRender: !0,
  cmpWillLoad: !0,
  cmpWillUpdate: !0,
  cmpWillRender: !0,
  connectedCallback: !0,
  disconnectedCallback: !0,
  element: !0,
  event: !0,
  hasRenderFn: !0,
  lifecycle: !0,
  hostListener: !0,
  hostListenerTargetWindow: !0,
  hostListenerTargetDocument: !0,
  hostListenerTargetBody: !0,
  hostListenerTargetParent: !1,
  hostListenerTarget: !0,
  member: !0,
  method: !0,
  mode: !0,
  observeAttribute: !0,
  prop: !0,
  propMutable: !0,
  reflect: !0,
  scoped: !0,
  shadowDom: !0,
  slot: !0,
  cssAnnotations: !0,
  state: !0,
  style: !0,
  formAssociated: !1,
  svg: !0,
  updatable: !0,
  vdomAttribute: !0,
  vdomXlink: !0,
  vdomClass: !0,
  vdomFunctional: !0,
  vdomKey: !0,
  vdomListener: !0,
  vdomRef: !0,
  vdomPropOrAttr: !0,
  vdomRender: !0,
  vdomStyle: !0,
  vdomText: !0,
  watchCallback: !0,
  taskQueue: !0,
  hotModuleReplacement: !1,
  isDebug: !1,
  isDev: !1,
  isTesting: !1,
  hydrateServerSide: !1,
  hydrateClientSide: !1,
  lifecycleDOMEvents: !1,
  lazyLoad: !1,
  profile: !1,
  slotRelocation: !0,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  appendChildSlotFix: !1,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  cloneNodeFix: !1,
  hydratedAttribute: !1,
  hydratedClass: !0,
  scriptDataOpts: !1,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  scopedSlotTextContentFix: !1,
  // TODO(STENCIL-854): Remove code related to legacy shadowDomShim field
  shadowDomShim: !1,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  slotChildNodesFix: !1,
  invisiblePrehydration: !0,
  propBoolean: !0,
  propNumber: !0,
  propString: !0,
  constructableCSS: !0,
  cmpShouldUpdate: !0,
  devTools: !1,
  shadowDelegatesFocus: !0,
  initializeNextTick: !1,
  asyncLoading: !1,
  asyncQueue: !1,
  transformTagName: !1,
  attachStyles: !0,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  experimentalSlotFixes: !1
};
let E, D, H, ot = !1, B = !1, W = !1, y = !1, I = null, M = !1;
const L = (t, e = "") => () => {
}, kt = "slot-fb{display:contents}slot-fb[hidden]{display:none}", Q = "http://www.w3.org/1999/xlink", X = {}, St = "http://www.w3.org/2000/svg", xt = "http://www.w3.org/1999/xhtml", Tt = (t) => t != null, N = (t) => (t = typeof t, t === "object" || t === "function");
function Lt(t) {
  var e, n, s;
  return (s = (n = (e = t.head) === null || e === void 0 ? void 0 : e.querySelector('meta[name="csp-nonce"]')) === null || n === void 0 ? void 0 : n.getAttribute("content")) !== null && s !== void 0 ? s : void 0;
}
const k = (t, e, ...n) => {
  let s = null, o = null, i = null, l = !1, a = !1;
  const r = [], u = (c) => {
    for (let f = 0; f < c.length; f++)
      s = c[f], Array.isArray(s) ? u(s) : s != null && typeof s != "boolean" && ((l = typeof t != "function" && !N(s)) && (s = String(s)), l && a ? r[r.length - 1].$text$ += s : r.push(l ? F(null, s) : s), a = l);
  };
  if (u(n), e) {
    e.key && (o = e.key), e.name && (i = e.name);
    {
      const c = e.className || e.class;
      c && (e.class = typeof c != "object" ? c : Object.keys(c).filter((f) => c[f]).join(" "));
    }
  }
  if (typeof t == "function")
    return t(e === null ? {} : e, r, Ot);
  const $ = F(t, null);
  return $.$attrs$ = e, r.length > 0 && ($.$children$ = r), $.$key$ = o, $.$name$ = i, $;
}, F = (t, e) => {
  const n = {
    $flags$: 0,
    $tag$: t,
    $text$: e,
    $elm$: null,
    $children$: null
  };
  return n.$attrs$ = null, n.$key$ = null, n.$name$ = null, n;
}, Et = {}, _t = (t) => t && t.$tag$ === Et, Ot = {
  forEach: (t, e) => t.map(Y).forEach(e),
  map: (t, e) => t.map(Y).map(e).map(Ht)
}, Y = (t) => ({
  vattrs: t.$attrs$,
  vchildren: t.$children$,
  vkey: t.$key$,
  vname: t.$name$,
  vtag: t.$tag$,
  vtext: t.$text$
}), Ht = (t) => {
  if (typeof t.vtag == "function") {
    const n = Object.assign({}, t.vattrs);
    return t.vkey && (n.key = t.vkey), t.vname && (n.name = t.vname), k(t.vtag, n, ...t.vchildren || []);
  }
  const e = F(t.vtag, t.vtext);
  return e.$attrs$ = t.vattrs, e.$children$ = t.vchildren, e.$key$ = t.vkey, e.$name$ = t.vname, e;
}, At = (t) => se.map((e) => e(t)).find((e) => !!e), Ct = (t, e) => t != null && !N(t) ? e & 4 ? t === "false" ? !1 : t === "" || !!t : e & 2 ? parseFloat(t) : e & 1 ? String(t) : t : t, G = /* @__PURE__ */ new WeakMap(), wt = (t, e, n) => {
  let s = U.get(t);
  le && n ? (s = s || new CSSStyleSheet(), typeof s == "string" ? s = e : s.replaceSync(e)) : s = e, U.set(t, s);
}, Rt = (t, e, n) => {
  var s;
  const o = lt(e, n), i = U.get(o);
  if (t = t.nodeType === 11 ? t : v, i)
    if (typeof i == "string") {
      t = t.head || t;
      let l = G.get(t), a;
      if (l || G.set(t, l = /* @__PURE__ */ new Set()), !l.has(o)) {
        {
          a = v.createElement("style"), a.innerHTML = i;
          const r = (s = h.$nonce$) !== null && s !== void 0 ? s : Lt(v);
          r != null && a.setAttribute("nonce", r), t.insertBefore(a, t.querySelector("link"));
        }
        e.$flags$ & 4 && (a.innerHTML += kt), l && l.add(o);
      }
    } else t.adoptedStyleSheets.includes(i) || (t.adoptedStyleSheets = [...t.adoptedStyleSheets, i]);
  return o;
}, Bt = (t) => {
  const e = t.$cmpMeta$, n = t.$hostElement$, s = e.$flags$, o = L("attachStyles", e.$tagName$), i = Rt(n.shadowRoot ? n.shadowRoot : n.getRootNode(), e, t.$modeName$);
  s & 10 && (n["s-sc"] = i, n.classList.add(i + "-h"), s & 2 && n.classList.add(i + "-s")), o();
}, lt = (t, e) => "sc-" + (e && t.$flags$ & 32 ? t.$tagName$ + "-" + e : t.$tagName$), K = (t, e, n, s, o, i) => {
  if (n !== s) {
    let l = et(t, e), a = e.toLowerCase();
    if (e === "class") {
      const r = t.classList, u = J(n), $ = J(s);
      r.remove(...u.filter((c) => c && !$.includes(c))), r.add(...$.filter((c) => c && !u.includes(c)));
    } else if (e === "style") {
      for (const r in n)
        (!s || s[r] == null) && (r.includes("-") ? t.style.removeProperty(r) : t.style[r] = "");
      for (const r in s)
        (!n || s[r] !== n[r]) && (r.includes("-") ? t.style.setProperty(r, s[r]) : t.style[r] = s[r]);
    } else if (e !== "key")
      if (e === "ref")
        s && s(t);
      else if (!t.__lookupSetter__(e) && e[0] === "o" && e[1] === "n") {
        if (e[2] === "-" ? e = e.slice(3) : et(P, a) ? e = a.slice(2) : e = a[2] + e.slice(3), n || s) {
          const r = e.endsWith(rt);
          e = e.replace(jt, ""), n && h.rel(t, e, n, r), s && h.ael(t, e, s, r);
        }
      } else {
        const r = N(s);
        if ((l || r && s !== null) && !o)
          try {
            if (t.tagName.includes("-"))
              t[e] = s;
            else {
              const $ = s ?? "";
              e === "list" ? l = !1 : (n == null || t[e] != $) && (t[e] = $);
            }
          } catch {
          }
        let u = !1;
        a !== (a = a.replace(/^xlink\:?/, "")) && (e = a, u = !0), s == null || s === !1 ? (s !== !1 || t.getAttribute(e) === "") && (u ? t.removeAttributeNS(Q, e) : t.removeAttribute(e)) : (!l || i & 4 || o) && !r && (s = s === !0 ? "" : s, u ? t.setAttributeNS(Q, e, s) : t.setAttribute(e, s));
      }
  }
}, Ft = /\s/, J = (t) => t ? t.split(Ft) : [], rt = "Capture", jt = new RegExp(rt + "$"), it = (t, e, n, s) => {
  const o = e.$elm$.nodeType === 11 && e.$elm$.host ? e.$elm$.host : e.$elm$, i = t && t.$attrs$ || X, l = e.$attrs$ || X;
  for (s in i)
    s in l || K(o, s, i[s], void 0, n, e.$flags$);
  for (s in l)
    K(o, s, i[s], l[s], n, e.$flags$);
}, j = (t, e, n, s) => {
  const o = e.$children$[n];
  let i = 0, l, a, r;
  if (ot || (W = !0, o.$tag$ === "slot" && (E && s.classList.add(E + "-s"), o.$flags$ |= o.$children$ ? (
    // slot element has fallback content
    2
  ) : (
    // slot element does not have fallback content
    1
  ))), o.$text$ !== null)
    l = o.$elm$ = v.createTextNode(o.$text$);
  else if (o.$flags$ & 1)
    l = o.$elm$ = v.createTextNode("");
  else {
    if (y || (y = o.$tag$ === "svg"), l = o.$elm$ = v.createElementNS(y ? St : xt, o.$flags$ & 2 ? "slot-fb" : o.$tag$), y && o.$tag$ === "foreignObject" && (y = !1), it(null, o, y), Tt(E) && l["s-si"] !== E && l.classList.add(l["s-si"] = E), o.$children$)
      for (i = 0; i < o.$children$.length; ++i)
        a = j(t, o, i, l), a && l.appendChild(a);
    o.$tag$ === "svg" ? y = !1 : l.tagName === "foreignObject" && (y = !0);
  }
  return l["s-hn"] = H, o.$flags$ & 3 && (l["s-sr"] = !0, l["s-cr"] = D, l["s-sn"] = o.$name$ || "", r = t && t.$children$ && t.$children$[n], r && r.$tag$ === o.$tag$ && t.$elm$ && A(t.$elm$, !1)), l;
}, A = (t, e) => {
  h.$flags$ |= 1;
  const n = t.childNodes;
  for (let s = n.length - 1; s >= 0; s--) {
    const o = n[s];
    o["s-hn"] !== H && o["s-ol"] && (dt(o).insertBefore(o, q(o)), o["s-ol"].remove(), o["s-ol"] = void 0, o["s-sh"] = void 0, W = !0), e && A(o, e);
  }
  h.$flags$ &= -2;
}, at = (t, e, n, s, o, i) => {
  let l = t["s-cr"] && t["s-cr"].parentNode || t, a;
  for (l.shadowRoot && l.tagName === H && (l = l.shadowRoot); o <= i; ++o)
    s[o] && (a = j(null, n, o, t), a && (s[o].$elm$ = a, l.insertBefore(a, q(e))));
}, ct = (t, e, n) => {
  for (let s = e; s <= n; ++s) {
    const o = t[s];
    if (o) {
      const i = o.$elm$;
      ut(o), i && (B = !0, i["s-ol"] ? i["s-ol"].remove() : A(i, !0), i.remove());
    }
  }
}, Ut = (t, e, n, s, o = !1) => {
  let i = 0, l = 0, a = 0, r = 0, u = e.length - 1, $ = e[0], c = e[u], f = s.length - 1, d = s[0], p = s[f], b, g;
  for (; i <= u && l <= f; )
    if ($ == null)
      $ = e[++i];
    else if (c == null)
      c = e[--u];
    else if (d == null)
      d = s[++l];
    else if (p == null)
      p = s[--f];
    else if (R($, d, o))
      _($, d, o), $ = e[++i], d = s[++l];
    else if (R(c, p, o))
      _(c, p, o), c = e[--u], p = s[--f];
    else if (R($, p, o))
      ($.$tag$ === "slot" || p.$tag$ === "slot") && A($.$elm$.parentNode, !1), _($, p, o), t.insertBefore($.$elm$, c.$elm$.nextSibling), $ = e[++i], p = s[--f];
    else if (R(c, d, o))
      ($.$tag$ === "slot" || p.$tag$ === "slot") && A(c.$elm$.parentNode, !1), _(c, d, o), t.insertBefore(c.$elm$, $.$elm$), c = e[--u], d = s[++l];
    else {
      for (a = -1, r = i; r <= u; ++r)
        if (e[r] && e[r].$key$ !== null && e[r].$key$ === d.$key$) {
          a = r;
          break;
        }
      a >= 0 ? (g = e[a], g.$tag$ !== d.$tag$ ? b = j(e && e[l], n, a, t) : (_(g, d, o), e[a] = void 0, b = g.$elm$), d = s[++l]) : (b = j(e && e[l], n, l, t), d = s[++l]), b && dt($.$elm$).insertBefore(b, q($.$elm$));
    }
  i > u ? at(t, s[f + 1] == null ? null : s[f + 1].$elm$, n, s, l, f) : l > f && ct(e, i, u);
}, R = (t, e, n = !1) => t.$tag$ === e.$tag$ ? t.$tag$ === "slot" ? t.$name$ === e.$name$ : n ? !0 : t.$key$ === e.$key$ : !1, q = (t) => t && t["s-ol"] || t, dt = (t) => (t["s-ol"] ? t["s-ol"] : t).parentNode, _ = (t, e, n = !1) => {
  const s = e.$elm$ = t.$elm$, o = t.$children$, i = e.$children$, l = e.$tag$, a = e.$text$;
  let r;
  a === null ? (y = l === "svg" ? !0 : l === "foreignObject" ? !1 : y, l === "slot" || it(t, e, y), o !== null && i !== null ? Ut(s, o, e, i, n) : i !== null ? (t.$text$ !== null && (s.textContent = ""), at(s, null, e, i, 0, i.length - 1)) : o !== null && ct(o, 0, o.length - 1), y && l === "svg" && (y = !1)) : (r = s["s-cr"]) ? r.parentNode.textContent = a : t.$text$ !== a && (s.data = a);
}, $t = (t) => {
  const e = t.childNodes;
  for (const n of e)
    if (n.nodeType === 1) {
      if (n["s-sr"]) {
        const s = n["s-sn"];
        n.hidden = !1;
        for (const o of e)
          if (o !== n) {
            if (o["s-hn"] !== n["s-hn"] || s !== "") {
              if (o.nodeType === 1 && (s === o.getAttribute("slot") || s === o["s-sn"])) {
                n.hidden = !0;
                break;
              }
            } else if (o.nodeType === 1 || o.nodeType === 3 && o.textContent.trim() !== "") {
              n.hidden = !0;
              break;
            }
          }
      }
      $t(n);
    }
}, m = [], ft = (t) => {
  let e, n, s;
  for (const o of t.childNodes) {
    if (o["s-sr"] && (e = o["s-cr"]) && e.parentNode) {
      n = e.parentNode.childNodes;
      const i = o["s-sn"];
      for (s = n.length - 1; s >= 0; s--)
        if (e = n[s], !e["s-cn"] && !e["s-nr"] && e["s-hn"] !== o["s-hn"] && !x.experimentalSlotFixes)
          if (Z(e, i)) {
            let l = m.find((a) => a.$nodeToRelocate$ === e);
            B = !0, e["s-sn"] = e["s-sn"] || i, l ? (l.$nodeToRelocate$["s-sh"] = o["s-hn"], l.$slotRefNode$ = o) : (e["s-sh"] = o["s-hn"], m.push({
              $slotRefNode$: o,
              $nodeToRelocate$: e
            })), e["s-sr"] && m.map((a) => {
              Z(a.$nodeToRelocate$, e["s-sn"]) && (l = m.find((r) => r.$nodeToRelocate$ === e), l && !a.$slotRefNode$ && (a.$slotRefNode$ = l.$slotRefNode$));
            });
          } else m.some((l) => l.$nodeToRelocate$ === e) || m.push({
            $nodeToRelocate$: e
          });
    }
    o.nodeType === 1 && ft(o);
  }
}, Z = (t, e) => t.nodeType === 1 ? t.getAttribute("slot") === null && e === "" || t.getAttribute("slot") === e : t["s-sn"] === e ? !0 : e === "", ut = (t) => {
  t.$attrs$ && t.$attrs$.ref && t.$attrs$.ref(null), t.$children$ && t.$children$.map(ut);
}, Pt = (t, e, n = !1) => {
  var s, o, i, l, a;
  const r = t.$hostElement$, u = t.$cmpMeta$, $ = t.$vnode$ || F(null, null), c = _t(e) ? e : k(null, null, e);
  if (H = r.tagName, u.$attrsToReflect$ && (c.$attrs$ = c.$attrs$ || {}, u.$attrsToReflect$.map(([f, d]) => c.$attrs$[d] = r[f])), n && c.$attrs$)
    for (const f of Object.keys(c.$attrs$))
      r.hasAttribute(f) && !["key", "ref", "style", "class"].includes(f) && (c.$attrs$[f] = r[f]);
  c.$tag$ = null, c.$flags$ |= 4, t.$vnode$ = c, c.$elm$ = $.$elm$ = r.shadowRoot || r, E = r["s-sc"], D = r["s-cr"], ot = (u.$flags$ & 1) !== 0, B = !1, _($, c, n);
  {
    if (h.$flags$ |= 1, W) {
      ft(c.$elm$);
      for (const f of m) {
        const d = f.$nodeToRelocate$;
        if (!d["s-ol"]) {
          const p = v.createTextNode("");
          p["s-nr"] = d, d.parentNode.insertBefore(d["s-ol"] = p, d);
        }
      }
      for (const f of m) {
        const d = f.$nodeToRelocate$, p = f.$slotRefNode$;
        if (p) {
          const b = p.parentNode;
          let g = p.nextSibling;
          {
            let T = (s = d["s-ol"]) === null || s === void 0 ? void 0 : s.previousSibling;
            for (; T; ) {
              let S = (o = T["s-nr"]) !== null && o !== void 0 ? o : null;
              if (S && S["s-sn"] === d["s-sn"] && b === S.parentNode && (S = S.nextSibling, !S || !S["s-nr"])) {
                g = S;
                break;
              }
              T = T.previousSibling;
            }
          }
          (!g && b !== d.parentNode || d.nextSibling !== g) && d !== g && (!d["s-hn"] && d["s-ol"] && (d["s-hn"] = d["s-ol"].parentNode.nodeName), b.insertBefore(d, g), d.nodeType === 1 && (d.hidden = (i = d["s-ih"]) !== null && i !== void 0 ? i : !1));
        } else
          d.nodeType === 1 && (n && (d["s-ih"] = (l = d.hidden) !== null && l !== void 0 ? l : !1), d.hidden = !0);
      }
    }
    B && $t(c.$elm$), h.$flags$ &= -2, m.length = 0;
  }
  if (x.experimentalScopedSlotChanges && u.$flags$ & 2)
    for (const f of c.$elm$.childNodes)
      f["s-hn"] !== H && !f["s-sh"] && (n && f["s-ih"] == null && (f["s-ih"] = (a = f.hidden) !== null && a !== void 0 ? a : !1), f.hidden = !0);
  D = void 0;
}, Dt = (t, e) => {
}, pt = (t, e) => (t.$flags$ |= 16, Dt(t, t.$ancestorComponent$), ae(() => Mt(t, e))), Mt = (t, e) => {
  const n = t.$hostElement$, s = L("scheduleUpdate", t.$cmpMeta$.$tagName$), o = n;
  let i;
  return e ? i = O(o, "componentWillLoad") : i = O(o, "componentWillUpdate"), i = V(i, () => O(o, "componentWillRender")), s(), V(i, () => Wt(t, o, e));
}, V = (t, e) => zt(t) ? t.then(e) : e(), zt = (t) => t instanceof Promise || t && t.then && typeof t.then == "function", Wt = async (t, e, n) => {
  const s = t.$hostElement$, o = L("update", t.$cmpMeta$.$tagName$);
  s["s-rc"], n && Bt(t);
  const i = L("render", t.$cmpMeta$.$tagName$);
  Nt(t, e, s, n), i(), o(), qt(t);
}, Nt = (t, e, n, s) => {
  try {
    I = e, e = e.render && e.render(), t.$flags$ &= -17, t.$flags$ |= 2, (x.hasRenderFn || x.reflect) && (x.vdomRender || x.reflect) && (x.hydrateServerSide || Pt(t, e, s));
  } catch (r) {
    w(r, t.$hostElement$);
  }
  return I = null, null;
}, qt = (t) => {
  const e = t.$cmpMeta$.$tagName$, n = t.$hostElement$, s = L("postUpdate", e), o = n;
  t.$ancestorComponent$, O(o, "componentDidRender"), t.$flags$ & 64 ? (O(o, "componentDidUpdate"), s()) : (t.$flags$ |= 64, O(o, "componentDidLoad"), s());
}, O = (t, e, n) => {
  if (t && t[e])
    try {
      return t[e](n);
    } catch (s) {
      w(s);
    }
}, It = (t, e) => C(t).$instanceValues$.get(e), Qt = (t, e, n, s) => {
  const o = C(t), i = t, l = o.$instanceValues$.get(e), a = o.$flags$, r = i;
  n = Ct(n, s.$members$[e][0]);
  const u = Number.isNaN(l) && Number.isNaN(n);
  if (n !== l && !u) {
    o.$instanceValues$.set(e, n);
    {
      if (s.$watchers$ && a & 128) {
        const c = s.$watchers$[e];
        c && c.map((f) => {
          try {
            r[f](n, l, e);
          } catch (d) {
            w(d, i);
          }
        });
      }
      if ((a & 18) === 2) {
        if (r.componentShouldUpdate && r.componentShouldUpdate(n, l, e) === !1)
          return;
        pt(o, !1);
      }
    }
  }
}, Xt = (t, e, n) => {
  var s;
  const o = t.prototype;
  if (e.$members$) {
    t.watchers && (e.$watchers$ = t.watchers);
    const i = Object.entries(e.$members$);
    i.map(([l, [a]]) => {
      (a & 31 || a & 32) && Object.defineProperty(o, l, {
        get() {
          return It(this, l);
        },
        set(r) {
          Qt(this, l, r, e);
        },
        configurable: !0,
        enumerable: !0
      });
    });
    {
      const l = /* @__PURE__ */ new Map();
      o.attributeChangedCallback = function(a, r, u) {
        h.jmp(() => {
          var $;
          const c = l.get(a);
          if (this.hasOwnProperty(c))
            u = this[c], delete this[c];
          else {
            if (o.hasOwnProperty(c) && typeof this[c] == "number" && this[c] == u)
              return;
            if (c == null) {
              const f = C(this), d = f == null ? void 0 : f.$flags$;
              if (d && !(d & 8) && d & 128 && u !== r) {
                const b = this, g = ($ = e.$watchers$) === null || $ === void 0 ? void 0 : $[a];
                g == null || g.forEach((T) => {
                  b[T] != null && b[T].call(b, u, r, a);
                });
              }
              return;
            }
          }
          this[c] = u === null && typeof this[c] == "boolean" ? !1 : u;
        });
      }, t.observedAttributes = Array.from(/* @__PURE__ */ new Set([
        ...Object.keys((s = e.$watchers$) !== null && s !== void 0 ? s : {}),
        ...i.filter(
          ([a, r]) => r[0] & 15
          /* MEMBER_FLAGS.HasAttribute */
        ).map(([a, r]) => {
          var u;
          const $ = r[1] || a;
          return l.set($, a), r[0] & 512 && ((u = e.$attrsToReflect$) === null || u === void 0 || u.push([a, $])), $;
        })
      ]));
    }
  }
  return t;
}, Yt = async (t, e, n, s) => {
  let o;
  if (!(e.$flags$ & 32) && (e.$flags$ |= 32, o = t.constructor, customElements.whenDefined(n.$tagName$).then(() => e.$flags$ |= 128), o.style)) {
    let l = o.style;
    typeof l != "string" && (l = l[e.$modeName$ = At(t)]);
    const a = lt(n, e.$modeName$);
    if (!U.has(a)) {
      const r = L("registerStyles", n.$tagName$);
      wt(a, l, !!(n.$flags$ & 1)), r();
    }
  }
  e.$ancestorComponent$, pt(e, !0);
}, tt = (t) => {
}, Gt = (t) => {
  if (!(h.$flags$ & 1)) {
    const e = C(t), n = e.$cmpMeta$, s = L("connectedCallback", n.$tagName$);
    e.$flags$ & 1 ? (gt(t, e, n.$listeners$), e != null && e.$lazyInstance$ ? tt(e.$lazyInstance$) : e != null && e.$onReadyPromise$ && e.$onReadyPromise$.then(() => tt(e.$lazyInstance$))) : (e.$flags$ |= 1, // TODO(STENCIL-854): Remove code related to legacy shadowDomShim field
    n.$flags$ & 12 && Kt(t), n.$members$ && Object.entries(n.$members$).map(([o, [i]]) => {
      if (i & 31 && t.hasOwnProperty(o)) {
        const l = t[o];
        delete t[o], t[o] = l;
      }
    }), Yt(t, e, n)), s();
  }
}, Kt = (t) => {
  const e = t["s-cr"] = v.createComment("");
  e["s-cn"] = !0, t.insertBefore(e, t.firstChild);
}, Jt = async (t) => {
  if (!(h.$flags$ & 1)) {
    const e = C(t);
    e.$rmListeners$ && (e.$rmListeners$.map((n) => n()), e.$rmListeners$ = void 0);
  }
}, ht = (t, e) => {
  const n = {
    $flags$: e[0],
    $tagName$: e[1]
  };
  n.$members$ = e[2], n.$listeners$ = e[3], n.$watchers$ = t.$watchers$, n.$attrsToReflect$ = [];
  const s = t.prototype.connectedCallback, o = t.prototype.disconnectedCallback;
  return Object.assign(t.prototype, {
    __registerHost() {
      ee(this, n);
    },
    connectedCallback() {
      Gt(this), s && s.call(this);
    },
    disconnectedCallback() {
      Jt(this), o && o.call(this);
    },
    __attachShadow() {
      this.attachShadow({
        mode: "open",
        delegatesFocus: !!(n.$flags$ & 16)
      });
    }
  }), t.is = n.$tagName$, Xt(t, n);
}, gt = (t, e, n, s) => {
  n && n.map(([o, i, l]) => {
    const a = Vt(t, o), r = Zt(e, l), u = te(o);
    h.ael(a, i, r, u), (e.$rmListeners$ = e.$rmListeners$ || []).push(() => h.rel(a, i, r, u));
  });
}, Zt = (t, e) => (n) => {
  try {
    x.lazyLoad || t.$hostElement$[e](n);
  } catch (s) {
    w(s);
  }
}, Vt = (t, e) => e & 4 ? v : e & 8 ? P : e & 16 ? v.body : t, te = (t) => ne ? {
  passive: (t & 1) !== 0,
  capture: (t & 2) !== 0
} : (t & 2) !== 0, bt = /* @__PURE__ */ new WeakMap(), C = (t) => bt.get(t), ee = (t, e) => {
  const n = {
    $flags$: 0,
    $hostElement$: t,
    $cmpMeta$: e,
    $instanceValues$: /* @__PURE__ */ new Map()
  };
  return gt(t, n, e.$listeners$), bt.set(t, n);
}, et = (t, e) => e in t, w = (t, e) => (0, console.error)(t, e), U = /* @__PURE__ */ new Map(), se = [], P = typeof window < "u" ? window : {}, v = P.document || { head: {} }, yt = P.HTMLElement || class {
}, h = {
  $flags$: 0,
  $resourcesUrl$: "",
  jmp: (t) => t(),
  raf: (t) => requestAnimationFrame(t),
  ael: (t, e, n, s) => t.addEventListener(e, n, s),
  rel: (t, e, n, s) => t.removeEventListener(e, n, s),
  ce: (t, e) => new CustomEvent(t, e)
}, ne = /* @__PURE__ */ (() => {
  let t = !1;
  try {
    v.addEventListener("e", null, Object.defineProperty({}, "passive", {
      get() {
        t = !0;
      }
    }));
  } catch {
  }
  return t;
})(), oe = (t) => Promise.resolve(t), le = /* @__PURE__ */ (() => {
  try {
    return new CSSStyleSheet(), typeof new CSSStyleSheet().replaceSync == "function";
  } catch {
  }
  return !1;
})(), st = [], vt = [], re = (t, e) => (n) => {
  t.push(n), M || (M = !0, e && h.$flags$ & 4 ? ie(z) : h.raf(z));
}, nt = (t) => {
  for (let e = 0; e < t.length; e++)
    try {
      t[e](performance.now());
    } catch (n) {
      w(n);
    }
  t.length = 0;
}, z = () => {
  nt(st), nt(vt), (M = st.length > 0) && h.raf(z);
}, ie = (t) => oe().then(t), ae = /* @__PURE__ */ re(vt, !0), ce = ".wave{display:flex;align-items:center;justify-content:center}.dot{width:4px;height:4px;background-color:var(--dot-color, white);border-radius:50%;margin:0 2px;animation:wave 1.5s infinite ease-in-out}.dot:nth-child(2){animation-delay:0.2s}.dot:nth-child(3){animation-delay:0.4s}@keyframes wave{0%,60%,100%{transform:translateY(0)}30%{transform:translateY(-4px)}}", de = ce, $e = ht(class extends yt {
  constructor() {
    super(), this.__registerHost(), this.__attachShadow(), this.width = "15", this.height = "15";
  }
  render() {
    return k("div", { key: "5ff9aa86f4d346ab894e59e1a4b96756a5f38231", class: "wave" }, k("div", { key: "35fa4038e2be7d2a18e65ada70f40f2d2e1001e8", class: "dot" }), k("div", { key: "f16983cd2790b6a781cdb71e317046f7fa27c264", class: "dot" }), k("div", { key: "c03e50ae47d23c1caa2c9c5c19f6d1ae274de1b7", class: "dot" }));
  }
  static get style() {
    return de;
  }
}, [1, "loading-icon", {
  width: [1],
  height: [1]
}]);
function fe() {
  if (typeof customElements > "u")
    return;
  ["loading-icon"].forEach((e) => {
    switch (e) {
      case "loading-icon":
        customElements.get(e) || customElements.define(e, $e);
        break;
    }
  });
}
const ue = "button.sc-button-component{display:flex;align-items:center;justify-content:center;gap:0.75rem;height:48px;border:none;border-radius:var(--nylas-border-radius-2x);font-size:1rem;line-height:24px;padding:0 1rem;cursor:pointer;font-family:var(--nylas-font-family);font-weight:600;width:100%}@media screen and (max-width: 768px){button.sc-button-component{width:100%}}button.button-primary.sc-button-component{background:var(--nylas-primary);color:var(--nylas-base-0);height:48px;border:none;border-radius:var(--nylas-border-radius-2x);font-size:1rem;line-height:24px;padding:0 1rem;cursor:pointer}button.button-primary.sc-button-component:hover{background:var(--nylas-base-600)}button.button-primary.sc-button-component:active{background:var(--nylas-base-800)}button.button-primary.sc-button-component:disabled{background:var(--nylas-base-300);cursor:not-allowed}button.button-basic.sc-button-component{border:1px solid var(--nylas-base-300);color:var(--nylas-base-700);background:transparent}button.button-basic.sc-button-component:hover{background:transparent;border-color:var(--nylas-primary)}button.button-basic.sc-button-component:active{background:transparent;border-color:var(--nylas-base-600);border-width:2px}button.button-basic.sc-button-component:disabled{background:transparent;border-color:var(--nylas-base-300);color:var(--nylas-base-300);cursor:not-allowed}button.button-destructive.sc-button-component{border:1px solid var(--nylas-base-300);color:var(--nylas-error);background:transparent}button.button-destructive.sc-button-component:hover{background:transparent;border-color:var(--nylas-error)}button.button-destructive.sc-button-component:active{background:transparent;border-color:var(--nylas-error-pressed);border-width:2px;color:var(--nylas-error-pressed)}button.button-destructive.sc-button-component:disabled{background:transparent;border-color:var(--nylas-base-300);color:var(--nylas-base-300);cursor:not-allowed}button.button-invisible.sc-button-component{background:transparent;color:var(--nylas-base-800);border:none}button.button-invisible.sc-button-component:hover{background:var(--nylas-base-50);color:var(--nylas-primary)}button.button-invisible.sc-button-component:active{background:var(--nylas-base-50);color:var(--nylas-base-850)}button.button-invisible.sc-button-component:disabled{background:transparent;color:var(--nylas-base-300);cursor:not-allowed}", pe = ue, mt = ht(class extends yt {
  constructor() {
    super(), this.__registerHost(), this.variant = "primary", this.disabled = !1, this.type = "button", this.isLoading = !1, this.clickHandler = void 0, this.mouseOverHandler = void 0, this.mouseOutHandler = void 0, this.focusHandler = void 0, this.blurHandler = void 0;
  }
  render() {
    const e = `button-${this.variant}` + (this.disabled ? " disabled" : "");
    return k("button", { key: "8e535c0a5cb3be758941eee5ac5bf9b1e65b9e42", class: e, disabled: this.disabled, type: "type", onClick: this.clickHandler, onMouseOver: this.mouseOverHandler, onMouseOut: this.mouseOutHandler, onFocus: this.focusHandler, onBlur: this.blurHandler }, !!this.isLoading && k("loading-icon", { slot: "icon" }), k("slot", { key: "cc75d33abd248961336b47c5ae558c87d1ddce1e" }));
  }
  static get style() {
    return pe;
  }
}, [6, "button-component", {
  variant: [1],
  disabled: [4],
  type: [1],
  isLoading: [4, "is-loading"],
  clickHandler: [16],
  mouseOverHandler: [16],
  mouseOutHandler: [16],
  focusHandler: [16],
  blurHandler: [16]
}]);
function he() {
  if (typeof customElements > "u")
    return;
  ["button-component", "loading-icon"].forEach((e) => {
    switch (e) {
      case "button-component":
        customElements.get(e) || customElements.define(e, mt);
        break;
      case "loading-icon":
        customElements.get(e) || fe();
        break;
    }
  });
}
const ye = mt, ve = he;
export {
  ye as ButtonComponent,
  ve as defineCustomElement
};
