// @nylas/.web-elements-GF2lXkj6/dist/collection/components/design-system/toggle-switch/toggle-switch.js
import { h } from "@stencil/core";
export class ToggleSwitchComponent {
    constructor() {
        this.name = 'toggle-switch';
        this.checked = false;
        this.label = '';
        this.isChecked = this.checked;
    }
    componentWillLoad() {
        this.isChecked = this.checked;
    }
    toggleCheck() {
        this.isChecked = !this.isChecked;
        this.nylasFormSwitchToggled.emit({
            checked: this.isChecked,
            name: this.name,
            label: this.label,
        });
    }
    render() {
        return (h("label", { key: 'c23c77986617f95474b950d7e34e0c8fde52de8e', class: "switch", part: "ts_label" }, h("input", { key: 'e1d164a9413c00949a1e29f12c45a9bdc9ba04e1', type: "checkbox", part: "ts_input", name: this.name, checked: this.isChecked, onChange: () => this.toggleCheck() }), h("div", { key: 'cfa88bfbb93261ffdf1344c4cb78fcfc8b71dd28', class: "slider round", part: "ts_slider" })));
    }
    static get is() { return "toggle-switch"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["toggle-switch.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["toggle-switch.css"]
        };
    }
    static get properties() {
        return {
            "name": {
                "type": "string",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "The name of the toggle switch. This is used to identify the toggle switch when submitting a form."
                },
                "attribute": "name",
                "reflect": false,
                "defaultValue": "'toggle-switch'"
            },
            "checked": {
                "type": "boolean",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "The default value of the toggle switch. This is the value that is displayed when the toggle switch is rendered."
                },
                "attribute": "checked",
                "reflect": false,
                "defaultValue": "false"
            },
            "label": {
                "type": "string",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "The label of the toggle switch. This is displayed next to the toggle switch."
                },
                "attribute": "label",
                "reflect": false,
                "defaultValue": "''"
            }
        };
    }
    static get states() {
        return {
            "isChecked": {}
        };
    }
    static get events() {
        return [{
                "method": "nylasFormSwitchToggled",
                "name": "nylasFormSwitchToggled",
                "bubbles": true,
                "cancelable": true,
                "composed": true,
                "docs": {
                    "tags": [],
                    "text": "This event is fired when the toggle switch is toggled."
                },
                "complexType": {
                    "original": "{\n    checked: boolean;\n    name: string;\n    label: string;\n  }",
                    "resolved": "{ checked: boolean; name: string; label: string; }",
                    "references": {}
                }
            }];
    }
}
