// @nylas/.web-elements-GF2lXkj6/dist/collection/components/scheduler-editor/nylas-event-limits/nylas-event-limits.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
        return Reflect.metadata(k, v);
};
import { RegisterComponent } from "../../../common/register-component";
import { checkForMissingSlots, debug } from "../../../utils/utils";
import { Host, h } from "@stencil/core";
export class NylasEventLimits {
    connectedCallback() {
        debug('nylas-event-limits', 'connectedCallback');
    }
    componentWillLoad() {
        debug('nylas-event-limits', 'componentWillLoad');
    }
    componentDidLoad() {
        debug('nylas-event-limits', 'componentDidLoad');
        checkForMissingSlots(['inputs'], this.el);
    }
    disconnectedCallback() {
        debug('nylas-event-limits', 'disconnectedCallback');
    }
    render() {
        return (h(Host, { key: 'ac880b066a9bc24e69f04bcb56367fc474ba42f3' }, h("nylas-form-card", { key: '6d1772fb3cf067055bb5fb5a77dcae7c68576a64', exportparts: "nfc__header: nel__header" }, h("h1", { key: '1f3315eb09333a10c0696bff3ce6e788e2abad4d', slot: "header-title", class: "nylas-event-limits__title" }, "Event limits"), h("div", { key: 'aa6cc3f24a9af450ac5f097a7093ab7ec3bec11f', slot: "content", class: "nylas-event-limits__content" }, h("slot", { key: 'faca426e5abf0c7eda9a4221c77475bc0b787f6d', name: "inputs" })))));
    }
    static get is() { return "nylas-event-limits"; }
    static get encapsulation() { return "shadow"; }
    static get formAssociated() { return true; }
    static get originalStyleUrls() {
        return {
            "$": ["nylas-event-limits.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["nylas-event-limits.css"]
        };
    }
    static get elementRef() { return "el"; }
    static get attachInternalsMemberName() { return "internals"; }
}
__decorate([
    RegisterComponent({
        name: 'nylas-event-limits',
        eventToProps: {},
        fireRegisterEvent: true,
    }),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", []),
    __metadata("design:returntype", void 0)
], NylasEventLimits.prototype, "render", null);
