// @nylas/.web-elements-GF2lXkj6/dist/cdn/star/star.es.js
const L = {
  allRenderFn: !1,
  cmpDidLoad: !0,
  cmpDidUnload: !1,
  cmpDidUpdate: !0,
  cmpDidRender: !0,
  cmpWillLoad: !0,
  cmpWillUpdate: !0,
  cmpWillRender: !0,
  connectedCallback: !0,
  disconnectedCallback: !0,
  element: !0,
  event: !0,
  hasRenderFn: !0,
  lifecycle: !0,
  hostListener: !0,
  hostListenerTargetWindow: !0,
  hostListenerTargetDocument: !0,
  hostListenerTargetBody: !0,
  hostListenerTargetParent: !1,
  hostListenerTarget: !0,
  member: !0,
  method: !0,
  mode: !0,
  observeAttribute: !0,
  prop: !0,
  propMutable: !0,
  reflect: !0,
  scoped: !0,
  shadowDom: !0,
  slot: !0,
  cssAnnotations: !0,
  state: !0,
  style: !0,
  formAssociated: !1,
  svg: !0,
  updatable: !0,
  vdomAttribute: !0,
  vdomXlink: !0,
  vdomClass: !0,
  vdomFunctional: !0,
  vdomKey: !0,
  vdomListener: !0,
  vdomRef: !0,
  vdomPropOrAttr: !0,
  vdomRender: !0,
  vdomStyle: !0,
  vdomText: !0,
  watchCallback: !0,
  taskQueue: !0,
  hotModuleReplacement: !1,
  isDebug: !1,
  isDev: !1,
  isTesting: !1,
  hydrateServerSide: !1,
  hydrateClientSide: !1,
  lifecycleDOMEvents: !1,
  lazyLoad: !1,
  profile: !1,
  slotRelocation: !0,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  appendChildSlotFix: !1,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  cloneNodeFix: !1,
  hydratedAttribute: !1,
  hydratedClass: !0,
  scriptDataOpts: !1,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  scopedSlotTextContentFix: !1,
  // TODO(STENCIL-854): Remove code related to legacy shadowDomShim field
  shadowDomShim: !1,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  slotChildNodesFix: !1,
  invisiblePrehydration: !0,
  propBoolean: !0,
  propNumber: !0,
  propString: !0,
  constructableCSS: !0,
  cmpShouldUpdate: !0,
  devTools: !1,
  shadowDelegatesFocus: !0,
  initializeNextTick: !1,
  asyncLoading: !1,
  asyncQueue: !1,
  transformTagName: !1,
  attachStyles: !0,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  experimentalSlotFixes: !1
};
let x, H, _, lt = !1, U = !1, W = !1, v = !1, I = null, M = !1;
const b = (t, e = "") => () => {
}, vt = "slot-fb{display:contents}slot-fb[hidden]{display:none}", Q = "http://www.w3.org/1999/xlink", X = {}, St = "http://www.w3.org/2000/svg", kt = "http://www.w3.org/1999/xhtml", mt = (t) => t != null, z = (t) => (t = typeof t, t === "object" || t === "function");
function Lt(t) {
  var e, n, s;
  return (s = (n = (e = t.head) === null || e === void 0 ? void 0 : e.querySelector('meta[name="csp-nonce"]')) === null || n === void 0 ? void 0 : n.getAttribute("content")) !== null && s !== void 0 ? s : void 0;
}
const w = (t, e, ...n) => {
  let s = null, l = null, r = null, o = !1, c = !1;
  const i = [], u = ($) => {
    for (let d = 0; d < $.length; d++)
      s = $[d], Array.isArray(s) ? u(s) : s != null && typeof s != "boolean" && ((o = typeof t != "function" && !z(s)) && (s = String(s)), o && c ? i[i.length - 1].$text$ += s : i.push(o ? P(null, s) : s), c = o);
  };
  if (u(n), e) {
    e.key && (l = e.key), e.name && (r = e.name);
    {
      const $ = e.className || e.class;
      $ && (e.class = typeof $ != "object" ? $ : Object.keys($).filter((d) => $[d]).join(" "));
    }
  }
  if (typeof t == "function")
    return t(e === null ? {} : e, i, xt);
  const f = P(t, null);
  return f.$attrs$ = e, i.length > 0 && (f.$children$ = i), f.$key$ = l, f.$name$ = r, f;
}, P = (t, e) => {
  const n = {
    $flags$: 0,
    $tag$: t,
    $text$: e,
    $elm$: null,
    $children$: null
  };
  return n.$attrs$ = null, n.$key$ = null, n.$name$ = null, n;
}, Tt = {}, bt = (t) => t && t.$tag$ === Tt, xt = {
  forEach: (t, e) => t.map(G).forEach(e),
  map: (t, e) => t.map(G).map(e).map(Ct)
}, G = (t) => ({
  vattrs: t.$attrs$,
  vchildren: t.$children$,
  vkey: t.$key$,
  vname: t.$name$,
  vtag: t.$tag$,
  vtext: t.$text$
}), Ct = (t) => {
  if (typeof t.vtag == "function") {
    const n = Object.assign({}, t.vattrs);
    return t.vkey && (n.key = t.vkey), t.vname && (n.name = t.vname), w(t.vtag, n, ...t.vchildren || []);
  }
  const e = P(t.vtag, t.vtext);
  return e.$attrs$ = t.vattrs, e.$children$ = t.vchildren, e.$key$ = t.vkey, e.$name$ = t.vname, e;
}, Et = (t) => te.map((e) => e(t)).find((e) => !!e), _t = (t, e) => t != null && !z(t) ? e & 4 ? t === "false" ? !1 : t === "" || !!t : e & 2 ? parseFloat(t) : e & 1 ? String(t) : t : t, K = /* @__PURE__ */ new WeakMap(), At = (t, e, n) => {
  let s = B.get(t);
  le && n ? (s = s || new CSSStyleSheet(), typeof s == "string" ? s = e : s.replaceSync(e)) : s = e, B.set(t, s);
}, Rt = (t, e, n) => {
  var s;
  const l = ot(e, n), r = B.get(l);
  if (t = t.nodeType === 11 ? t : S, r)
    if (typeof r == "string") {
      t = t.head || t;
      let o = K.get(t), c;
      if (o || K.set(t, o = /* @__PURE__ */ new Set()), !o.has(l)) {
        {
          c = S.createElement("style"), c.innerHTML = r;
          const i = (s = p.$nonce$) !== null && s !== void 0 ? s : Lt(S);
          i != null && c.setAttribute("nonce", i), t.insertBefore(c, t.querySelector("link"));
        }
        e.$flags$ & 4 && (c.innerHTML += vt), o && o.add(l);
      }
    } else t.adoptedStyleSheets.includes(r) || (t.adoptedStyleSheets = [...t.adoptedStyleSheets, r]);
  return l;
}, Ot = (t) => {
  const e = t.$cmpMeta$, n = t.$hostElement$, s = e.$flags$, l = b("attachStyles", e.$tagName$), r = Rt(n.shadowRoot ? n.shadowRoot : n.getRootNode(), e, t.$modeName$);
  s & 10 && (n["s-sc"] = r, n.classList.add(r + "-h"), s & 2 && n.classList.add(r + "-s")), l();
}, ot = (t, e) => "sc-" + (e && t.$flags$ & 32 ? t.$tagName$ + "-" + e : t.$tagName$), J = (t, e, n, s, l, r) => {
  if (n !== s) {
    let o = et(t, e), c = e.toLowerCase();
    if (e === "class") {
      const i = t.classList, u = Y(n), f = Y(s);
      i.remove(...u.filter(($) => $ && !f.includes($))), i.add(...f.filter(($) => $ && !u.includes($)));
    } else if (e === "style") {
      for (const i in n)
        (!s || s[i] == null) && (i.includes("-") ? t.style.removeProperty(i) : t.style[i] = "");
      for (const i in s)
        (!n || s[i] !== n[i]) && (i.includes("-") ? t.style.setProperty(i, s[i]) : t.style[i] = s[i]);
    } else if (e !== "key")
      if (e === "ref")
        s && s(t);
      else if (!t.__lookupSetter__(e) && e[0] === "o" && e[1] === "n") {
        if (e[2] === "-" ? e = e.slice(3) : et(D, c) ? e = c.slice(2) : e = c[2] + e.slice(3), n || s) {
          const i = e.endsWith(it);
          e = e.replace(Ut, ""), n && p.rel(t, e, n, i), s && p.ael(t, e, s, i);
        }
      } else {
        const i = z(s);
        if ((o || i && s !== null) && !l)
          try {
            if (t.tagName.includes("-"))
              t[e] = s;
            else {
              const f = s ?? "";
              e === "list" ? o = !1 : (n == null || t[e] != f) && (t[e] = f);
            }
          } catch {
          }
        let u = !1;
        c !== (c = c.replace(/^xlink\:?/, "")) && (e = c, u = !0), s == null || s === !1 ? (s !== !1 || t.getAttribute(e) === "") && (u ? t.removeAttributeNS(Q, e) : t.removeAttribute(e)) : (!o || r & 4 || l) && !i && (s = s === !0 ? "" : s, u ? t.setAttributeNS(Q, e, s) : t.setAttribute(e, s));
      }
  }
}, Ft = /\s/, Y = (t) => t ? t.split(Ft) : [], it = "Capture", Ut = new RegExp(it + "$"), rt = (t, e, n, s) => {
  const l = e.$elm$.nodeType === 11 && e.$elm$.host ? e.$elm$.host : e.$elm$, r = t && t.$attrs$ || X, o = e.$attrs$ || X;
  for (s in r)
    s in o || J(l, s, r[s], void 0, n, e.$flags$);
  for (s in o)
    J(l, s, r[s], o[s], n, e.$flags$);
}, j = (t, e, n, s) => {
  const l = e.$children$[n];
  let r = 0, o, c, i;
  if (lt || (W = !0, l.$tag$ === "slot" && (x && s.classList.add(x + "-s"), l.$flags$ |= l.$children$ ? (
    // slot element has fallback content
    2
  ) : (
    // slot element does not have fallback content
    1
  ))), l.$text$ !== null)
    o = l.$elm$ = S.createTextNode(l.$text$);
  else if (l.$flags$ & 1)
    o = l.$elm$ = S.createTextNode("");
  else {
    if (v || (v = l.$tag$ === "svg"), o = l.$elm$ = S.createElementNS(v ? St : kt, l.$flags$ & 2 ? "slot-fb" : l.$tag$), v && l.$tag$ === "foreignObject" && (v = !1), rt(null, l, v), mt(x) && o["s-si"] !== x && o.classList.add(o["s-si"] = x), l.$children$)
      for (r = 0; r < l.$children$.length; ++r)
        c = j(t, l, r, o), c && o.appendChild(c);
    l.$tag$ === "svg" ? v = !1 : o.tagName === "foreignObject" && (v = !0);
  }
  return o["s-hn"] = _, l.$flags$ & 3 && (o["s-sr"] = !0, o["s-cr"] = H, o["s-sn"] = l.$name$ || "", i = t && t.$children$ && t.$children$[n], i && i.$tag$ === l.$tag$ && t.$elm$ && A(t.$elm$, !1)), o;
}, A = (t, e) => {
  p.$flags$ |= 1;
  const n = t.childNodes;
  for (let s = n.length - 1; s >= 0; s--) {
    const l = n[s];
    l["s-hn"] !== _ && l["s-ol"] && (at(l).insertBefore(l, q(l)), l["s-ol"].remove(), l["s-ol"] = void 0, l["s-sh"] = void 0, W = !0), e && A(l, e);
  }
  p.$flags$ &= -2;
}, ct = (t, e, n, s, l, r) => {
  let o = t["s-cr"] && t["s-cr"].parentNode || t, c;
  for (o.shadowRoot && o.tagName === _ && (o = o.shadowRoot); l <= r; ++l)
    s[l] && (c = j(null, n, l, t), c && (s[l].$elm$ = c, o.insertBefore(c, q(e))));
}, $t = (t, e, n) => {
  for (let s = e; s <= n; ++s) {
    const l = t[s];
    if (l) {
      const r = l.$elm$;
      ut(l), r && (U = !0, r["s-ol"] ? r["s-ol"].remove() : A(r, !0), r.remove());
    }
  }
}, wt = (t, e, n, s, l = !1) => {
  let r = 0, o = 0, c = 0, i = 0, u = e.length - 1, f = e[0], $ = e[u], d = s.length - 1, a = s[0], h = s[d], y, g;
  for (; r <= u && o <= d; )
    if (f == null)
      f = e[++r];
    else if ($ == null)
      $ = e[--u];
    else if (a == null)
      a = s[++o];
    else if (h == null)
      h = s[--d];
    else if (F(f, a, l))
      C(f, a, l), f = e[++r], a = s[++o];
    else if (F($, h, l))
      C($, h, l), $ = e[--u], h = s[--d];
    else if (F(f, h, l))
      (f.$tag$ === "slot" || h.$tag$ === "slot") && A(f.$elm$.parentNode, !1), C(f, h, l), t.insertBefore(f.$elm$, $.$elm$.nextSibling), f = e[++r], h = s[--d];
    else if (F($, a, l))
      (f.$tag$ === "slot" || h.$tag$ === "slot") && A($.$elm$.parentNode, !1), C($, a, l), t.insertBefore($.$elm$, f.$elm$), $ = e[--u], a = s[++o];
    else {
      for (c = -1, i = r; i <= u; ++i)
        if (e[i] && e[i].$key$ !== null && e[i].$key$ === a.$key$) {
          c = i;
          break;
        }
      c >= 0 ? (g = e[c], g.$tag$ !== a.$tag$ ? y = j(e && e[o], n, c, t) : (C(g, a, l), e[c] = void 0, y = g.$elm$), a = s[++o]) : (y = j(e && e[o], n, o, t), a = s[++o]), y && at(f.$elm$).insertBefore(y, q(f.$elm$));
    }
  r > u ? ct(t, s[d + 1] == null ? null : s[d + 1].$elm$, n, s, o, d) : o > d && $t(e, r, u);
}, F = (t, e, n = !1) => t.$tag$ === e.$tag$ ? t.$tag$ === "slot" ? t.$name$ === e.$name$ : n ? !0 : t.$key$ === e.$key$ : !1, q = (t) => t && t["s-ol"] || t, at = (t) => (t["s-ol"] ? t["s-ol"] : t).parentNode, C = (t, e, n = !1) => {
  const s = e.$elm$ = t.$elm$, l = t.$children$, r = e.$children$, o = e.$tag$, c = e.$text$;
  let i;
  c === null ? (v = o === "svg" ? !0 : o === "foreignObject" ? !1 : v, o === "slot" || rt(t, e, v), l !== null && r !== null ? wt(s, l, e, r, n) : r !== null ? (t.$text$ !== null && (s.textContent = ""), ct(s, null, e, r, 0, r.length - 1)) : l !== null && $t(l, 0, l.length - 1), v && o === "svg" && (v = !1)) : (i = s["s-cr"]) ? i.parentNode.textContent = c : t.$text$ !== c && (s.data = c);
}, ft = (t) => {
  const e = t.childNodes;
  for (const n of e)
    if (n.nodeType === 1) {
      if (n["s-sr"]) {
        const s = n["s-sn"];
        n.hidden = !1;
        for (const l of e)
          if (l !== n) {
            if (l["s-hn"] !== n["s-hn"] || s !== "") {
              if (l.nodeType === 1 && (s === l.getAttribute("slot") || s === l["s-sn"])) {
                n.hidden = !0;
                break;
              }
            } else if (l.nodeType === 1 || l.nodeType === 3 && l.textContent.trim() !== "") {
              n.hidden = !0;
              break;
            }
          }
      }
      ft(n);
    }
}, k = [], dt = (t) => {
  let e, n, s;
  for (const l of t.childNodes) {
    if (l["s-sr"] && (e = l["s-cr"]) && e.parentNode) {
      n = e.parentNode.childNodes;
      const r = l["s-sn"];
      for (s = n.length - 1; s >= 0; s--)
        if (e = n[s], !e["s-cn"] && !e["s-nr"] && e["s-hn"] !== l["s-hn"] && !L.experimentalSlotFixes)
          if (Z(e, r)) {
            let o = k.find((c) => c.$nodeToRelocate$ === e);
            U = !0, e["s-sn"] = e["s-sn"] || r, o ? (o.$nodeToRelocate$["s-sh"] = l["s-hn"], o.$slotRefNode$ = l) : (e["s-sh"] = l["s-hn"], k.push({
              $slotRefNode$: l,
              $nodeToRelocate$: e
            })), e["s-sr"] && k.map((c) => {
              Z(c.$nodeToRelocate$, e["s-sn"]) && (o = k.find((i) => i.$nodeToRelocate$ === e), o && !c.$slotRefNode$ && (c.$slotRefNode$ = o.$slotRefNode$));
            });
          } else k.some((o) => o.$nodeToRelocate$ === e) || k.push({
            $nodeToRelocate$: e
          });
    }
    l.nodeType === 1 && dt(l);
  }
}, Z = (t, e) => t.nodeType === 1 ? t.getAttribute("slot") === null && e === "" || t.getAttribute("slot") === e : t["s-sn"] === e ? !0 : e === "", ut = (t) => {
  t.$attrs$ && t.$attrs$.ref && t.$attrs$.ref(null), t.$children$ && t.$children$.map(ut);
}, Pt = (t, e, n = !1) => {
  var s, l, r, o, c;
  const i = t.$hostElement$, u = t.$cmpMeta$, f = t.$vnode$ || P(null, null), $ = bt(e) ? e : w(null, null, e);
  if (_ = i.tagName, u.$attrsToReflect$ && ($.$attrs$ = $.$attrs$ || {}, u.$attrsToReflect$.map(([d, a]) => $.$attrs$[a] = i[d])), n && $.$attrs$)
    for (const d of Object.keys($.$attrs$))
      i.hasAttribute(d) && !["key", "ref", "style", "class"].includes(d) && ($.$attrs$[d] = i[d]);
  $.$tag$ = null, $.$flags$ |= 4, t.$vnode$ = $, $.$elm$ = f.$elm$ = i.shadowRoot || i, x = i["s-sc"], H = i["s-cr"], lt = (u.$flags$ & 1) !== 0, U = !1, C(f, $, n);
  {
    if (p.$flags$ |= 1, W) {
      dt($.$elm$);
      for (const d of k) {
        const a = d.$nodeToRelocate$;
        if (!a["s-ol"]) {
          const h = S.createTextNode("");
          h["s-nr"] = a, a.parentNode.insertBefore(a["s-ol"] = h, a);
        }
      }
      for (const d of k) {
        const a = d.$nodeToRelocate$, h = d.$slotRefNode$;
        if (h) {
          const y = h.parentNode;
          let g = h.nextSibling;
          {
            let T = (s = a["s-ol"]) === null || s === void 0 ? void 0 : s.previousSibling;
            for (; T; ) {
              let m = (l = T["s-nr"]) !== null && l !== void 0 ? l : null;
              if (m && m["s-sn"] === a["s-sn"] && y === m.parentNode && (m = m.nextSibling, !m || !m["s-nr"])) {
                g = m;
                break;
              }
              T = T.previousSibling;
            }
          }
          (!g && y !== a.parentNode || a.nextSibling !== g) && a !== g && (!a["s-hn"] && a["s-ol"] && (a["s-hn"] = a["s-ol"].parentNode.nodeName), y.insertBefore(a, g), a.nodeType === 1 && (a.hidden = (r = a["s-ih"]) !== null && r !== void 0 ? r : !1));
        } else
          a.nodeType === 1 && (n && (a["s-ih"] = (o = a.hidden) !== null && o !== void 0 ? o : !1), a.hidden = !0);
      }
    }
    U && ft($.$elm$), p.$flags$ &= -2, k.length = 0;
  }
  if (L.experimentalScopedSlotChanges && u.$flags$ & 2)
    for (const d of $.$elm$.childNodes)
      d["s-hn"] !== _ && !d["s-sh"] && (n && d["s-ih"] == null && (d["s-ih"] = (c = d.hidden) !== null && c !== void 0 ? c : !1), d.hidden = !0);
  H = void 0;
}, jt = (t, e) => {
}, ht = (t, e) => (t.$flags$ |= 16, jt(t, t.$ancestorComponent$), re(() => Bt(t, e))), Bt = (t, e) => {
  const n = t.$hostElement$, s = b("scheduleUpdate", t.$cmpMeta$.$tagName$), l = n;
  let r;
  return e ? r = E(l, "componentWillLoad") : r = E(l, "componentWillUpdate"), r = V(r, () => E(l, "componentWillRender")), s(), V(r, () => Ht(t, l, e));
}, V = (t, e) => Dt(t) ? t.then(e) : e(), Dt = (t) => t instanceof Promise || t && t.then && typeof t.then == "function", Ht = async (t, e, n) => {
  const s = t.$hostElement$, l = b("update", t.$cmpMeta$.$tagName$);
  s["s-rc"], n && Ot(t);
  const r = b("render", t.$cmpMeta$.$tagName$);
  Mt(t, e, s, n), r(), l(), Nt(t);
}, Mt = (t, e, n, s) => {
  try {
    I = e, e = e.render && e.render(), t.$flags$ &= -17, t.$flags$ |= 2, (L.hasRenderFn || L.reflect) && (L.vdomRender || L.reflect) && (L.hydrateServerSide || Pt(t, e, s));
  } catch (i) {
    O(i, t.$hostElement$);
  }
  return I = null, null;
}, Nt = (t) => {
  const e = t.$cmpMeta$.$tagName$, n = t.$hostElement$, s = b("postUpdate", e), l = n;
  t.$ancestorComponent$, E(l, "componentDidRender"), t.$flags$ & 64 ? (E(l, "componentDidUpdate"), s()) : (t.$flags$ |= 64, E(l, "componentDidLoad"), s());
}, E = (t, e, n) => {
  if (t && t[e])
    try {
      return t[e](n);
    } catch (s) {
      O(s);
    }
}, Wt = (t, e) => R(t).$instanceValues$.get(e), zt = (t, e, n, s) => {
  const l = R(t), r = t, o = l.$instanceValues$.get(e), c = l.$flags$, i = r;
  n = _t(n, s.$members$[e][0]);
  const u = Number.isNaN(o) && Number.isNaN(n);
  if (n !== o && !u) {
    l.$instanceValues$.set(e, n);
    {
      if (s.$watchers$ && c & 128) {
        const $ = s.$watchers$[e];
        $ && $.map((d) => {
          try {
            i[d](n, o, e);
          } catch (a) {
            O(a, r);
          }
        });
      }
      if ((c & 18) === 2) {
        if (i.componentShouldUpdate && i.componentShouldUpdate(n, o, e) === !1)
          return;
        ht(l, !1);
      }
    }
  }
}, qt = (t, e, n) => {
  var s;
  const l = t.prototype;
  if (e.$members$) {
    t.watchers && (e.$watchers$ = t.watchers);
    const r = Object.entries(e.$members$);
    r.map(([o, [c]]) => {
      (c & 31 || c & 32) && Object.defineProperty(l, o, {
        get() {
          return Wt(this, o);
        },
        set(i) {
          zt(this, o, i, e);
        },
        configurable: !0,
        enumerable: !0
      });
    });
    {
      const o = /* @__PURE__ */ new Map();
      l.attributeChangedCallback = function(c, i, u) {
        p.jmp(() => {
          var f;
          const $ = o.get(c);
          if (this.hasOwnProperty($))
            u = this[$], delete this[$];
          else {
            if (l.hasOwnProperty($) && typeof this[$] == "number" && this[$] == u)
              return;
            if ($ == null) {
              const d = R(this), a = d == null ? void 0 : d.$flags$;
              if (a && !(a & 8) && a & 128 && u !== i) {
                const y = this, g = (f = e.$watchers$) === null || f === void 0 ? void 0 : f[c];
                g == null || g.forEach((T) => {
                  y[T] != null && y[T].call(y, u, i, c);
                });
              }
              return;
            }
          }
          this[$] = u === null && typeof this[$] == "boolean" ? !1 : u;
        });
      }, t.observedAttributes = Array.from(/* @__PURE__ */ new Set([
        ...Object.keys((s = e.$watchers$) !== null && s !== void 0 ? s : {}),
        ...r.filter(
          ([c, i]) => i[0] & 15
          /* MEMBER_FLAGS.HasAttribute */
        ).map(([c, i]) => {
          var u;
          const f = i[1] || c;
          return o.set(f, c), i[0] & 512 && ((u = e.$attrsToReflect$) === null || u === void 0 || u.push([c, f])), f;
        })
      ]));
    }
  }
  return t;
}, It = async (t, e, n, s) => {
  let l;
  if (!(e.$flags$ & 32) && (e.$flags$ |= 32, l = t.constructor, customElements.whenDefined(n.$tagName$).then(() => e.$flags$ |= 128), l.style)) {
    let o = l.style;
    typeof o != "string" && (o = o[e.$modeName$ = Et(t)]);
    const c = ot(n, e.$modeName$);
    if (!B.has(c)) {
      const i = b("registerStyles", n.$tagName$);
      At(c, o, !!(n.$flags$ & 1)), i();
    }
  }
  e.$ancestorComponent$, ht(e, !0);
}, tt = (t) => {
}, Qt = (t) => {
  if (!(p.$flags$ & 1)) {
    const e = R(t), n = e.$cmpMeta$, s = b("connectedCallback", n.$tagName$);
    e.$flags$ & 1 ? (pt(t, e, n.$listeners$), e != null && e.$lazyInstance$ ? tt(e.$lazyInstance$) : e != null && e.$onReadyPromise$ && e.$onReadyPromise$.then(() => tt(e.$lazyInstance$))) : (e.$flags$ |= 1, // TODO(STENCIL-854): Remove code related to legacy shadowDomShim field
    n.$flags$ & 12 && Xt(t), n.$members$ && Object.entries(n.$members$).map(([l, [r]]) => {
      if (r & 31 && t.hasOwnProperty(l)) {
        const o = t[l];
        delete t[l], t[l] = o;
      }
    }), It(t, e, n)), s();
  }
}, Xt = (t) => {
  const e = t["s-cr"] = S.createComment("");
  e["s-cn"] = !0, t.insertBefore(e, t.firstChild);
}, Gt = async (t) => {
  if (!(p.$flags$ & 1)) {
    const e = R(t);
    e.$rmListeners$ && (e.$rmListeners$.map((n) => n()), e.$rmListeners$ = void 0);
  }
}, Kt = (t, e) => {
  const n = {
    $flags$: e[0],
    $tagName$: e[1]
  };
  n.$members$ = e[2], n.$listeners$ = e[3], n.$watchers$ = t.$watchers$, n.$attrsToReflect$ = [];
  const s = t.prototype.connectedCallback, l = t.prototype.disconnectedCallback;
  return Object.assign(t.prototype, {
    __registerHost() {
      Vt(this, n);
    },
    connectedCallback() {
      Qt(this), s && s.call(this);
    },
    disconnectedCallback() {
      Gt(this), l && l.call(this);
    },
    __attachShadow() {
      this.attachShadow({
        mode: "open",
        delegatesFocus: !!(n.$flags$ & 16)
      });
    }
  }), t.is = n.$tagName$, qt(t, n);
}, pt = (t, e, n, s) => {
  n && n.map(([l, r, o]) => {
    const c = Yt(t, l), i = Jt(e, o), u = Zt(l);
    p.ael(c, r, i, u), (e.$rmListeners$ = e.$rmListeners$ || []).push(() => p.rel(c, r, i, u));
  });
}, Jt = (t, e) => (n) => {
  try {
    L.lazyLoad || t.$hostElement$[e](n);
  } catch (s) {
    O(s);
  }
}, Yt = (t, e) => e & 4 ? S : e & 8 ? D : e & 16 ? S.body : t, Zt = (t) => se ? {
  passive: (t & 1) !== 0,
  capture: (t & 2) !== 0
} : (t & 2) !== 0, gt = /* @__PURE__ */ new WeakMap(), R = (t) => gt.get(t), Vt = (t, e) => {
  const n = {
    $flags$: 0,
    $hostElement$: t,
    $cmpMeta$: e,
    $instanceValues$: /* @__PURE__ */ new Map()
  };
  return pt(t, n, e.$listeners$), gt.set(t, n);
}, et = (t, e) => e in t, O = (t, e) => (0, console.error)(t, e), B = /* @__PURE__ */ new Map(), te = [], D = typeof window < "u" ? window : {}, S = D.document || { head: {} }, ee = D.HTMLElement || class {
}, p = {
  $flags$: 0,
  $resourcesUrl$: "",
  jmp: (t) => t(),
  raf: (t) => requestAnimationFrame(t),
  ael: (t, e, n, s) => t.addEventListener(e, n, s),
  rel: (t, e, n, s) => t.removeEventListener(e, n, s),
  ce: (t, e) => new CustomEvent(t, e)
}, se = /* @__PURE__ */ (() => {
  let t = !1;
  try {
    S.addEventListener("e", null, Object.defineProperty({}, "passive", {
      get() {
        t = !0;
      }
    }));
  } catch {
  }
  return t;
})(), ne = (t) => Promise.resolve(t), le = /* @__PURE__ */ (() => {
  try {
    return new CSSStyleSheet(), typeof new CSSStyleSheet().replaceSync == "function";
  } catch {
  }
  return !1;
})(), st = [], yt = [], oe = (t, e) => (n) => {
  t.push(n), M || (M = !0, e && p.$flags$ & 4 ? ie(N) : p.raf(N));
}, nt = (t) => {
  for (let e = 0; e < t.length; e++)
    try {
      t[e](performance.now());
    } catch (n) {
      O(n);
    }
  t.length = 0;
}, N = () => {
  nt(st), nt(yt), (M = st.length > 0) && p.raf(N);
}, ie = (t) => ne().then(t), re = /* @__PURE__ */ oe(yt, !0), ce = ".sc-star-icon-h{display:flex}", $e = ce, ae = Kt(class extends ee {
  constructor() {
    super(), this.__registerHost(), this.width = "15", this.height = "15";
  }
  render() {
    return w("svg", { key: "298c7143a575df19f6e9fa155cb4c5377a074d9e", width: this.width, height: this.height, viewBox: "0 0 24 24", fill: "auto", xmlns: "http://www.w3.org/2000/svg" }, w("path", { key: "8aa881c0befd74349f594ab11a428b9a901d53f8", d: "M9.15316 5.40838C10.4198 3.13613 11.0531 2 12 2C12.9469 2 13.5802 3.13612 14.8468 5.40837L15.1745 5.99623C15.5345 6.64193 15.7144 6.96479 15.9951 7.17781C16.2757 7.39083 16.6251 7.4699 17.3241 7.62805L17.9605 7.77203C20.4201 8.32856 21.65 8.60682 21.9426 9.54773C22.2352 10.4886 21.3968 11.4691 19.7199 13.4299L19.2861 13.9372C18.8096 14.4944 18.5713 14.773 18.4641 15.1177C18.357 15.4624 18.393 15.8341 18.465 16.5776L18.5306 17.2544C18.7841 19.8706 18.9109 21.1787 18.1449 21.7602C17.3788 22.3417 16.2273 21.8115 13.9243 20.7512L13.3285 20.4768C12.6741 20.1755 12.3469 20.0248 12 20.0248C11.6531 20.0248 11.3259 20.1755 10.6715 20.4768L10.0757 20.7512C7.77268 21.8115 6.62118 22.3417 5.85515 21.7602C5.08912 21.1787 5.21588 19.8706 5.4694 17.2544L5.53498 16.5776C5.60703 15.8341 5.64305 15.4624 5.53586 15.1177C5.42868 14.773 5.19043 14.4944 4.71392 13.9372L4.2801 13.4299C2.60325 11.4691 1.76482 10.4886 2.05742 9.54773C2.35002 8.60682 3.57986 8.32856 6.03954 7.77203L6.67589 7.62805C7.37485 7.4699 7.72433 7.39083 8.00494 7.17781C8.28555 6.96479 8.46553 6.64194 8.82547 5.99623L9.15316 5.40838Z", stroke: "auto", "stroke-width": "1.5" }));
  }
  static get style() {
    return $e;
  }
}, [2, "star-icon", {
  width: [1],
  height: [1]
}]);
function de() {
  if (typeof customElements > "u")
    return;
  ["star-icon"].forEach((e) => {
    switch (e) {
      case "star-icon":
        customElements.get(e) || customElements.define(e, ae);
        break;
    }
  });
}
export {
  ae as S,
  de as d
};
