// @nylas/.web-elements-GF2lXkj6/dist/collection/components/scheduler-editor/nylas-form-card/nylas-form-card.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
        return Reflect.metadata(k, v);
};
import { RegisterComponent } from "../../../common/register-component";
import { checkForMissingSlots, debug } from "../../../utils/utils";
import { Host, h } from "@stencil/core";
export class NylasFormCard {
    connectedCallback() {
        debug('nylas-event-card', 'connectedCallback');
    }
    componentWillLoad() {
        debug('nylas-event-card', 'componentWillLoad');
    }
    componentDidLoad() {
        debug('nylas-event-card', 'componentDidLoad');
    }
    disconnectedCallback() {
        debug('nylas-event-card', 'disconnectedCallback');
        const slots = ['header-title', 'header-subtitle', 'content'];
        checkForMissingSlots(slots, this.host);
    }
    render() {
        return (h(Host, { key: 'ceadfc37c47ece06fb9e188692f12d9b7c1c069d' }, h("div", { key: '2f05c5a7258c800b26dc55d6ffd2d3505dd746c7', class: "nylas-form-card", part: "nfc" }, h("div", { key: '96de3e134413e04a4631d013bb5c3a03c2b07b10', class: "nylas-form-card__header", part: "nfc__header" }, h("slot", { key: 'bd3dc03079117538110f3011775eee108b41da9e', name: "header-title" }), h("slot", { key: '9c140e29b9ff9d8e7ea214dcd0d1eb9c4b752e20', name: "header-subtitle" })), h("div", { key: '1c75e8768e0b59ae4fd2b18f301905ecc8595a82', class: "nylas-form-card__content" }, h("slot", { key: '2b4c082461c47313cb506dced2018af660a20ec8', name: "content" })))));
    }
    static get is() { return "nylas-form-card"; }
    static get encapsulation() { return "shadow"; }
    static get formAssociated() { return true; }
    static get originalStyleUrls() {
        return {
            "$": ["nylas-form-card.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["nylas-form-card.css"]
        };
    }
    static get elementRef() { return "host"; }
    static get attachInternalsMemberName() { return "internals"; }
}
__decorate([
    RegisterComponent({
        name: 'nylas-form-card',
        eventToProps: {},
        fireRegisterEvent: true,
    }),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", []),
    __metadata("design:returntype", void 0)
], NylasFormCard.prototype, "render", null);
