// @nylas/.web-elements-GF2lXkj6/dist/collection/components/mailbox/nylas-threads-search/nylas-threads-search.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
        return Reflect.metadata(k, v);
};
import { RegisterComponent } from "../../../common/register-component";
import { debug } from "../../../utils/utils";
import { Host, h } from "@stencil/core";
export class NylasThreadsSearch {
    constructor() {
        this.onSearch = async (event) => {
            event.preventDefault();
            event.stopPropagation();
            const target = event.target;
            const value = target.value;
            if (value === this.internalQuery) {
                return;
            }
            this.internalQuery = value;
            this.search.emit(this.internalQuery);
        };
        this.query = '';
        this.internalQuery = '';
    }
    connectedCallback() { }
    disconnectedCallback() { }
    async componentWillLoad() {
        debug(`[nylas-threads-search] Component will load`);
    }
    async componentDidLoad() {
        debug(`[nylas-threads-search] Component did load`);
    }
    onQueryChange() {
        this.internalQuery = this.query;
    }
    render() {
        return (h(Host, { key: '6d49d740fb64cff220ddf6b74ca0393a3b6fd83a' }, h("sp-theme", { key: 'fda27523444c0c7350cdfdf6d1a29dd1d63158f5', scale: "medium", color: "dark" }, h("sp-search", { key: 'bb840884220a5008c144f90775ea10291c507b51', quiet: true, onSubmit: this.onSearch, onBlur: this.onSearch, value: this.internalQuery }))));
    }
    static get is() { return "nylas-threads-search"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["nylas-threads-search.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["nylas-threads-search.css"]
        };
    }
    static get properties() {
        return {
            "query": {
                "type": "string",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "The query to search for threads"
                },
                "attribute": "query",
                "reflect": false,
                "defaultValue": "''"
            }
        };
    }
    static get states() {
        return {
            "internalQuery": {}
        };
    }
    static get events() {
        return [{
                "method": "search",
                "name": "search",
                "bubbles": true,
                "cancelable": true,
                "composed": true,
                "docs": {
                    "tags": [],
                    "text": "Event emitted when the query changes"
                },
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                }
            }];
    }
    static get watchers() {
        return [{
                "propName": "query",
                "methodName": "onQueryChange"
            }];
    }
}
__decorate([
    RegisterComponent({
        name: 'nylas-pagination',
        stateToProps: new Map([['mailbox.searchQuery', 'query']]),
        eventToProps: {
            search: async (_event, nylasConnector) => {
                await nylasConnector.threads.getThreads({
                    search_query_native: _event.detail || undefined,
                    page_token: undefined,
                });
            },
        },
        fireRegisterEvent: true,
    }),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", []),
    __metadata("design:returntype", void 0)
], NylasThreadsSearch.prototype, "render", null);
