// @nylas/.web-elements-GF2lXkj6/dist/nylas-web-elements/input-component.entry.js
import { r as registerInstance, a as createEvent, h, e as getElement } from './index-b5cb18eb.js';
import { o as sanitize } from './utils-674a951a.js';
import './_commonjsHelpers-6cbf8316.js';

const inputComponentCss = ":host{display:block;height:auto;--nylas-primary:#2563eb;--nylas-error:#cc4841;--nylas-error-pressed:#992222;--nylas-wraning:#f06c00;--nylas-success:#16a392;--nylas-info:#2b8fc2;--nylas-base-0:#ffffff;--nylas-base-25:#fcfcfd;--nylas-base-50:#f8f9fc;--nylas-base-100:#eaecf5;--nylas-base-200:#d5d9eb;--nylas-base-300:#b3b8d8;--nylas-base-400:#717bbc;--nylas-base-500:#4e5ba6;--nylas-base-600:#3e4784;--nylas-base-700:#263f72;--nylas-base-800:#293056;--nylas-base-900:#101323;--nylas-base-950:#0e101b;--nylas-border-radius:0.25rem;--nylas-border-radius-2x:0.5rem;--nylas-border-radius-3x:0.75rem;--nylas-font-family:\"Inter\", sans-serif;width:100%}label{display:flex;color:var(--nylas-base-800, #293056);font-size:14px;font-style:normal;font-weight:400;line-height:150%;flex-direction:column;gap:4px;font-family:var(--nylas-font-family);font-size:16px;color:var(--nylas-base-800)}label.error{color:var(--nylas-error)}label span.required{color:var(--nylas-error);padding:0 0.25rem}label p{margin:0}label .error{color:var(--nylas-error)}.input_wrapper{display:flex;align-items:center;gap:1rem}input[type=text],input[type=email]{display:flex;height:48px;padding-left:16px;align-items:center;gap:8px;align-self:stretch;border-radius:var(--nylas-border-radius-2x);border:1px solid var(--nylas-base-300);background:var(--nylas-base-0);width:-webkit-fill-available;color:var(--nylas-base-900);font-size:16px;font-style:normal;font-weight:400;line-height:150%;width:-moz-available;width:-webkit-fill-available}input[type=text]:focus,input[type=email]:focus{outline-color:var(--nylas-primary)}input[type=text].error,input[type=email].error{border-radius:var(--nylas-border-radius-2x);font-weight:600;border-color:var(--nylas-error);border-width:2px}input[type=text].error::placeholder,input[type=email].error::placeholder{font-weight:400}input[type=text]::placeholder,input[type=email]::placeholder{color:var(--nylas-base-300)}";

const DefaultPattern = {
    email: /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$/,
    phone_number: /^\+?\d{1,15}$/,
};
const DefaultPlaceholder = {
    email: 'name@example.com',
    phone_number: '1234567890',
};
const InputComponent = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.nylasFormInputChanged = createEvent(this, "nylasFormInputChanged", 7);
        this.nylasFormInputFocused = createEvent(this, "nylasFormInputFocused", 7);
        this.nylasFormInputBlurred = createEvent(this, "nylasFormInputBlurred", 7);
        this.name = 'input';
        this.defaultValue = undefined;
        this.label = '';
        this.type = 'text';
        this.placeholder = DefaultPlaceholder[this.type];
        this.required = false;
        this.readOnly = false;
        this.autoFocus = false;
        this.pattern = undefined;
        this.maxLength = 255;
        this.patternError = '';
        this.requiredError = '';
        this.value = undefined;
        this.error = '';
    }
    handleDefaultValueChange(newValue) {
        this.value = sanitize(newValue);
        if (this.value) {
            this.validatePattern(this.value);
        }
    }
    componentDidLoad() {
        this.value = sanitize(this.defaultValue || '');
        if (this.type !== 'text' && !this.pattern) {
            this.pattern = DefaultPattern[this.type];
        }
        if (this.value) {
            this.validatePattern(this.value);
        }
    }
    handleBookingFormSubmitted(event) {
        this.validatePattern(this.value);
        if (this.error) {
            event.preventDefault();
        }
    }
    async handleFormSubmitted(event) {
        if (this.el?.getAttribute('data-page-styling')) {
            this.validatePattern(this.value);
            if (this.error) {
                event.preventDefault();
            }
        }
    }
    handleInput(e) {
        this.error = '';
        this.value = sanitize(e.target.value);
        this.nylasFormInputChanged.emit({
            value: this.value,
            name: this.name,
            label: this.label,
            error: this.error,
            type: this.type,
        });
    }
    handleBlur() {
        this.nylasFormInputBlurred.emit({
            value: this.value,
            name: this.name,
        });
    }
    handleFocus() {
        this.nylasFormInputFocused.emit({
            value: this.value,
            name: this.name,
        });
    }
    validatePattern(value) {
        this.error = '';
        if (this.required && !value) {
            this.error = this.requiredError || 'This field is required.';
            return;
        }
        if (!this.pattern || (!value && !this.required))
            return;
        if (this.pattern.test(value)) {
            this.error = '';
        }
        else {
            this.error = this.patternError || 'Invalid format.';
        }
    }
    render() {
        return (h("label", { key: '207cb06847b7a045c4917180affc535169b55596', part: "ic__label", class: { error: !!this.error } }, this.label && (h("p", null, h("span", { class: "label" }, this.label), this.required && h("span", { class: "required" }, "*"))), h("div", { key: '8da4d52acd543376116d9851df43eb54b3e3fec1', part: "ic__input_wrapper", class: "input_wrapper" }, h("input", { key: '541168da152c81e01dfa0263f8c1f0b6e73e09f2', type: "text", name: this.name, part: "ic__input", readOnly: this.readOnly, autoFocus: this.autoFocus, value: this.value, maxLength: this.maxLength, placeholder: this.placeholder, class: { error: !!this.error }, onInput: e => this.handleInput(e), onFocus: () => this.handleFocus(), onBlur: () => this.handleBlur() }), h("slot", { key: '9e4ccbf52802dc59bda1d6b738c45026b8b78bd4', name: "additional-input" })), this.error && h("span", { class: "error help-text" }, this.error)));
    }
    get el() { return getElement(this); }
    static get watchers() { return {
        "defaultValue": ["handleDefaultValueChange"]
    }; }
};
InputComponent.style = inputComponentCss;

export { InputComponent as input_component };
