// @nylas/.web-elements-GF2lXkj6/dist/nylas-web-elements/nylas-form-card.entry.js
import { r as registerInstance, h, i as Host, e as getElement } from './index-b5cb18eb.js';
import { R as RegisterComponent } from './register-component-8057b2e5.js';
import { a as debug, H as checkForMissingSlots } from './utils-674a951a.js';
import './_commonjsHelpers-6cbf8316.js';

const nylasFormCardCss = ":host{display:block;--nylas-primary:#2563eb;--nylas-error:#cc4841;--nylas-error-pressed:#992222;--nylas-wraning:#f06c00;--nylas-success:#16a392;--nylas-info:#2b8fc2;--nylas-base-0:#ffffff;--nylas-base-25:#fcfcfd;--nylas-base-50:#f8f9fc;--nylas-base-100:#eaecf5;--nylas-base-200:#d5d9eb;--nylas-base-300:#b3b8d8;--nylas-base-400:#717bbc;--nylas-base-500:#4e5ba6;--nylas-base-600:#3e4784;--nylas-base-700:#263f72;--nylas-base-800:#293056;--nylas-base-900:#101323;--nylas-base-950:#0e101b;--nylas-border-radius:0.25rem;--nylas-border-radius-2x:0.5rem;--nylas-border-radius-3x:0.75rem;--nylas-font-family:\"Inter\", sans-serif}.nylas-form-card{width:100%;border-radius:8px;border:1px solid var(--nylas-base-200)}.nylas-form-card__header{padding:1rem;border-bottom:1px solid var(--nylas-base-200)}";

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (undefined && undefined.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
        return Reflect.metadata(k, v);
};
const NylasFormCard = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        if (hostRef.$hostElement$["s-ei"]) {
            this.internals = hostRef.$hostElement$["s-ei"];
        }
        else {
            this.internals = hostRef.$hostElement$.attachInternals();
            hostRef.$hostElement$["s-ei"] = this.internals;
        }
    }
    connectedCallback() {
        debug('nylas-event-card', 'connectedCallback');
    }
    componentWillLoad() {
        debug('nylas-event-card', 'componentWillLoad');
    }
    componentDidLoad() {
        debug('nylas-event-card', 'componentDidLoad');
    }
    disconnectedCallback() {
        debug('nylas-event-card', 'disconnectedCallback');
        const slots = ['header-title', 'header-subtitle', 'content'];
        checkForMissingSlots(slots, this.host);
    }
    render() {
        return (h(Host, { key: 'ceadfc37c47ece06fb9e188692f12d9b7c1c069d' }, h("div", { key: '2f05c5a7258c800b26dc55d6ffd2d3505dd746c7', class: "nylas-form-card", part: "nfc" }, h("div", { key: '96de3e134413e04a4631d013bb5c3a03c2b07b10', class: "nylas-form-card__header", part: "nfc__header" }, h("slot", { key: 'bd3dc03079117538110f3011775eee108b41da9e', name: "header-title" }), h("slot", { key: '9c140e29b9ff9d8e7ea214dcd0d1eb9c4b752e20', name: "header-subtitle" })), h("div", { key: '1c75e8768e0b59ae4fd2b18f301905ecc8595a82', class: "nylas-form-card__content" }, h("slot", { key: '2b4c082461c47313cb506dced2018af660a20ec8', name: "content" })))));
    }
    static get formAssociated() { return true; }
    get host() { return getElement(this); }
};
__decorate([
    RegisterComponent({
        name: 'nylas-form-card',
        eventToProps: {},
        fireRegisterEvent: true,
    }),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", []),
    __metadata("design:returntype", void 0)
], NylasFormCard.prototype, "render", null);
NylasFormCard.style = nylasFormCardCss;

export { NylasFormCard as nylas_form_card };
