// @nylas/.web-elements-GF2lXkj6/dist/cdn/spam/spam.es.js
const T = {
  allRenderFn: !1,
  cmpDidLoad: !0,
  cmpDidUnload: !1,
  cmpDidUpdate: !0,
  cmpDidRender: !0,
  cmpWillLoad: !0,
  cmpWillUpdate: !0,
  cmpWillRender: !0,
  connectedCallback: !0,
  disconnectedCallback: !0,
  element: !0,
  event: !0,
  hasRenderFn: !0,
  lifecycle: !0,
  hostListener: !0,
  hostListenerTargetWindow: !0,
  hostListenerTargetDocument: !0,
  hostListenerTargetBody: !0,
  hostListenerTargetParent: !1,
  hostListenerTarget: !0,
  member: !0,
  method: !0,
  mode: !0,
  observeAttribute: !0,
  prop: !0,
  propMutable: !0,
  reflect: !0,
  scoped: !0,
  shadowDom: !0,
  slot: !0,
  cssAnnotations: !0,
  state: !0,
  style: !0,
  formAssociated: !1,
  svg: !0,
  updatable: !0,
  vdomAttribute: !0,
  vdomXlink: !0,
  vdomClass: !0,
  vdomFunctional: !0,
  vdomKey: !0,
  vdomListener: !0,
  vdomRef: !0,
  vdomPropOrAttr: !0,
  vdomRender: !0,
  vdomStyle: !0,
  vdomText: !0,
  watchCallback: !0,
  taskQueue: !0,
  hotModuleReplacement: !1,
  isDebug: !1,
  isDev: !1,
  isTesting: !1,
  hydrateServerSide: !1,
  hydrateClientSide: !1,
  lifecycleDOMEvents: !1,
  lazyLoad: !1,
  profile: !1,
  slotRelocation: !0,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  appendChildSlotFix: !1,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  cloneNodeFix: !1,
  hydratedAttribute: !1,
  hydratedClass: !0,
  scriptDataOpts: !1,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  scopedSlotTextContentFix: !1,
  // TODO(STENCIL-854): Remove code related to legacy shadowDomShim field
  shadowDomShim: !1,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  slotChildNodesFix: !1,
  invisiblePrehydration: !0,
  propBoolean: !0,
  propNumber: !0,
  propString: !0,
  constructableCSS: !0,
  cmpShouldUpdate: !0,
  devTools: !1,
  shadowDelegatesFocus: !0,
  initializeNextTick: !1,
  asyncLoading: !1,
  asyncQueue: !1,
  transformTagName: !1,
  attachStyles: !0,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  experimentalSlotFixes: !1
};
let L, H, R, ot = !1, U = !1, W = !1, v = !1, I = null, M = !1;
const x = (t, e = "") => () => {
}, vt = "slot-fb{display:contents}slot-fb[hidden]{display:none}", Q = "http://www.w3.org/1999/xlink", X = {}, St = "http://www.w3.org/2000/svg", mt = "http://www.w3.org/1999/xhtml", kt = (t) => t != null, z = (t) => (t = typeof t, t === "object" || t === "function");
function Tt(t) {
  var e, n, s;
  return (s = (n = (e = t.head) === null || e === void 0 ? void 0 : e.querySelector('meta[name="csp-nonce"]')) === null || n === void 0 ? void 0 : n.getAttribute("content")) !== null && s !== void 0 ? s : void 0;
}
const A = (t, e, ...n) => {
  let s = null, o = null, r = null, l = !1, c = !1;
  const i = [], u = (a) => {
    for (let d = 0; d < a.length; d++)
      s = a[d], Array.isArray(s) ? u(s) : s != null && typeof s != "boolean" && ((l = typeof t != "function" && !z(s)) && (s = String(s)), l && c ? i[i.length - 1].$text$ += s : i.push(l ? P(null, s) : s), c = l);
  };
  if (u(n), e) {
    e.key && (o = e.key), e.name && (r = e.name);
    {
      const a = e.className || e.class;
      a && (e.class = typeof a != "object" ? a : Object.keys(a).filter((d) => a[d]).join(" "));
    }
  }
  if (typeof t == "function")
    return t(e === null ? {} : e, i, Lt);
  const f = P(t, null);
  return f.$attrs$ = e, i.length > 0 && (f.$children$ = i), f.$key$ = o, f.$name$ = r, f;
}, P = (t, e) => {
  const n = {
    $flags$: 0,
    $tag$: t,
    $text$: e,
    $elm$: null,
    $children$: null
  };
  return n.$attrs$ = null, n.$key$ = null, n.$name$ = null, n;
}, bt = {}, xt = (t) => t && t.$tag$ === bt, Lt = {
  forEach: (t, e) => t.map(G).forEach(e),
  map: (t, e) => t.map(G).map(e).map(Et)
}, G = (t) => ({
  vattrs: t.$attrs$,
  vchildren: t.$children$,
  vkey: t.$key$,
  vname: t.$name$,
  vtag: t.$tag$,
  vtext: t.$text$
}), Et = (t) => {
  if (typeof t.vtag == "function") {
    const n = Object.assign({}, t.vattrs);
    return t.vkey && (n.key = t.vkey), t.vname && (n.name = t.vname), A(t.vtag, n, ...t.vchildren || []);
  }
  const e = P(t.vtag, t.vtext);
  return e.$attrs$ = t.vattrs, e.$children$ = t.vchildren, e.$key$ = t.vkey, e.$name$ = t.vname, e;
}, _t = (t) => te.map((e) => e(t)).find((e) => !!e), At = (t, e) => t != null && !z(t) ? e & 4 ? t === "false" ? !1 : t === "" || !!t : e & 2 ? parseFloat(t) : e & 1 ? String(t) : t : t, K = /* @__PURE__ */ new WeakMap(), Rt = (t, e, n) => {
  let s = B.get(t);
  oe && n ? (s = s || new CSSStyleSheet(), typeof s == "string" ? s = e : s.replaceSync(e)) : s = e, B.set(t, s);
}, Ct = (t, e, n) => {
  var s;
  const o = lt(e, n), r = B.get(o);
  if (t = t.nodeType === 11 ? t : S, r)
    if (typeof r == "string") {
      t = t.head || t;
      let l = K.get(t), c;
      if (l || K.set(t, l = /* @__PURE__ */ new Set()), !l.has(o)) {
        {
          c = S.createElement("style"), c.innerHTML = r;
          const i = (s = p.$nonce$) !== null && s !== void 0 ? s : Tt(S);
          i != null && c.setAttribute("nonce", i), t.insertBefore(c, t.querySelector("link"));
        }
        e.$flags$ & 4 && (c.innerHTML += vt), l && l.add(o);
      }
    } else t.adoptedStyleSheets.includes(r) || (t.adoptedStyleSheets = [...t.adoptedStyleSheets, r]);
  return o;
}, Ot = (t) => {
  const e = t.$cmpMeta$, n = t.$hostElement$, s = e.$flags$, o = x("attachStyles", e.$tagName$), r = Ct(n.shadowRoot ? n.shadowRoot : n.getRootNode(), e, t.$modeName$);
  s & 10 && (n["s-sc"] = r, n.classList.add(r + "-h"), s & 2 && n.classList.add(r + "-s")), o();
}, lt = (t, e) => "sc-" + (e && t.$flags$ & 32 ? t.$tagName$ + "-" + e : t.$tagName$), J = (t, e, n, s, o, r) => {
  if (n !== s) {
    let l = et(t, e), c = e.toLowerCase();
    if (e === "class") {
      const i = t.classList, u = Y(n), f = Y(s);
      i.remove(...u.filter((a) => a && !f.includes(a))), i.add(...f.filter((a) => a && !u.includes(a)));
    } else if (e === "style") {
      for (const i in n)
        (!s || s[i] == null) && (i.includes("-") ? t.style.removeProperty(i) : t.style[i] = "");
      for (const i in s)
        (!n || s[i] !== n[i]) && (i.includes("-") ? t.style.setProperty(i, s[i]) : t.style[i] = s[i]);
    } else if (e !== "key")
      if (e === "ref")
        s && s(t);
      else if (!t.__lookupSetter__(e) && e[0] === "o" && e[1] === "n") {
        if (e[2] === "-" ? e = e.slice(3) : et(D, c) ? e = c.slice(2) : e = c[2] + e.slice(3), n || s) {
          const i = e.endsWith(it);
          e = e.replace(Ft, ""), n && p.rel(t, e, n, i), s && p.ael(t, e, s, i);
        }
      } else {
        const i = z(s);
        if ((l || i && s !== null) && !o)
          try {
            if (t.tagName.includes("-"))
              t[e] = s;
            else {
              const f = s ?? "";
              e === "list" ? l = !1 : (n == null || t[e] != f) && (t[e] = f);
            }
          } catch {
          }
        let u = !1;
        c !== (c = c.replace(/^xlink\:?/, "")) && (e = c, u = !0), s == null || s === !1 ? (s !== !1 || t.getAttribute(e) === "") && (u ? t.removeAttributeNS(Q, e) : t.removeAttribute(e)) : (!l || r & 4 || o) && !i && (s = s === !0 ? "" : s, u ? t.setAttributeNS(Q, e, s) : t.setAttribute(e, s));
      }
  }
}, wt = /\s/, Y = (t) => t ? t.split(wt) : [], it = "Capture", Ft = new RegExp(it + "$"), rt = (t, e, n, s) => {
  const o = e.$elm$.nodeType === 11 && e.$elm$.host ? e.$elm$.host : e.$elm$, r = t && t.$attrs$ || X, l = e.$attrs$ || X;
  for (s in r)
    s in l || J(o, s, r[s], void 0, n, e.$flags$);
  for (s in l)
    J(o, s, r[s], l[s], n, e.$flags$);
}, j = (t, e, n, s) => {
  const o = e.$children$[n];
  let r = 0, l, c, i;
  if (ot || (W = !0, o.$tag$ === "slot" && (L && s.classList.add(L + "-s"), o.$flags$ |= o.$children$ ? (
    // slot element has fallback content
    2
  ) : (
    // slot element does not have fallback content
    1
  ))), o.$text$ !== null)
    l = o.$elm$ = S.createTextNode(o.$text$);
  else if (o.$flags$ & 1)
    l = o.$elm$ = S.createTextNode("");
  else {
    if (v || (v = o.$tag$ === "svg"), l = o.$elm$ = S.createElementNS(v ? St : mt, o.$flags$ & 2 ? "slot-fb" : o.$tag$), v && o.$tag$ === "foreignObject" && (v = !1), rt(null, o, v), kt(L) && l["s-si"] !== L && l.classList.add(l["s-si"] = L), o.$children$)
      for (r = 0; r < o.$children$.length; ++r)
        c = j(t, o, r, l), c && l.appendChild(c);
    o.$tag$ === "svg" ? v = !1 : l.tagName === "foreignObject" && (v = !0);
  }
  return l["s-hn"] = R, o.$flags$ & 3 && (l["s-sr"] = !0, l["s-cr"] = H, l["s-sn"] = o.$name$ || "", i = t && t.$children$ && t.$children$[n], i && i.$tag$ === o.$tag$ && t.$elm$ && C(t.$elm$, !1)), l;
}, C = (t, e) => {
  p.$flags$ |= 1;
  const n = t.childNodes;
  for (let s = n.length - 1; s >= 0; s--) {
    const o = n[s];
    o["s-hn"] !== R && o["s-ol"] && ($t(o).insertBefore(o, q(o)), o["s-ol"].remove(), o["s-ol"] = void 0, o["s-sh"] = void 0, W = !0), e && C(o, e);
  }
  p.$flags$ &= -2;
}, ct = (t, e, n, s, o, r) => {
  let l = t["s-cr"] && t["s-cr"].parentNode || t, c;
  for (l.shadowRoot && l.tagName === R && (l = l.shadowRoot); o <= r; ++o)
    s[o] && (c = j(null, n, o, t), c && (s[o].$elm$ = c, l.insertBefore(c, q(e))));
}, at = (t, e, n) => {
  for (let s = e; s <= n; ++s) {
    const o = t[s];
    if (o) {
      const r = o.$elm$;
      ut(o), r && (U = !0, r["s-ol"] ? r["s-ol"].remove() : C(r, !0), r.remove());
    }
  }
}, Ut = (t, e, n, s, o = !1) => {
  let r = 0, l = 0, c = 0, i = 0, u = e.length - 1, f = e[0], a = e[u], d = s.length - 1, $ = s[0], h = s[d], y, g;
  for (; r <= u && l <= d; )
    if (f == null)
      f = e[++r];
    else if (a == null)
      a = e[--u];
    else if ($ == null)
      $ = s[++l];
    else if (h == null)
      h = s[--d];
    else if (F(f, $, o))
      E(f, $, o), f = e[++r], $ = s[++l];
    else if (F(a, h, o))
      E(a, h, o), a = e[--u], h = s[--d];
    else if (F(f, h, o))
      (f.$tag$ === "slot" || h.$tag$ === "slot") && C(f.$elm$.parentNode, !1), E(f, h, o), t.insertBefore(f.$elm$, a.$elm$.nextSibling), f = e[++r], h = s[--d];
    else if (F(a, $, o))
      (f.$tag$ === "slot" || h.$tag$ === "slot") && C(a.$elm$.parentNode, !1), E(a, $, o), t.insertBefore(a.$elm$, f.$elm$), a = e[--u], $ = s[++l];
    else {
      for (c = -1, i = r; i <= u; ++i)
        if (e[i] && e[i].$key$ !== null && e[i].$key$ === $.$key$) {
          c = i;
          break;
        }
      c >= 0 ? (g = e[c], g.$tag$ !== $.$tag$ ? y = j(e && e[l], n, c, t) : (E(g, $, o), e[c] = void 0, y = g.$elm$), $ = s[++l]) : (y = j(e && e[l], n, l, t), $ = s[++l]), y && $t(f.$elm$).insertBefore(y, q(f.$elm$));
    }
  r > u ? ct(t, s[d + 1] == null ? null : s[d + 1].$elm$, n, s, l, d) : l > d && at(e, r, u);
}, F = (t, e, n = !1) => t.$tag$ === e.$tag$ ? t.$tag$ === "slot" ? t.$name$ === e.$name$ : n ? !0 : t.$key$ === e.$key$ : !1, q = (t) => t && t["s-ol"] || t, $t = (t) => (t["s-ol"] ? t["s-ol"] : t).parentNode, E = (t, e, n = !1) => {
  const s = e.$elm$ = t.$elm$, o = t.$children$, r = e.$children$, l = e.$tag$, c = e.$text$;
  let i;
  c === null ? (v = l === "svg" ? !0 : l === "foreignObject" ? !1 : v, l === "slot" || rt(t, e, v), o !== null && r !== null ? Ut(s, o, e, r, n) : r !== null ? (t.$text$ !== null && (s.textContent = ""), ct(s, null, e, r, 0, r.length - 1)) : o !== null && at(o, 0, o.length - 1), v && l === "svg" && (v = !1)) : (i = s["s-cr"]) ? i.parentNode.textContent = c : t.$text$ !== c && (s.data = c);
}, ft = (t) => {
  const e = t.childNodes;
  for (const n of e)
    if (n.nodeType === 1) {
      if (n["s-sr"]) {
        const s = n["s-sn"];
        n.hidden = !1;
        for (const o of e)
          if (o !== n) {
            if (o["s-hn"] !== n["s-hn"] || s !== "") {
              if (o.nodeType === 1 && (s === o.getAttribute("slot") || s === o["s-sn"])) {
                n.hidden = !0;
                break;
              }
            } else if (o.nodeType === 1 || o.nodeType === 3 && o.textContent.trim() !== "") {
              n.hidden = !0;
              break;
            }
          }
      }
      ft(n);
    }
}, m = [], dt = (t) => {
  let e, n, s;
  for (const o of t.childNodes) {
    if (o["s-sr"] && (e = o["s-cr"]) && e.parentNode) {
      n = e.parentNode.childNodes;
      const r = o["s-sn"];
      for (s = n.length - 1; s >= 0; s--)
        if (e = n[s], !e["s-cn"] && !e["s-nr"] && e["s-hn"] !== o["s-hn"] && !T.experimentalSlotFixes)
          if (Z(e, r)) {
            let l = m.find((c) => c.$nodeToRelocate$ === e);
            U = !0, e["s-sn"] = e["s-sn"] || r, l ? (l.$nodeToRelocate$["s-sh"] = o["s-hn"], l.$slotRefNode$ = o) : (e["s-sh"] = o["s-hn"], m.push({
              $slotRefNode$: o,
              $nodeToRelocate$: e
            })), e["s-sr"] && m.map((c) => {
              Z(c.$nodeToRelocate$, e["s-sn"]) && (l = m.find((i) => i.$nodeToRelocate$ === e), l && !c.$slotRefNode$ && (c.$slotRefNode$ = l.$slotRefNode$));
            });
          } else m.some((l) => l.$nodeToRelocate$ === e) || m.push({
            $nodeToRelocate$: e
          });
    }
    o.nodeType === 1 && dt(o);
  }
}, Z = (t, e) => t.nodeType === 1 ? t.getAttribute("slot") === null && e === "" || t.getAttribute("slot") === e : t["s-sn"] === e ? !0 : e === "", ut = (t) => {
  t.$attrs$ && t.$attrs$.ref && t.$attrs$.ref(null), t.$children$ && t.$children$.map(ut);
}, Pt = (t, e, n = !1) => {
  var s, o, r, l, c;
  const i = t.$hostElement$, u = t.$cmpMeta$, f = t.$vnode$ || P(null, null), a = xt(e) ? e : A(null, null, e);
  if (R = i.tagName, u.$attrsToReflect$ && (a.$attrs$ = a.$attrs$ || {}, u.$attrsToReflect$.map(([d, $]) => a.$attrs$[$] = i[d])), n && a.$attrs$)
    for (const d of Object.keys(a.$attrs$))
      i.hasAttribute(d) && !["key", "ref", "style", "class"].includes(d) && (a.$attrs$[d] = i[d]);
  a.$tag$ = null, a.$flags$ |= 4, t.$vnode$ = a, a.$elm$ = f.$elm$ = i.shadowRoot || i, L = i["s-sc"], H = i["s-cr"], ot = (u.$flags$ & 1) !== 0, U = !1, E(f, a, n);
  {
    if (p.$flags$ |= 1, W) {
      dt(a.$elm$);
      for (const d of m) {
        const $ = d.$nodeToRelocate$;
        if (!$["s-ol"]) {
          const h = S.createTextNode("");
          h["s-nr"] = $, $.parentNode.insertBefore($["s-ol"] = h, $);
        }
      }
      for (const d of m) {
        const $ = d.$nodeToRelocate$, h = d.$slotRefNode$;
        if (h) {
          const y = h.parentNode;
          let g = h.nextSibling;
          {
            let b = (s = $["s-ol"]) === null || s === void 0 ? void 0 : s.previousSibling;
            for (; b; ) {
              let k = (o = b["s-nr"]) !== null && o !== void 0 ? o : null;
              if (k && k["s-sn"] === $["s-sn"] && y === k.parentNode && (k = k.nextSibling, !k || !k["s-nr"])) {
                g = k;
                break;
              }
              b = b.previousSibling;
            }
          }
          (!g && y !== $.parentNode || $.nextSibling !== g) && $ !== g && (!$["s-hn"] && $["s-ol"] && ($["s-hn"] = $["s-ol"].parentNode.nodeName), y.insertBefore($, g), $.nodeType === 1 && ($.hidden = (r = $["s-ih"]) !== null && r !== void 0 ? r : !1));
        } else
          $.nodeType === 1 && (n && ($["s-ih"] = (l = $.hidden) !== null && l !== void 0 ? l : !1), $.hidden = !0);
      }
    }
    U && ft(a.$elm$), p.$flags$ &= -2, m.length = 0;
  }
  if (T.experimentalScopedSlotChanges && u.$flags$ & 2)
    for (const d of a.$elm$.childNodes)
      d["s-hn"] !== R && !d["s-sh"] && (n && d["s-ih"] == null && (d["s-ih"] = (c = d.hidden) !== null && c !== void 0 ? c : !1), d.hidden = !0);
  H = void 0;
}, jt = (t, e) => {
}, ht = (t, e) => (t.$flags$ |= 16, jt(t, t.$ancestorComponent$), re(() => Bt(t, e))), Bt = (t, e) => {
  const n = t.$hostElement$, s = x("scheduleUpdate", t.$cmpMeta$.$tagName$), o = n;
  let r;
  return e ? r = _(o, "componentWillLoad") : r = _(o, "componentWillUpdate"), r = V(r, () => _(o, "componentWillRender")), s(), V(r, () => Ht(t, o, e));
}, V = (t, e) => Dt(t) ? t.then(e) : e(), Dt = (t) => t instanceof Promise || t && t.then && typeof t.then == "function", Ht = async (t, e, n) => {
  const s = t.$hostElement$, o = x("update", t.$cmpMeta$.$tagName$);
  s["s-rc"], n && Ot(t);
  const r = x("render", t.$cmpMeta$.$tagName$);
  Mt(t, e, s, n), r(), o(), Nt(t);
}, Mt = (t, e, n, s) => {
  try {
    I = e, e = e.render && e.render(), t.$flags$ &= -17, t.$flags$ |= 2, (T.hasRenderFn || T.reflect) && (T.vdomRender || T.reflect) && (T.hydrateServerSide || Pt(t, e, s));
  } catch (i) {
    w(i, t.$hostElement$);
  }
  return I = null, null;
}, Nt = (t) => {
  const e = t.$cmpMeta$.$tagName$, n = t.$hostElement$, s = x("postUpdate", e), o = n;
  t.$ancestorComponent$, _(o, "componentDidRender"), t.$flags$ & 64 ? (_(o, "componentDidUpdate"), s()) : (t.$flags$ |= 64, _(o, "componentDidLoad"), s());
}, _ = (t, e, n) => {
  if (t && t[e])
    try {
      return t[e](n);
    } catch (s) {
      w(s);
    }
}, Wt = (t, e) => O(t).$instanceValues$.get(e), zt = (t, e, n, s) => {
  const o = O(t), r = t, l = o.$instanceValues$.get(e), c = o.$flags$, i = r;
  n = At(n, s.$members$[e][0]);
  const u = Number.isNaN(l) && Number.isNaN(n);
  if (n !== l && !u) {
    o.$instanceValues$.set(e, n);
    {
      if (s.$watchers$ && c & 128) {
        const a = s.$watchers$[e];
        a && a.map((d) => {
          try {
            i[d](n, l, e);
          } catch ($) {
            w($, r);
          }
        });
      }
      if ((c & 18) === 2) {
        if (i.componentShouldUpdate && i.componentShouldUpdate(n, l, e) === !1)
          return;
        ht(o, !1);
      }
    }
  }
}, qt = (t, e, n) => {
  var s;
  const o = t.prototype;
  if (e.$members$) {
    t.watchers && (e.$watchers$ = t.watchers);
    const r = Object.entries(e.$members$);
    r.map(([l, [c]]) => {
      (c & 31 || c & 32) && Object.defineProperty(o, l, {
        get() {
          return Wt(this, l);
        },
        set(i) {
          zt(this, l, i, e);
        },
        configurable: !0,
        enumerable: !0
      });
    });
    {
      const l = /* @__PURE__ */ new Map();
      o.attributeChangedCallback = function(c, i, u) {
        p.jmp(() => {
          var f;
          const a = l.get(c);
          if (this.hasOwnProperty(a))
            u = this[a], delete this[a];
          else {
            if (o.hasOwnProperty(a) && typeof this[a] == "number" && this[a] == u)
              return;
            if (a == null) {
              const d = O(this), $ = d == null ? void 0 : d.$flags$;
              if ($ && !($ & 8) && $ & 128 && u !== i) {
                const y = this, g = (f = e.$watchers$) === null || f === void 0 ? void 0 : f[c];
                g == null || g.forEach((b) => {
                  y[b] != null && y[b].call(y, u, i, c);
                });
              }
              return;
            }
          }
          this[a] = u === null && typeof this[a] == "boolean" ? !1 : u;
        });
      }, t.observedAttributes = Array.from(/* @__PURE__ */ new Set([
        ...Object.keys((s = e.$watchers$) !== null && s !== void 0 ? s : {}),
        ...r.filter(
          ([c, i]) => i[0] & 15
          /* MEMBER_FLAGS.HasAttribute */
        ).map(([c, i]) => {
          var u;
          const f = i[1] || c;
          return l.set(f, c), i[0] & 512 && ((u = e.$attrsToReflect$) === null || u === void 0 || u.push([c, f])), f;
        })
      ]));
    }
  }
  return t;
}, It = async (t, e, n, s) => {
  let o;
  if (!(e.$flags$ & 32) && (e.$flags$ |= 32, o = t.constructor, customElements.whenDefined(n.$tagName$).then(() => e.$flags$ |= 128), o.style)) {
    let l = o.style;
    typeof l != "string" && (l = l[e.$modeName$ = _t(t)]);
    const c = lt(n, e.$modeName$);
    if (!B.has(c)) {
      const i = x("registerStyles", n.$tagName$);
      Rt(c, l, !!(n.$flags$ & 1)), i();
    }
  }
  e.$ancestorComponent$, ht(e, !0);
}, tt = (t) => {
}, Qt = (t) => {
  if (!(p.$flags$ & 1)) {
    const e = O(t), n = e.$cmpMeta$, s = x("connectedCallback", n.$tagName$);
    e.$flags$ & 1 ? (pt(t, e, n.$listeners$), e != null && e.$lazyInstance$ ? tt(e.$lazyInstance$) : e != null && e.$onReadyPromise$ && e.$onReadyPromise$.then(() => tt(e.$lazyInstance$))) : (e.$flags$ |= 1, // TODO(STENCIL-854): Remove code related to legacy shadowDomShim field
    n.$flags$ & 12 && Xt(t), n.$members$ && Object.entries(n.$members$).map(([o, [r]]) => {
      if (r & 31 && t.hasOwnProperty(o)) {
        const l = t[o];
        delete t[o], t[o] = l;
      }
    }), It(t, e, n)), s();
  }
}, Xt = (t) => {
  const e = t["s-cr"] = S.createComment("");
  e["s-cn"] = !0, t.insertBefore(e, t.firstChild);
}, Gt = async (t) => {
  if (!(p.$flags$ & 1)) {
    const e = O(t);
    e.$rmListeners$ && (e.$rmListeners$.map((n) => n()), e.$rmListeners$ = void 0);
  }
}, Kt = (t, e) => {
  const n = {
    $flags$: e[0],
    $tagName$: e[1]
  };
  n.$members$ = e[2], n.$listeners$ = e[3], n.$watchers$ = t.$watchers$, n.$attrsToReflect$ = [];
  const s = t.prototype.connectedCallback, o = t.prototype.disconnectedCallback;
  return Object.assign(t.prototype, {
    __registerHost() {
      Vt(this, n);
    },
    connectedCallback() {
      Qt(this), s && s.call(this);
    },
    disconnectedCallback() {
      Gt(this), o && o.call(this);
    },
    __attachShadow() {
      this.attachShadow({
        mode: "open",
        delegatesFocus: !!(n.$flags$ & 16)
      });
    }
  }), t.is = n.$tagName$, qt(t, n);
}, pt = (t, e, n, s) => {
  n && n.map(([o, r, l]) => {
    const c = Yt(t, o), i = Jt(e, l), u = Zt(o);
    p.ael(c, r, i, u), (e.$rmListeners$ = e.$rmListeners$ || []).push(() => p.rel(c, r, i, u));
  });
}, Jt = (t, e) => (n) => {
  try {
    T.lazyLoad || t.$hostElement$[e](n);
  } catch (s) {
    w(s);
  }
}, Yt = (t, e) => e & 4 ? S : e & 8 ? D : e & 16 ? S.body : t, Zt = (t) => se ? {
  passive: (t & 1) !== 0,
  capture: (t & 2) !== 0
} : (t & 2) !== 0, gt = /* @__PURE__ */ new WeakMap(), O = (t) => gt.get(t), Vt = (t, e) => {
  const n = {
    $flags$: 0,
    $hostElement$: t,
    $cmpMeta$: e,
    $instanceValues$: /* @__PURE__ */ new Map()
  };
  return pt(t, n, e.$listeners$), gt.set(t, n);
}, et = (t, e) => e in t, w = (t, e) => (0, console.error)(t, e), B = /* @__PURE__ */ new Map(), te = [], D = typeof window < "u" ? window : {}, S = D.document || { head: {} }, ee = D.HTMLElement || class {
}, p = {
  $flags$: 0,
  $resourcesUrl$: "",
  jmp: (t) => t(),
  raf: (t) => requestAnimationFrame(t),
  ael: (t, e, n, s) => t.addEventListener(e, n, s),
  rel: (t, e, n, s) => t.removeEventListener(e, n, s),
  ce: (t, e) => new CustomEvent(t, e)
}, se = /* @__PURE__ */ (() => {
  let t = !1;
  try {
    S.addEventListener("e", null, Object.defineProperty({}, "passive", {
      get() {
        t = !0;
      }
    }));
  } catch {
  }
  return t;
})(), ne = (t) => Promise.resolve(t), oe = /* @__PURE__ */ (() => {
  try {
    return new CSSStyleSheet(), typeof new CSSStyleSheet().replaceSync == "function";
  } catch {
  }
  return !1;
})(), st = [], yt = [], le = (t, e) => (n) => {
  t.push(n), M || (M = !0, e && p.$flags$ & 4 ? ie(N) : p.raf(N));
}, nt = (t) => {
  for (let e = 0; e < t.length; e++)
    try {
      t[e](performance.now());
    } catch (n) {
      w(n);
    }
  t.length = 0;
}, N = () => {
  nt(st), nt(yt), (M = st.length > 0) && p.raf(N);
}, ie = (t) => ne().then(t), re = /* @__PURE__ */ le(yt, !0), ce = ".sc-spam-icon-h{display:flex}", ae = ce, $e = Kt(class extends ee {
  constructor() {
    super(), this.__registerHost(), this.width = "15", this.height = "15";
  }
  render() {
    return A("svg", { key: "9afff7db1e837c023fb53790efccfcef7a65f93d", width: this.width, height: this.height, viewBox: "0 0 24 24", fill: "auto", xmlns: "http://www.w3.org/2000/svg" }, A("path", { key: "c5a7463010ca4c71cfe05a6c440233a6a015cb91", d: "M20 3H4C2.89543 3 2 3.89543 2 5V19C2 20.1046 2.89543 21 4 21H20C21.1046 21 22 20.1046 22 19V5C22 3.89543 21.1046 3 20 3Z", stroke: "auto", "stroke-width": "1.5" }), A("path", { key: "6af1f5931964cb49518df288e37377986df3844e", d: "M2 5L12 12.5L22 5", stroke: "auto", "stroke-width": "1.5" }));
  }
  static get style() {
    return ae;
  }
}, [2, "spam-icon", {
  width: [1],
  height: [1]
}]);
function de() {
  if (typeof customElements > "u")
    return;
  ["spam-icon"].forEach((e) => {
    switch (e) {
      case "spam-icon":
        customElements.get(e) || customElements.define(e, $e);
        break;
    }
  });
}
export {
  $e as S,
  de as d
};
