// @nylas/.web-elements-GF2lXkj6/dist/cdn/person-clipboard/person-clipboard.es.js
const C = {
  allRenderFn: !1,
  cmpDidLoad: !0,
  cmpDidUnload: !1,
  cmpDidUpdate: !0,
  cmpDidRender: !0,
  cmpWillLoad: !0,
  cmpWillUpdate: !0,
  cmpWillRender: !0,
  connectedCallback: !0,
  disconnectedCallback: !0,
  element: !0,
  event: !0,
  hasRenderFn: !0,
  lifecycle: !0,
  hostListener: !0,
  hostListenerTargetWindow: !0,
  hostListenerTargetDocument: !0,
  hostListenerTargetBody: !0,
  hostListenerTargetParent: !1,
  hostListenerTarget: !0,
  member: !0,
  method: !0,
  mode: !0,
  observeAttribute: !0,
  prop: !0,
  propMutable: !0,
  reflect: !0,
  scoped: !0,
  shadowDom: !0,
  slot: !0,
  cssAnnotations: !0,
  state: !0,
  style: !0,
  formAssociated: !1,
  svg: !0,
  updatable: !0,
  vdomAttribute: !0,
  vdomXlink: !0,
  vdomClass: !0,
  vdomFunctional: !0,
  vdomKey: !0,
  vdomListener: !0,
  vdomRef: !0,
  vdomPropOrAttr: !0,
  vdomRender: !0,
  vdomStyle: !0,
  vdomText: !0,
  watchCallback: !0,
  taskQueue: !0,
  hotModuleReplacement: !1,
  isDebug: !1,
  isDev: !1,
  isTesting: !1,
  hydrateServerSide: !1,
  hydrateClientSide: !1,
  lifecycleDOMEvents: !1,
  lazyLoad: !1,
  profile: !1,
  slotRelocation: !0,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  appendChildSlotFix: !1,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  cloneNodeFix: !1,
  hydratedAttribute: !1,
  hydratedClass: !0,
  scriptDataOpts: !1,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  scopedSlotTextContentFix: !1,
  // TODO(STENCIL-854): Remove code related to legacy shadowDomShim field
  shadowDomShim: !1,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  slotChildNodesFix: !1,
  invisiblePrehydration: !0,
  propBoolean: !0,
  propNumber: !0,
  propString: !0,
  constructableCSS: !0,
  cmpShouldUpdate: !0,
  devTools: !1,
  shadowDelegatesFocus: !0,
  initializeNextTick: !1,
  asyncLoading: !1,
  asyncQueue: !1,
  transformTagName: !1,
  attachStyles: !0,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  experimentalSlotFixes: !1
};
let x, B, _, le = !1, P = !1, W = !1, v = !1, Z = null, D = !1;
const T = (e, t = "") => () => {
}, ve = "slot-fb{display:contents}slot-fb[hidden]{display:none}", I = "http://www.w3.org/1999/xlink", Q = {}, Se = "http://www.w3.org/2000/svg", me = "http://www.w3.org/1999/xhtml", ke = (e) => e != null, z = (e) => (e = typeof e, e === "object" || e === "function");
function Ce(e) {
  var t, n, s;
  return (s = (n = (t = e.head) === null || t === void 0 ? void 0 : t.querySelector('meta[name="csp-nonce"]')) === null || n === void 0 ? void 0 : n.getAttribute("content")) !== null && s !== void 0 ? s : void 0;
}
const M = (e, t, ...n) => {
  let s = null, l = null, r = null, o = !1, c = !1;
  const i = [], u = ($) => {
    for (let d = 0; d < $.length; d++)
      s = $[d], Array.isArray(s) ? u(s) : s != null && typeof s != "boolean" && ((o = typeof e != "function" && !z(s)) && (s = String(s)), o && c ? i[i.length - 1].$text$ += s : i.push(o ? F(null, s) : s), c = o);
  };
  if (u(n), t) {
    t.key && (l = t.key), t.name && (r = t.name);
    {
      const $ = t.className || t.class;
      $ && (t.class = typeof $ != "object" ? $ : Object.keys($).filter((d) => $[d]).join(" "));
    }
  }
  if (typeof e == "function")
    return e(t === null ? {} : t, i, xe);
  const f = F(e, null);
  return f.$attrs$ = t, i.length > 0 && (f.$children$ = i), f.$key$ = l, f.$name$ = r, f;
}, F = (e, t) => {
  const n = {
    $flags$: 0,
    $tag$: e,
    $text$: t,
    $elm$: null,
    $children$: null
  };
  return n.$attrs$ = null, n.$key$ = null, n.$name$ = null, n;
}, be = {}, Te = (e) => e && e.$tag$ === be, xe = {
  forEach: (e, t) => e.map(X).forEach(t),
  map: (e, t) => e.map(X).map(t).map(Le)
}, X = (e) => ({
  vattrs: e.$attrs$,
  vchildren: e.$children$,
  vkey: e.$key$,
  vname: e.$name$,
  vtag: e.$tag$,
  vtext: e.$text$
}), Le = (e) => {
  if (typeof e.vtag == "function") {
    const n = Object.assign({}, e.vattrs);
    return e.vkey && (n.key = e.vkey), e.vname && (n.name = e.vname), M(e.vtag, n, ...e.vchildren || []);
  }
  const t = F(e.vtag, e.vtext);
  return t.$attrs$ = e.vattrs, t.$children$ = e.vchildren, t.$key$ = e.vkey, t.$name$ = e.vname, t;
}, Ee = (e) => et.map((t) => t(e)).find((t) => !!t), _e = (e, t) => e != null && !z(e) ? t & 4 ? e === "false" ? !1 : e === "" || !!e : t & 2 ? parseFloat(e) : t & 1 ? String(e) : e : e, G = /* @__PURE__ */ new WeakMap(), Ae = (e, t, n) => {
  let s = j.get(e);
  lt && n ? (s = s || new CSSStyleSheet(), typeof s == "string" ? s = t : s.replaceSync(t)) : s = t, j.set(e, s);
}, Re = (e, t, n) => {
  var s;
  const l = oe(t, n), r = j.get(l);
  if (e = e.nodeType === 11 ? e : S, r)
    if (typeof r == "string") {
      e = e.head || e;
      let o = G.get(e), c;
      if (o || G.set(e, o = /* @__PURE__ */ new Set()), !o.has(l)) {
        {
          c = S.createElement("style"), c.innerHTML = r;
          const i = (s = h.$nonce$) !== null && s !== void 0 ? s : Ce(S);
          i != null && c.setAttribute("nonce", i), e.insertBefore(c, e.querySelector("link"));
        }
        t.$flags$ & 4 && (c.innerHTML += ve), o && o.add(l);
      }
    } else e.adoptedStyleSheets.includes(r) || (e.adoptedStyleSheets = [...e.adoptedStyleSheets, r]);
  return l;
}, He = (e) => {
  const t = e.$cmpMeta$, n = e.$hostElement$, s = t.$flags$, l = T("attachStyles", t.$tagName$), r = Re(n.shadowRoot ? n.shadowRoot : n.getRootNode(), t, e.$modeName$);
  s & 10 && (n["s-sc"] = r, n.classList.add(r + "-h"), s & 2 && n.classList.add(r + "-s")), l();
}, oe = (e, t) => "sc-" + (t && e.$flags$ & 32 ? e.$tagName$ + "-" + t : e.$tagName$), K = (e, t, n, s, l, r) => {
  if (n !== s) {
    let o = te(e, t), c = t.toLowerCase();
    if (t === "class") {
      const i = e.classList, u = J(n), f = J(s);
      i.remove(...u.filter(($) => $ && !f.includes($))), i.add(...f.filter(($) => $ && !u.includes($)));
    } else if (t === "style") {
      for (const i in n)
        (!s || s[i] == null) && (i.includes("-") ? e.style.removeProperty(i) : e.style[i] = "");
      for (const i in s)
        (!n || s[i] !== n[i]) && (i.includes("-") ? e.style.setProperty(i, s[i]) : e.style[i] = s[i]);
    } else if (t !== "key")
      if (t === "ref")
        s && s(e);
      else if (!e.__lookupSetter__(t) && t[0] === "o" && t[1] === "n") {
        if (t[2] === "-" ? t = t.slice(3) : te(w, c) ? t = c.slice(2) : t = c[2] + t.slice(3), n || s) {
          const i = t.endsWith(ie);
          t = t.replace(Pe, ""), n && h.rel(e, t, n, i), s && h.ael(e, t, s, i);
        }
      } else {
        const i = z(s);
        if ((o || i && s !== null) && !l)
          try {
            if (e.tagName.includes("-"))
              e[t] = s;
            else {
              const f = s ?? "";
              t === "list" ? o = !1 : (n == null || e[t] != f) && (e[t] = f);
            }
          } catch {
          }
        let u = !1;
        c !== (c = c.replace(/^xlink\:?/, "")) && (t = c, u = !0), s == null || s === !1 ? (s !== !1 || e.getAttribute(t) === "") && (u ? e.removeAttributeNS(I, t) : e.removeAttribute(t)) : (!o || r & 4 || l) && !i && (s = s === !0 ? "" : s, u ? e.setAttributeNS(I, t, s) : e.setAttribute(t, s));
      }
  }
}, Oe = /\s/, J = (e) => e ? e.split(Oe) : [], ie = "Capture", Pe = new RegExp(ie + "$"), re = (e, t, n, s) => {
  const l = t.$elm$.nodeType === 11 && t.$elm$.host ? t.$elm$.host : t.$elm$, r = e && e.$attrs$ || Q, o = t.$attrs$ || Q;
  for (s in r)
    s in o || K(l, s, r[s], void 0, n, t.$flags$);
  for (s in o)
    K(l, s, r[s], o[s], n, t.$flags$);
}, U = (e, t, n, s) => {
  const l = t.$children$[n];
  let r = 0, o, c, i;
  if (le || (W = !0, l.$tag$ === "slot" && (x && s.classList.add(x + "-s"), l.$flags$ |= l.$children$ ? (
    // slot element has fallback content
    2
  ) : (
    // slot element does not have fallback content
    1
  ))), l.$text$ !== null)
    o = l.$elm$ = S.createTextNode(l.$text$);
  else if (l.$flags$ & 1)
    o = l.$elm$ = S.createTextNode("");
  else {
    if (v || (v = l.$tag$ === "svg"), o = l.$elm$ = S.createElementNS(v ? Se : me, l.$flags$ & 2 ? "slot-fb" : l.$tag$), v && l.$tag$ === "foreignObject" && (v = !1), re(null, l, v), ke(x) && o["s-si"] !== x && o.classList.add(o["s-si"] = x), l.$children$)
      for (r = 0; r < l.$children$.length; ++r)
        c = U(e, l, r, o), c && o.appendChild(c);
    l.$tag$ === "svg" ? v = !1 : o.tagName === "foreignObject" && (v = !0);
  }
  return o["s-hn"] = _, l.$flags$ & 3 && (o["s-sr"] = !0, o["s-cr"] = B, o["s-sn"] = l.$name$ || "", i = e && e.$children$ && e.$children$[n], i && i.$tag$ === l.$tag$ && e.$elm$ && A(e.$elm$, !1)), o;
}, A = (e, t) => {
  h.$flags$ |= 1;
  const n = e.childNodes;
  for (let s = n.length - 1; s >= 0; s--) {
    const l = n[s];
    l["s-hn"] !== _ && l["s-ol"] && (ae(l).insertBefore(l, q(l)), l["s-ol"].remove(), l["s-ol"] = void 0, l["s-sh"] = void 0, W = !0), t && A(l, t);
  }
  h.$flags$ &= -2;
}, ce = (e, t, n, s, l, r) => {
  let o = e["s-cr"] && e["s-cr"].parentNode || e, c;
  for (o.shadowRoot && o.tagName === _ && (o = o.shadowRoot); l <= r; ++l)
    s[l] && (c = U(null, n, l, e), c && (s[l].$elm$ = c, o.insertBefore(c, q(t))));
}, $e = (e, t, n) => {
  for (let s = t; s <= n; ++s) {
    const l = e[s];
    if (l) {
      const r = l.$elm$;
      ue(l), r && (P = !0, r["s-ol"] ? r["s-ol"].remove() : A(r, !0), r.remove());
    }
  }
}, Me = (e, t, n, s, l = !1) => {
  let r = 0, o = 0, c = 0, i = 0, u = t.length - 1, f = t[0], $ = t[u], d = s.length - 1, a = s[0], p = s[d], y, g;
  for (; r <= u && o <= d; )
    if (f == null)
      f = t[++r];
    else if ($ == null)
      $ = t[--u];
    else if (a == null)
      a = s[++o];
    else if (p == null)
      p = s[--d];
    else if (O(f, a, l))
      L(f, a, l), f = t[++r], a = s[++o];
    else if (O($, p, l))
      L($, p, l), $ = t[--u], p = s[--d];
    else if (O(f, p, l))
      (f.$tag$ === "slot" || p.$tag$ === "slot") && A(f.$elm$.parentNode, !1), L(f, p, l), e.insertBefore(f.$elm$, $.$elm$.nextSibling), f = t[++r], p = s[--d];
    else if (O($, a, l))
      (f.$tag$ === "slot" || p.$tag$ === "slot") && A($.$elm$.parentNode, !1), L($, a, l), e.insertBefore($.$elm$, f.$elm$), $ = t[--u], a = s[++o];
    else {
      for (c = -1, i = r; i <= u; ++i)
        if (t[i] && t[i].$key$ !== null && t[i].$key$ === a.$key$) {
          c = i;
          break;
        }
      c >= 0 ? (g = t[c], g.$tag$ !== a.$tag$ ? y = U(t && t[o], n, c, e) : (L(g, a, l), t[c] = void 0, y = g.$elm$), a = s[++o]) : (y = U(t && t[o], n, o, e), a = s[++o]), y && ae(f.$elm$).insertBefore(y, q(f.$elm$));
    }
  r > u ? ce(e, s[d + 1] == null ? null : s[d + 1].$elm$, n, s, o, d) : o > d && $e(t, r, u);
}, O = (e, t, n = !1) => e.$tag$ === t.$tag$ ? e.$tag$ === "slot" ? e.$name$ === t.$name$ : n ? !0 : e.$key$ === t.$key$ : !1, q = (e) => e && e["s-ol"] || e, ae = (e) => (e["s-ol"] ? e["s-ol"] : e).parentNode, L = (e, t, n = !1) => {
  const s = t.$elm$ = e.$elm$, l = e.$children$, r = t.$children$, o = t.$tag$, c = t.$text$;
  let i;
  c === null ? (v = o === "svg" ? !0 : o === "foreignObject" ? !1 : v, o === "slot" || re(e, t, v), l !== null && r !== null ? Me(s, l, t, r, n) : r !== null ? (e.$text$ !== null && (s.textContent = ""), ce(s, null, t, r, 0, r.length - 1)) : l !== null && $e(l, 0, l.length - 1), v && o === "svg" && (v = !1)) : (i = s["s-cr"]) ? i.parentNode.textContent = c : e.$text$ !== c && (s.data = c);
}, fe = (e) => {
  const t = e.childNodes;
  for (const n of t)
    if (n.nodeType === 1) {
      if (n["s-sr"]) {
        const s = n["s-sn"];
        n.hidden = !1;
        for (const l of t)
          if (l !== n) {
            if (l["s-hn"] !== n["s-hn"] || s !== "") {
              if (l.nodeType === 1 && (s === l.getAttribute("slot") || s === l["s-sn"])) {
                n.hidden = !0;
                break;
              }
            } else if (l.nodeType === 1 || l.nodeType === 3 && l.textContent.trim() !== "") {
              n.hidden = !0;
              break;
            }
          }
      }
      fe(n);
    }
}, m = [], de = (e) => {
  let t, n, s;
  for (const l of e.childNodes) {
    if (l["s-sr"] && (t = l["s-cr"]) && t.parentNode) {
      n = t.parentNode.childNodes;
      const r = l["s-sn"];
      for (s = n.length - 1; s >= 0; s--)
        if (t = n[s], !t["s-cn"] && !t["s-nr"] && t["s-hn"] !== l["s-hn"] && !C.experimentalSlotFixes)
          if (Y(t, r)) {
            let o = m.find((c) => c.$nodeToRelocate$ === t);
            P = !0, t["s-sn"] = t["s-sn"] || r, o ? (o.$nodeToRelocate$["s-sh"] = l["s-hn"], o.$slotRefNode$ = l) : (t["s-sh"] = l["s-hn"], m.push({
              $slotRefNode$: l,
              $nodeToRelocate$: t
            })), t["s-sr"] && m.map((c) => {
              Y(c.$nodeToRelocate$, t["s-sn"]) && (o = m.find((i) => i.$nodeToRelocate$ === t), o && !c.$slotRefNode$ && (c.$slotRefNode$ = o.$slotRefNode$));
            });
          } else m.some((o) => o.$nodeToRelocate$ === t) || m.push({
            $nodeToRelocate$: t
          });
    }
    l.nodeType === 1 && de(l);
  }
}, Y = (e, t) => e.nodeType === 1 ? e.getAttribute("slot") === null && t === "" || e.getAttribute("slot") === t : e["s-sn"] === t ? !0 : t === "", ue = (e) => {
  e.$attrs$ && e.$attrs$.ref && e.$attrs$.ref(null), e.$children$ && e.$children$.map(ue);
}, Fe = (e, t, n = !1) => {
  var s, l, r, o, c;
  const i = e.$hostElement$, u = e.$cmpMeta$, f = e.$vnode$ || F(null, null), $ = Te(t) ? t : M(null, null, t);
  if (_ = i.tagName, u.$attrsToReflect$ && ($.$attrs$ = $.$attrs$ || {}, u.$attrsToReflect$.map(([d, a]) => $.$attrs$[a] = i[d])), n && $.$attrs$)
    for (const d of Object.keys($.$attrs$))
      i.hasAttribute(d) && !["key", "ref", "style", "class"].includes(d) && ($.$attrs$[d] = i[d]);
  $.$tag$ = null, $.$flags$ |= 4, e.$vnode$ = $, $.$elm$ = f.$elm$ = i.shadowRoot || i, x = i["s-sc"], B = i["s-cr"], le = (u.$flags$ & 1) !== 0, P = !1, L(f, $, n);
  {
    if (h.$flags$ |= 1, W) {
      de($.$elm$);
      for (const d of m) {
        const a = d.$nodeToRelocate$;
        if (!a["s-ol"]) {
          const p = S.createTextNode("");
          p["s-nr"] = a, a.parentNode.insertBefore(a["s-ol"] = p, a);
        }
      }
      for (const d of m) {
        const a = d.$nodeToRelocate$, p = d.$slotRefNode$;
        if (p) {
          const y = p.parentNode;
          let g = p.nextSibling;
          {
            let b = (s = a["s-ol"]) === null || s === void 0 ? void 0 : s.previousSibling;
            for (; b; ) {
              let k = (l = b["s-nr"]) !== null && l !== void 0 ? l : null;
              if (k && k["s-sn"] === a["s-sn"] && y === k.parentNode && (k = k.nextSibling, !k || !k["s-nr"])) {
                g = k;
                break;
              }
              b = b.previousSibling;
            }
          }
          (!g && y !== a.parentNode || a.nextSibling !== g) && a !== g && (!a["s-hn"] && a["s-ol"] && (a["s-hn"] = a["s-ol"].parentNode.nodeName), y.insertBefore(a, g), a.nodeType === 1 && (a.hidden = (r = a["s-ih"]) !== null && r !== void 0 ? r : !1));
        } else
          a.nodeType === 1 && (n && (a["s-ih"] = (o = a.hidden) !== null && o !== void 0 ? o : !1), a.hidden = !0);
      }
    }
    P && fe($.$elm$), h.$flags$ &= -2, m.length = 0;
  }
  if (C.experimentalScopedSlotChanges && u.$flags$ & 2)
    for (const d of $.$elm$.childNodes)
      d["s-hn"] !== _ && !d["s-sh"] && (n && d["s-ih"] == null && (d["s-ih"] = (c = d.hidden) !== null && c !== void 0 ? c : !1), d.hidden = !0);
  B = void 0;
}, Ue = (e, t) => {
}, pe = (e, t) => (e.$flags$ |= 16, Ue(e, e.$ancestorComponent$), rt(() => je(e, t))), je = (e, t) => {
  const n = e.$hostElement$, s = T("scheduleUpdate", e.$cmpMeta$.$tagName$), l = n;
  let r;
  return t ? r = E(l, "componentWillLoad") : r = E(l, "componentWillUpdate"), r = V(r, () => E(l, "componentWillRender")), s(), V(r, () => Be(e, l, t));
}, V = (e, t) => we(e) ? e.then(t) : t(), we = (e) => e instanceof Promise || e && e.then && typeof e.then == "function", Be = async (e, t, n) => {
  const s = e.$hostElement$, l = T("update", e.$cmpMeta$.$tagName$);
  s["s-rc"], n && He(e);
  const r = T("render", e.$cmpMeta$.$tagName$);
  De(e, t, s, n), r(), l(), Ne(e);
}, De = (e, t, n, s) => {
  try {
    Z = t, t = t.render && t.render(), e.$flags$ &= -17, e.$flags$ |= 2, (C.hasRenderFn || C.reflect) && (C.vdomRender || C.reflect) && (C.hydrateServerSide || Fe(e, t, s));
  } catch (i) {
    H(i, e.$hostElement$);
  }
  return Z = null, null;
}, Ne = (e) => {
  const t = e.$cmpMeta$.$tagName$, n = e.$hostElement$, s = T("postUpdate", t), l = n;
  e.$ancestorComponent$, E(l, "componentDidRender"), e.$flags$ & 64 ? (E(l, "componentDidUpdate"), s()) : (e.$flags$ |= 64, E(l, "componentDidLoad"), s());
}, E = (e, t, n) => {
  if (e && e[t])
    try {
      return e[t](n);
    } catch (s) {
      H(s);
    }
}, We = (e, t) => R(e).$instanceValues$.get(t), ze = (e, t, n, s) => {
  const l = R(e), r = e, o = l.$instanceValues$.get(t), c = l.$flags$, i = r;
  n = _e(n, s.$members$[t][0]);
  const u = Number.isNaN(o) && Number.isNaN(n);
  if (n !== o && !u) {
    l.$instanceValues$.set(t, n);
    {
      if (s.$watchers$ && c & 128) {
        const $ = s.$watchers$[t];
        $ && $.map((d) => {
          try {
            i[d](n, o, t);
          } catch (a) {
            H(a, r);
          }
        });
      }
      if ((c & 18) === 2) {
        if (i.componentShouldUpdate && i.componentShouldUpdate(n, o, t) === !1)
          return;
        pe(l, !1);
      }
    }
  }
}, qe = (e, t, n) => {
  var s;
  const l = e.prototype;
  if (t.$members$) {
    e.watchers && (t.$watchers$ = e.watchers);
    const r = Object.entries(t.$members$);
    r.map(([o, [c]]) => {
      (c & 31 || c & 32) && Object.defineProperty(l, o, {
        get() {
          return We(this, o);
        },
        set(i) {
          ze(this, o, i, t);
        },
        configurable: !0,
        enumerable: !0
      });
    });
    {
      const o = /* @__PURE__ */ new Map();
      l.attributeChangedCallback = function(c, i, u) {
        h.jmp(() => {
          var f;
          const $ = o.get(c);
          if (this.hasOwnProperty($))
            u = this[$], delete this[$];
          else {
            if (l.hasOwnProperty($) && typeof this[$] == "number" && this[$] == u)
              return;
            if ($ == null) {
              const d = R(this), a = d == null ? void 0 : d.$flags$;
              if (a && !(a & 8) && a & 128 && u !== i) {
                const y = this, g = (f = t.$watchers$) === null || f === void 0 ? void 0 : f[c];
                g == null || g.forEach((b) => {
                  y[b] != null && y[b].call(y, u, i, c);
                });
              }
              return;
            }
          }
          this[$] = u === null && typeof this[$] == "boolean" ? !1 : u;
        });
      }, e.observedAttributes = Array.from(/* @__PURE__ */ new Set([
        ...Object.keys((s = t.$watchers$) !== null && s !== void 0 ? s : {}),
        ...r.filter(
          ([c, i]) => i[0] & 15
          /* MEMBER_FLAGS.HasAttribute */
        ).map(([c, i]) => {
          var u;
          const f = i[1] || c;
          return o.set(f, c), i[0] & 512 && ((u = t.$attrsToReflect$) === null || u === void 0 || u.push([c, f])), f;
        })
      ]));
    }
  }
  return e;
}, Ze = async (e, t, n, s) => {
  let l;
  if (!(t.$flags$ & 32) && (t.$flags$ |= 32, l = e.constructor, customElements.whenDefined(n.$tagName$).then(() => t.$flags$ |= 128), l.style)) {
    let o = l.style;
    typeof o != "string" && (o = o[t.$modeName$ = Ee(e)]);
    const c = oe(n, t.$modeName$);
    if (!j.has(c)) {
      const i = T("registerStyles", n.$tagName$);
      Ae(c, o, !!(n.$flags$ & 1)), i();
    }
  }
  t.$ancestorComponent$, pe(t, !0);
}, ee = (e) => {
}, Ie = (e) => {
  if (!(h.$flags$ & 1)) {
    const t = R(e), n = t.$cmpMeta$, s = T("connectedCallback", n.$tagName$);
    t.$flags$ & 1 ? (he(e, t, n.$listeners$), t != null && t.$lazyInstance$ ? ee(t.$lazyInstance$) : t != null && t.$onReadyPromise$ && t.$onReadyPromise$.then(() => ee(t.$lazyInstance$))) : (t.$flags$ |= 1, // TODO(STENCIL-854): Remove code related to legacy shadowDomShim field
    n.$flags$ & 12 && Qe(e), n.$members$ && Object.entries(n.$members$).map(([l, [r]]) => {
      if (r & 31 && e.hasOwnProperty(l)) {
        const o = e[l];
        delete e[l], e[l] = o;
      }
    }), Ze(e, t, n)), s();
  }
}, Qe = (e) => {
  const t = e["s-cr"] = S.createComment("");
  t["s-cn"] = !0, e.insertBefore(t, e.firstChild);
}, Xe = async (e) => {
  if (!(h.$flags$ & 1)) {
    const t = R(e);
    t.$rmListeners$ && (t.$rmListeners$.map((n) => n()), t.$rmListeners$ = void 0);
  }
}, Ge = (e, t) => {
  const n = {
    $flags$: t[0],
    $tagName$: t[1]
  };
  n.$members$ = t[2], n.$listeners$ = t[3], n.$watchers$ = e.$watchers$, n.$attrsToReflect$ = [];
  const s = e.prototype.connectedCallback, l = e.prototype.disconnectedCallback;
  return Object.assign(e.prototype, {
    __registerHost() {
      Ve(this, n);
    },
    connectedCallback() {
      Ie(this), s && s.call(this);
    },
    disconnectedCallback() {
      Xe(this), l && l.call(this);
    },
    __attachShadow() {
      this.attachShadow({
        mode: "open",
        delegatesFocus: !!(n.$flags$ & 16)
      });
    }
  }), e.is = n.$tagName$, qe(e, n);
}, he = (e, t, n, s) => {
  n && n.map(([l, r, o]) => {
    const c = Je(e, l), i = Ke(t, o), u = Ye(l);
    h.ael(c, r, i, u), (t.$rmListeners$ = t.$rmListeners$ || []).push(() => h.rel(c, r, i, u));
  });
}, Ke = (e, t) => (n) => {
  try {
    C.lazyLoad || e.$hostElement$[t](n);
  } catch (s) {
    H(s);
  }
}, Je = (e, t) => t & 4 ? S : t & 8 ? w : t & 16 ? S.body : e, Ye = (e) => st ? {
  passive: (e & 1) !== 0,
  capture: (e & 2) !== 0
} : (e & 2) !== 0, ge = /* @__PURE__ */ new WeakMap(), R = (e) => ge.get(e), Ve = (e, t) => {
  const n = {
    $flags$: 0,
    $hostElement$: e,
    $cmpMeta$: t,
    $instanceValues$: /* @__PURE__ */ new Map()
  };
  return he(e, n, t.$listeners$), ge.set(e, n);
}, te = (e, t) => t in e, H = (e, t) => (0, console.error)(e, t), j = /* @__PURE__ */ new Map(), et = [], w = typeof window < "u" ? window : {}, S = w.document || { head: {} }, tt = w.HTMLElement || class {
}, h = {
  $flags$: 0,
  $resourcesUrl$: "",
  jmp: (e) => e(),
  raf: (e) => requestAnimationFrame(e),
  ael: (e, t, n, s) => e.addEventListener(t, n, s),
  rel: (e, t, n, s) => e.removeEventListener(t, n, s),
  ce: (e, t) => new CustomEvent(e, t)
}, st = /* @__PURE__ */ (() => {
  let e = !1;
  try {
    S.addEventListener("e", null, Object.defineProperty({}, "passive", {
      get() {
        e = !0;
      }
    }));
  } catch {
  }
  return e;
})(), nt = (e) => Promise.resolve(e), lt = /* @__PURE__ */ (() => {
  try {
    return new CSSStyleSheet(), typeof new CSSStyleSheet().replaceSync == "function";
  } catch {
  }
  return !1;
})(), se = [], ye = [], ot = (e, t) => (n) => {
  e.push(n), D || (D = !0, t && h.$flags$ & 4 ? it(N) : h.raf(N));
}, ne = (e) => {
  for (let t = 0; t < e.length; t++)
    try {
      e[t](performance.now());
    } catch (n) {
      H(n);
    }
  e.length = 0;
}, N = () => {
  ne(se), ne(ye), (D = se.length > 0) && h.raf(N);
}, it = (e) => nt().then(e), rt = /* @__PURE__ */ ot(ye, !0), ct = ".sc-person-clipboard-icon-h{display:flex}", $t = ct, at = Ge(class extends tt {
  constructor() {
    super(), this.__registerHost(), this.width = "12", this.height = "17";
  }
  render() {
    return M("svg", { key: "c339794a491d77de3c5482524be059c78bf6a626", width: this.width, height: this.height, fill: "none", xmlns: "http://www.w3.org/2000/svg" }, M("path", { key: "91396827e84c8bda655a95b5f854c1851d969d5e", d: "M4 7.5C4 6.39543 4.89543 5.5 6 5.5C7.10457 5.5 8 6.39543 8 7.5C8 8.60457 7.10457 9.5 6 9.5C4.89543 9.5 4 8.60457 4 7.5ZM6 6.5C5.44772 6.5 5 6.94772 5 7.5C5 8.05228 5.44772 8.5 6 8.5C6.55228 8.5 7 8.05228 7 7.5C7 6.94772 6.55228 6.5 6 6.5ZM3.72077 10.5C3.10838 10.5 2.40961 10.9146 2.38992 11.6973C2.37693 12.2135 2.48426 12.9381 3.04816 13.5319C3.61451 14.1283 4.55301 14.5 6.00001 14.5C7.44702 14.5 8.38552 14.1283 8.95187 13.5319C9.51577 12.9381 9.6231 12.2135 9.61011 11.6973C9.59042 10.9146 8.89165 10.5 8.27925 10.5H3.72077ZM3.3896 11.7225C3.39099 11.6675 3.41262 11.6226 3.46275 11.582C3.51876 11.5366 3.6101 11.5 3.72077 11.5H8.27925C8.38993 11.5 8.48127 11.5366 8.53728 11.582C8.58741 11.6226 8.60904 11.6675 8.61042 11.7225C8.61975 12.0933 8.53969 12.5138 8.22676 12.8433C7.91628 13.1702 7.28849 13.5 6.00001 13.5C4.71154 13.5 4.08375 13.1702 3.77327 12.8433C3.46033 12.5138 3.38027 12.0933 3.3896 11.7225ZM5 0.5C3.89543 0.5 3 1.39543 3 2.5H1.5C0.671573 2.5 0 3.17157 0 4V15C0 15.8284 0.671572 16.5 1.5 16.5H10.5C11.3284 16.5 12 15.8284 12 15V4C12 3.17157 11.3284 2.5 10.5 2.5H9C9 1.39543 8.10457 0.5 7 0.5H5ZM8.73244 3.5H10.5C10.7761 3.5 11 3.72386 11 4V15C11 15.2761 10.7761 15.5 10.5 15.5H1.5C1.22386 15.5 1 15.2761 1 15V4C1 3.72386 1.22386 3.5 1.5 3.5H3.26756C3.61337 4.0978 4.25972 4.5 5 4.5H7C7.74028 4.5 8.38663 4.0978 8.73244 3.5ZM4 2.5C4 1.94772 4.44772 1.5 5 1.5H7C7.55228 1.5 8 1.94772 8 2.5C8 3.05228 7.55228 3.5 7 3.5H5C4.44772 3.5 4 3.05228 4 2.5Z", fill: "currentColor" }));
  }
  static get style() {
    return $t;
  }
}, [2, "person-clipboard-icon", {
  width: [1],
  height: [1]
}]);
function dt() {
  if (typeof customElements > "u")
    return;
  ["person-clipboard-icon"].forEach((t) => {
    switch (t) {
      case "person-clipboard-icon":
        customElements.get(t) || customElements.define(t, at);
        break;
    }
  });
}
export {
  at as P,
  dt as d
};
