// @nylas/.web-elements-GF2lXkj6/dist/esm/nylas-booking-form.entry.js
import { r as registerInstance, a as createEvent, h, i as Host, e as getElement } from './index-b5cb18eb.js';
import { R as RegisterComponent } from './register-component-8057b2e5.js';
import { s as sanitizeHtml_1, i as instance, a as debug, o as sanitize } from './utils-674a951a.js';
import './_commonjsHelpers-6cbf8316.js';

const nylasBookingFormCss = ":host{display:block;height:100%}.nylas-booking-form{width:100%;padding:1.5rem;box-sizing:border-box;height:450px;overflow-y:auto;height:100%}form{height:100%;display:flex;flex-direction:column;justify-content:space-between;align-items:center}sp-textfield{--spectrum-corner-radius-100:var(--nylas-border-radius-2x);--spectrum-textfield-border-color:var(--nylas-base-300);position:relative;width:100%;height:48px;margin-bottom:16px;color:var(--nylas-base-500)}.cta{width:100%;display:grid;grid-template-columns:1fr 1fr;gap:0.5rem;align-items:flex-start;justify-content:space-between;gap:0.5rem;padding:14px;box-sizing:border-box;border-top:1px solid var(--nylas-base-200)}@media screen and (max-width: 768px){.cta{grid-template-columns:1fr}}.input-wrapper{width:100%;display:flex;flex-direction:column;align-items:flex-start;gap:4px;margin-bottom:1rem}.input-wrapper input::placeholder{color:var(--nylas-base-300)}.input-wrapper label,.input-wrapper input{font-family:var(--nylas-font-family);font-size:16px}.input-wrapper .help-text{margin:0;color:var(--nylas-error)}.button-wrapper{width:100%;align-items:flex-start;height:max-content;position:relative}.button-content{display:flex;gap:4px;align-items:center}sp-infield-button{--spectrum-infield-button-background-color:transparent;--spectrum-infield-button-background-color-hover:transparent;--spectrum-infield-button-background-color-active:transparent;--spectrum-infield-button-border-color:transparent;--spectrum-infield-button-border-width:1px;--spectrum-infield-button-border-radius:0;color:var(--nylas-base-600);background:transparent;border:none;outline:1px solid var(--nylas-base-300);border-radius:0 var(--nylas-border-radius-2x) var(--nylas-border-radius-2x) 0;height:52px;width:52px;display:flex;position:absolute;right:0;align-items:center;justify-content:center;cursor:pointer}sp-infield-button sp-icon-cross300{color:var(--nylas-base-600)}sp-infield-button.error{border-width:2px;border-color:var(--nylas-error)}input[type=text],input[type=email]{display:flex;height:48px;padding-left:16px;align-items:center;gap:8px;align-self:stretch;border-radius:var(--nylas-border-radius-2x);border:1px solid var(--nylas-base-300);background:var(--nylas-base-0);width:-webkit-fill-available;color:var(--nylas-base-900);font-size:16px;font-style:normal;font-weight:400;line-height:150%}input[type=text]:focus,input[type=email]:focus{outline-color:var(--nylas-primary)}input[type=text].error,input[type=email].error{border-radius:var(--nylas-border-radius-2x);font-weight:600;border-color:var(--nylas-error);border-width:2px}input[type=text].error::placeholder,input[type=email].error::placeholder{font-weight:400}input[type=text]::placeholder,input[type=email]::placeholder{color:var(--nylas-base-300)}input[type=text].guest-email,input[type=email].guest-email{color:var(--nylas-base-900);padding-right:52px}label{display:flex;color:var(--nylas-base-800, #293056);font-size:14px;font-style:normal;font-weight:400;line-height:150%}label.error{color:var(--nylas-error)}label span.required{color:var(--nylas-error);padding:0 0.25rem}button-component.remove-guest{position:absolute;height:51px;right:0}button-component.remove-guest.error button{border-color:var(--nylas-error)}button-component.remove-guest button{height:inherit;border-top-left-radius:initial;border-bottom-left-radius:initial;border:none !important;border-left:1px solid var(--nylas-base-300) !important}button-component.remove-guest button:hover,button-component.remove-guest button:focus{outline:1px solid var(--nylas-primary) !important;color:var(--nylas-primary)}button-component.back button{--dot-color:var(--nylas-base-800)}.guest-email-input{position:relative;display:flex;width:100%}";

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (undefined && undefined.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
        return Reflect.metadata(k, v);
};
const emailRegex = /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$/;
const NylasBookingForm = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.nameChanged = createEvent(this, "nameChanged", 7);
        this.emailChanged = createEvent(this, "emailChanged", 7);
        this.backButtonClicked = createEvent(this, "backButtonClicked", 7);
        this.detailsConfirmed = createEvent(this, "detailsConfirmed", 7);
        this.bookingFormError = createEvent(this, "bookingFormError", 7);
        this.bookingFormSubmitted = createEvent(this, "bookingFormSubmitted", 7);
        this.handleGuestChange = (guestIndex, email) => {
            this.resetGuestEmailError(guestIndex);
            const guests = [...this.guestEmails];
            guests[guestIndex] = sanitizeHtml_1(email);
            this.guestEmails = [...guests];
        };
        this.handleGuestBlur = (guestIndex, email) => {
            this.resetGuestEmailError(guestIndex);
            if (email === '') {
                this.guestEmailErrors[guestIndex] = instance.t('fieldRequired', { field: instance.t('email') });
                return;
            }
            if (!emailRegex.test(email)) {
                this.guestEmailErrors[guestIndex] = instance.t('invalidInputFormat', { field: instance.t('email') });
                return;
            }
            this.handleGuestChange(guestIndex, email);
        };
        this.addGuestButtonClickedHandler = e => {
            e.preventDefault();
            this.guestEmails = [...this.guestEmails, ''];
        };
        this.removeGuestButtonClickHandler = (e, guestIndex) => {
            e.preventDefault();
            this.resetGuestEmailError(guestIndex);
            const guests = [...this.guestEmails];
            guests.splice(guestIndex, 1);
            this.guestEmails = [...guests];
        };
        this.handleBackButtonClicked = e => {
            e.preventDefault();
            this.backButtonLoading = true;
            this.backButtonClicked.emit();
        };
        this.bookButtonClickedHandler = async (e) => {
            e.preventDefault();
            const formSubmittedEvent = this.bookingFormSubmitted.emit();
            if (formSubmittedEvent.defaultPrevented) {
                return;
            }
            debug('nylas-booking-form', 'bookButtonClickedHandler', this.bookingInfo);
            const name = this.name || this.bookingInfo?.primaryParticipant?.name;
            const email = this.email || this.bookingInfo?.primaryParticipant?.email;
            if (!name || name === '') {
                this.isNameValid = false;
                this.validationError.name = instance.t('fieldRequired', { field: instance.t('name') }) || 'Name is required';
                this.bookingFormError.emit({
                    title: 'Booking form error',
                    description: 'Name is required',
                });
                return;
            }
            if (!email || email === '') {
                this.isEmailValid = false;
                this.validationError.email = instance.t('fieldRequired', { field: instance.t('email') }) || 'Email is required';
                this.bookingFormError.emit({
                    title: 'Booking form error',
                    description: 'Email is required',
                });
                return;
            }
            if (!emailRegex.test(email)) {
                this.isEmailValid = false;
                this.validationError.email = instance.t('invalidInputFormat', { field: instance.t('email') });
                this.bookingFormError.emit({
                    title: 'Booking form error',
                    description: instance.t('invalidInputFormat', { field: instance.t('email') }) || 'Invalid email',
                });
                return;
            }
            let guests = [];
            if (this.guestEmails.length > 0) {
                let hasError = false;
                this.guestEmails.forEach((email, i) => {
                    if (email === '') {
                        hasError = true;
                        this.guestEmailErrors = { ...this.guestEmailErrors, [i]: instance.t('fieldRequired', { field: instance.t('email') }) };
                    }
                    else if (!emailRegex.test(email)) {
                        hasError = true;
                        this.guestEmailErrors = { ...this.guestEmailErrors, [i]: instance.t('invalidInputFormat', { field: instance.t('email') }) };
                    }
                });
                if (hasError) {
                    this.bookingFormError.emit({
                        title: 'Booking form error',
                        description: instance.t('invalidInputFormat', { field: instance.t('guestEmail') }) || 'Invalid guest email',
                    });
                    return;
                }
                guests = this.guestEmails.map((email) => ({ name: email.trim(), email: email.trim() }));
            }
            else if (this.bookingInfo?.guests) {
                guests = this.bookingInfo?.guests;
            }
            const bookingInfoAdditionalFields = this.bookingInfo?.additionalFields || [];
            this.detailsConfirmed.emit({
                primaryParticipant: {
                    name: name,
                    email: email,
                },
                guests: guests,
                additionalFields: { ...bookingInfoAdditionalFields, ...this.additionalFields },
            });
        };
        this.updateAdditionalFields = (name, value) => {
            if (this.additionalFields) {
                this.additionalFields[name] = value;
            }
        };
        this.isLoading = undefined;
        this.bookingInfo = undefined;
        this.eventInfo = undefined;
        this.configSettings = undefined;
        this.name = '';
        this.email = '';
        this.guestEmails = [];
        this.guestEmailErrors = {};
        this.isNameValid = true;
        this.isEmailValid = true;
        this.additionalFields = {};
        this.validationError = { name: '', email: '' };
        this.backButtonLoading = false;
    }
    connectedCallback() { }
    disconnectedCallback() { }
    componentWillLoad() {
        debug('nylas-booking-form', 'componentWillLoad', this.bookingInfo);
        if (this.bookingInfo && this.bookingInfo?.primaryParticipant) {
            this.name = this.bookingInfo.primaryParticipant?.name;
            this.email = this.bookingInfo.primaryParticipant?.email;
        }
    }
    bookingInfoChangedHandler(newValue) {
        if (newValue && newValue.primaryParticipant) {
            this.name = newValue.primaryParticipant.name;
            this.email = newValue.primaryParticipant.email;
        }
    }
    changeName(name) {
        this.isNameValid = true;
        this.validationError.name = '';
        this.name = sanitize(name);
        this.nameChanged.emit(name);
    }
    changeEmail(email) {
        this.isEmailValid = true;
        this.validationError.email = '';
        this.email = sanitizeHtml_1(email);
        this.emailChanged.emit(email);
    }
    resetGuestEmailError(guestIndex) {
        const errors = { ...this.guestEmailErrors };
        delete errors[guestIndex];
        this.guestEmailErrors = { ...errors };
    }
    nylasFormInputChangedHandler(event) {
        switch (event.detail.name) {
            case 'name':
                this.changeName(event.detail.value);
                break;
            case 'email':
                this.changeEmail(event.detail.value);
                break;
            default:
                if (event.detail.error) {
                    this.bookingFormError.emit({
                        title: 'Booking form error',
                        description: event.detail.error,
                    });
                    return;
                }
                this.updateAdditionalFields(event.detail.name, event.detail.value);
                break;
        }
    }
    selectOptionChangedHandler(event) {
        if (event.detail.error) {
            this.bookingFormError.emit({
                title: 'Booking form error',
                description: event.detail.error,
            });
            return;
        }
        this.updateAdditionalFields(event.detail.name, event.detail.value);
    }
    switchToggledHandler(event) {
        this.updateAdditionalFields(event.detail.name, event.detail.checked ? 'true' : 'false');
    }
    checkboxToggledHandler(event) {
        this.updateAdditionalFields(event.detail.name, event.detail.checked ? 'true' : 'false');
    }
    radioChangedHandler(event) {
        this.updateAdditionalFields(event.detail.name, event.detail.value);
    }
    dropdownDefaultSelectedHandler(event) {
        this.updateAdditionalFields(event.detail.name, event.detail.value);
    }
    render() {
        return (h(Host, { key: 'e6226715a8817b1ac42f46e3efd78cc2aa1d413c' }, h("form", { key: 'd6e4aff3c6f9b978bef7cc7c1becdd8f84f2897d', onSubmit: e => this.bookButtonClickedHandler(e), noValidate: true }, h("div", { key: 'c3561940f06b86fab4eb248626cb63a09cbf1353', class: "nylas-booking-form", part: "nbf" }, h("div", { key: '02072469ac898b656c9829c484a24415d9e8e597', class: "input-wrapper", part: "nbf__input-wrapper" }, h("input-component", { key: '301574dd5c31489404efc426bc41d413b659c0a4', label: instance.t('name'), name: "name", id: "name", defaultValue: this.bookingInfo?.primaryParticipant?.name || this.name, placeholder: instance.t('namePlaceholder'), type: "text", required: true, requiredError: instance.t('fieldRequired', { field: instance.t('name') }), patternError: instance.t('invalidInputFormat', { field: instance.t('name') }), part: "nbf__input-textfield" })), h("div", { key: 'af992058321d3efbc622e335400da283123fe54c', class: "input-wrapper", part: "nbf__input-wrapper" }, h("input-component", { key: '3815bf80434a7178dcdc75fde5ee80db5c651179', label: instance.t('email'), name: "email", id: "email", pattern: emailRegex, defaultValue: this.bookingInfo?.primaryParticipant?.email || this.email, placeholder: instance.t('emailPlaceholder'), type: "email", required: true, requiredError: instance.t('fieldRequired', { field: instance.t('email') }), patternError: instance.t('invalidInputFormat', { field: instance.t('email') }), part: "nbf__input-textfield" })), this.configSettings?.scheduler?.hide_additional_guests !== true && (h("div", { class: "input-wrapper", part: "nbf__input-wrapper" }, this.guestEmails.map((email, index) => (h("div", { class: "input-wrapper button-wrapper" }, h("label", { class: {
                error: !!this.guestEmailErrors[index],
            } }, instance.t('guestEmail')), h("div", { class: "guest-email-input" }, h("input", { type: "email", id: `guest-email-${index}`, maxLength: 100, class: {
                'guest-email': true,
                'error': !!this.guestEmailErrors[index],
            }, placeholder: instance.t('guestEmailPlaceholder'), value: email, "data-index": index, part: "nbf__input-textfield", onBlur: (e) => this.handleGuestBlur(index, e.target?.value), onInput: (e) => this.handleGuestChange(index, e.target?.value) }), h("button-component", { variant: "basic", onClick: (e) => this.removeGuestButtonClickHandler(e, index), class: {
                'remove-guest': true,
                'error': !!this.guestEmailErrors[index],
            } }, h("close-icon", null))), h("p", { class: "help-text" }, this.guestEmailErrors[index])))), h("button-component", { class: "add-guest", variant: "invisible", part: "nbf__button-ghost", onClick: e => this.addGuestButtonClickedHandler(e) }, h("div", { class: "button-content" }, h("add-circle-icon", { slot: "icon" }), this.guestEmails.length > 0 ? `${instance.t('addAnotherGuest')}` : `${instance.t('addGuest')}`)))), Object.entries((this.configSettings?.scheduler?.additional_fields || {}))
            ?.sort((a, b) => a[1].order - b[1].order)
            .map(x => {
            const field = x[1];
            const label = field.label;
            const key = x[0];
            const defaultValue = this.bookingInfo?.additionalFields?.[key]?.value || '';
            switch (field.type) {
                case 'dropdown':
                    return (h("div", { class: "input-wrapper", part: "nbf__input-wrapper" }, h("select-dropdown", { label: label, name: key, id: key, withSearch: false, defaultSelectedOption: defaultValue ? { value: defaultValue, label: defaultValue } : undefined, options: field.options?.map(option => ({ value: option, label: option })), required: field.required, exportparts: "sd_dropdown: nbf__dropdown, sd_dropdown-button: nbf__dropdown-button, sd_dropdown-content: nbf__dropdown-content" })));
                case 'checkbox':
                    return (h("div", { class: "input-wrapper", part: "nbf__input-wrapper" }, h("checkbox-component", { label: label, name: key, id: key, checked: defaultValue === 'true', required: field.required, part: "nbf__checkbox-component" })));
                case 'radio_button':
                    return (h("div", { class: "input-wrapper", part: "nbf__input-wrapper" }, h("radio-button-group", { label: label, name: key, id: key, defaultSelectedValue: defaultValue || field.options?.[0] || '', options: field.options?.map(option => ({ value: option, label: option })), required: field.required, part: "nbf__radio-button-group" })));
                case 'multi_line_text':
                    return (h("div", { class: "input-wrapper", part: "nbf__input-wrapper" }, h("textarea-component", { label: label, defaultValue: defaultValue, name: key, id: key, required: field.required, part: "nbf__textarea-component" })));
                default:
                    return (h("div", { class: "input-wrapper", part: "nbf__input-wrapper" }, h("input-component", { label: label, name: key, id: key, type: field.type, defaultValue: defaultValue || '', requiredError: instance.t('fieldRequired', { field: label }), required: field.required, part: "nbf__input-textfield" })));
            }
        })), h("div", { key: '0f18461673aa4542e73b9316509aa7b5a2fba6f3', class: "cta" }, h("button-component", { key: '7041a279f4acc3fb2029321dec40b9fffd9d9a6d', variant: 'basic', class: "back", part: "nbf__button-outline", isLoading: this.backButtonLoading && this.isLoading, onClick: this.handleBackButtonClicked }, h("slot", { key: '6bf9092b2143c0b48cf56f2159d91283f804c3a3', name: "booking-form-back-label" }, `${instance.t('backButton')}`)), h("button-component", { key: '2e97b28476b04ede3687737c78bfa6a8814b42c9', variant: 'primary', isLoading: !this.backButtonLoading && this.isLoading, disabled: this.isLoading, part: "nbf__button-primary", type: "submit" }, h("slot", { key: 'e708cfa7c230cced4f75c814a660af51a1b8dfe4', name: "booking-form-book-label" }, `${instance.t('bookNowButton')}`))))));
    }
    get host() { return getElement(this); }
    static get watchers() { return {
        "bookingInfo": ["bookingInfoChangedHandler"]
    }; }
};
__decorate([
    RegisterComponent({
        name: 'nylas-booking-form',
        stateToProps: new Map([
            ['scheduler.isLoading', 'isLoading'],
            ['scheduler.bookingInfo', 'bookingInfo'],
            ['scheduler.eventInfo', 'eventInfo'],
            ['scheduler.configSettings', 'configSettings'],
        ]),
        eventToProps: {
            backButtonClicked: async (_event, nylasSchedulerConnector) => {
                nylasSchedulerConnector.scheduler.toggleAdditionalData(false);
            },
            nameChanged: async (event, nylasSchedulerConnector) => {
                nylasSchedulerConnector.scheduler.setParticipantName(event.detail);
            },
            emailChanged: async (event, nylasSchedulerConnector) => {
                nylasSchedulerConnector.scheduler.setParticipantEmail(event.detail);
            },
            detailsConfirmed: async (event, _nylasSchedulerConnector) => {
                debug('nylas-booking-form', 'detailsConfirmed', event.detail);
            },
            bookingFormSubmitted: async (event, _nylasSchedulerConnector) => {
                debug('nylas-booking-form', 'bookingFormSubmitted', event.detail);
            },
            bookingFormError: async (event, _nylasSchedulerConnector) => {
                debug('nylas-booking-form', 'bookingFormError', event.detail);
            },
        },
        fireRegisterEvent: true,
    }),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", []),
    __metadata("design:returntype", void 0)
], NylasBookingForm.prototype, "render", null);
NylasBookingForm.style = nylasBookingFormCss;

export { NylasBookingForm as nylas_booking_form };
