// @nylas/.web-elements-GF2lXkj6/dist/esm/toggle-switch.entry.js
import { r as registerInstance, a as createEvent, h } from './index-b5cb18eb.js';

const toggleSwitchCss = ".switch{--nylas-primary:#2563eb;--nylas-error:#cc4841;--nylas-error-pressed:#992222;--nylas-wraning:#f06c00;--nylas-success:#16a392;--nylas-info:#2b8fc2;--nylas-base-0:#ffffff;--nylas-base-25:#fcfcfd;--nylas-base-50:#f8f9fc;--nylas-base-100:#eaecf5;--nylas-base-200:#d5d9eb;--nylas-base-300:#b3b8d8;--nylas-base-400:#717bbc;--nylas-base-500:#4e5ba6;--nylas-base-600:#3e4784;--nylas-base-700:#263f72;--nylas-base-800:#293056;--nylas-base-900:#101323;--nylas-base-950:#0e101b;--nylas-border-radius:0.25rem;--nylas-border-radius-2x:0.5rem;--nylas-border-radius-3x:0.75rem;--nylas-font-family:\"Inter\", sans-serif;display:inline-block;height:34px;position:relative;width:60px}.switch input{display:none}.slider{background-color:var(--nylas-base-300);bottom:0;cursor:pointer;left:0;position:absolute;right:0;top:0;transition:0.4s}.slider:before{background-color:var(--nylas-base-0);bottom:4px;content:\"\";height:32px;left:4px;position:absolute;transition:0.4s;width:32px}input:checked+.slider{background-color:var(--nylas-success)}input:checked+.slider:before{transform:translateX(32px)}.slider.round{border-radius:35px;width:72px;height:40px}.slider.round:before{border-radius:50%}";

const ToggleSwitchComponent = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.nylasFormSwitchToggled = createEvent(this, "nylasFormSwitchToggled", 7);
        this.name = 'toggle-switch';
        this.checked = false;
        this.label = '';
        this.isChecked = this.checked;
    }
    componentWillLoad() {
        this.isChecked = this.checked;
    }
    toggleCheck() {
        this.isChecked = !this.isChecked;
        this.nylasFormSwitchToggled.emit({
            checked: this.isChecked,
            name: this.name,
            label: this.label,
        });
    }
    render() {
        return (h("label", { key: 'c23c77986617f95474b950d7e34e0c8fde52de8e', class: "switch", part: "ts_label" }, h("input", { key: 'e1d164a9413c00949a1e29f12c45a9bdc9ba04e1', type: "checkbox", part: "ts_input", name: this.name, checked: this.isChecked, onChange: () => this.toggleCheck() }), h("div", { key: 'cfa88bfbb93261ffdf1344c4cb78fcfc8b71dd28', class: "slider round", part: "ts_slider" })));
    }
};
ToggleSwitchComponent.style = toggleSwitchCss;

export { ToggleSwitchComponent as toggle_switch };
