// @nylas/.web-elements-GF2lXkj6/dist/components/arrow.js
import { proxyCustomElement, HTMLElement, h } from '@stencil/core/internal/client';

const iconCss = ".sc-arrow-icon-h{display:flex}";

const ArrowIcon = proxyCustomElement(class ArrowIcon extends HTMLElement {
    constructor() {
        super();
        this.__registerHost();
        this.width = '15';
        this.height = '15';
    }
    render() {
        return (h("svg", { key: 'daefefeb12e2b050fb85a40756ed665264d6e160', width: this.width, height: this.height, viewBox: "0 0 24 24", fill: "none", xmlns: "http://www.w3.org/2000/svg" }, h("path", { key: 'e9e38a9bfb652dc0af744b91ae4d1b588949edf5', d: "M5 12H19M12 5L19 12L12 19", stroke: "currentColor", "stroke-width": "2", "stroke-linecap": "round", "stroke-linejoin": "round" })));
    }
    static get style() { return iconCss; }
}, [2, "arrow-icon", {
        "width": [1],
        "height": [1]
    }]);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["arrow-icon"];
    components.forEach(tagName => { switch (tagName) {
        case "arrow-icon":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, ArrowIcon);
            }
            break;
    } });
}

export { ArrowIcon as A, defineCustomElement as d };
