// @nylas/.web-elements-GF2lXkj6/dist/components/bold.js
import { proxyCustomElement, HTMLElement, h } from '@stencil/core/internal/client';

const iconCss = ".sc-bold-icon-h{display:flex}";

const BoldIcon = proxyCustomElement(class BoldIcon extends HTMLElement {
    constructor() {
        super();
        this.__registerHost();
        this.width = '15';
        this.height = '15';
    }
    render() {
        return (h("svg", { key: '93ec49e234ab8841098d7024f4000f2047a950fd', width: this.width, height: this.height, viewBox: "0 0 24 24", fill: "none", xmlns: "http://www.w3.org/2000/svg" }, h("path", { key: '1713de8140949f52a489b05f2ab92cc2b09b49ec', d: "M6 4V20M9.5 4H15.5C17.7091 4 19.5 5.79086 19.5 8C19.5 10.2091 17.7091 12 15.5 12H9.5H16.5C18.7091 12 20.5 13.7909 20.5 16C20.5 18.2091 18.7091 20 16.5 20H9.5M9.5 4V20M9.5 4H4M9.5 20H4", stroke: "currentColor", "stroke-width": "2", "stroke-linecap": "round", "stroke-linejoin": "round" })));
    }
    static get style() { return iconCss; }
}, [2, "bold-icon", {
        "width": [1],
        "height": [1]
    }]);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["bold-icon"];
    components.forEach(tagName => { switch (tagName) {
        case "bold-icon":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, BoldIcon);
            }
            break;
    } });
}

export { BoldIcon as B, defineCustomElement as d };
