// @nylas/.web-elements-GF2lXkj6/dist/components/button-component2.js
import { proxyCustomElement, HTMLElement, h } from '@stencil/core/internal/client';
import { d as defineCustomElement$1 } from './loading.js';

const buttonComponentCss = "button.sc-button-component{display:flex;align-items:center;justify-content:center;gap:0.75rem;height:48px;border:none;border-radius:var(--nylas-border-radius-2x);font-size:1rem;line-height:24px;padding:0 1rem;cursor:pointer;font-family:var(--nylas-font-family);font-weight:600;width:100%}@media screen and (max-width: 768px){button.sc-button-component{width:100%}}button.button-primary.sc-button-component{background:var(--nylas-primary);color:var(--nylas-base-0);height:48px;border:none;border-radius:var(--nylas-border-radius-2x);font-size:1rem;line-height:24px;padding:0 1rem;cursor:pointer}button.button-primary.sc-button-component:hover{background:var(--nylas-base-600)}button.button-primary.sc-button-component:active{background:var(--nylas-base-800)}button.button-primary.sc-button-component:disabled{background:var(--nylas-base-300);cursor:not-allowed}button.button-basic.sc-button-component{border:1px solid var(--nylas-base-300);color:var(--nylas-base-700);background:transparent}button.button-basic.sc-button-component:hover{background:transparent;border-color:var(--nylas-primary)}button.button-basic.sc-button-component:active{background:transparent;border-color:var(--nylas-base-600);border-width:2px}button.button-basic.sc-button-component:disabled{background:transparent;border-color:var(--nylas-base-300);color:var(--nylas-base-300);cursor:not-allowed}button.button-destructive.sc-button-component{border:1px solid var(--nylas-base-300);color:var(--nylas-error);background:transparent}button.button-destructive.sc-button-component:hover{background:transparent;border-color:var(--nylas-error)}button.button-destructive.sc-button-component:active{background:transparent;border-color:var(--nylas-error-pressed);border-width:2px;color:var(--nylas-error-pressed)}button.button-destructive.sc-button-component:disabled{background:transparent;border-color:var(--nylas-base-300);color:var(--nylas-base-300);cursor:not-allowed}button.button-invisible.sc-button-component{background:transparent;color:var(--nylas-base-800);border:none}button.button-invisible.sc-button-component:hover{background:var(--nylas-base-50);color:var(--nylas-primary)}button.button-invisible.sc-button-component:active{background:var(--nylas-base-50);color:var(--nylas-base-850)}button.button-invisible.sc-button-component:disabled{background:transparent;color:var(--nylas-base-300);cursor:not-allowed}";

const ButtonComponent = proxyCustomElement(class ButtonComponent extends HTMLElement {
    constructor() {
        super();
        this.__registerHost();
        this.variant = 'primary';
        this.disabled = false;
        this.type = 'button';
        this.isLoading = false;
        this.clickHandler = undefined;
        this.mouseOverHandler = undefined;
        this.mouseOutHandler = undefined;
        this.focusHandler = undefined;
        this.blurHandler = undefined;
    }
    render() {
        const className = `button-${this.variant}` + (this.disabled ? ' disabled' : '');
        return (h("button", { key: '8e535c0a5cb3be758941eee5ac5bf9b1e65b9e42', class: className, disabled: this.disabled, type: "type", onClick: this.clickHandler, onMouseOver: this.mouseOverHandler, onMouseOut: this.mouseOutHandler, onFocus: this.focusHandler, onBlur: this.blurHandler }, !!this.isLoading && h("loading-icon", { slot: "icon" }), h("slot", { key: 'cc75d33abd248961336b47c5ae558c87d1ddce1e' })));
    }
    static get style() { return buttonComponentCss; }
}, [6, "button-component", {
        "variant": [1],
        "disabled": [4],
        "type": [1],
        "isLoading": [4, "is-loading"],
        "clickHandler": [16],
        "mouseOverHandler": [16],
        "mouseOutHandler": [16],
        "focusHandler": [16],
        "blurHandler": [16]
    }]);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["button-component", "loading-icon"];
    components.forEach(tagName => { switch (tagName) {
        case "button-component":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, ButtonComponent);
            }
            break;
        case "loading-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$1();
            }
            break;
    } });
}

export { ButtonComponent as B, defineCustomElement as d };
