// @nylas/.web-elements-GF2lXkj6/dist/components/checkmark.js
import { proxyCustomElement, HTMLElement, h } from '@stencil/core/internal/client';

const iconCss = ".sc-checkmark-icon-h{display:flex}";

const CheckmarkIcon = proxyCustomElement(class CheckmarkIcon extends HTMLElement {
    constructor() {
        super();
        this.__registerHost();
        this.width = '18';
        this.height = '14';
    }
    render() {
        return (h("svg", { key: 'e8ee67dad2ae6438b3358babc1350245ec04de72', width: this.width, height: this.height, viewBox: "0 0 18 14", fill: "none", xmlns: "http://www.w3.org/2000/svg" }, h("path", { key: 'de655451a6d3d50b7b6ba66c5267842ff3385b89', d: "M1.53033 7.46967C1.23744 7.17678 0.762563 7.17678 0.46967 7.46967C0.176777 7.76256 0.176777 8.23744 0.46967 8.53033L4.96967 13.0303C5.26256 13.3232 5.73744 13.3232 6.03033 13.0303L17.0303 2.03033C17.3232 1.73744 17.3232 1.26256 17.0303 0.96967C16.7374 0.676777 16.2626 0.676777 15.9697 0.96967L5.5 11.4393L1.53033 7.46967Z", fill: "#293056" })));
    }
    static get style() { return iconCss; }
}, [2, "checkmark-icon", {
        "width": [1],
        "height": [1]
    }]);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["checkmark-icon"];
    components.forEach(tagName => { switch (tagName) {
        case "checkmark-icon":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, CheckmarkIcon);
            }
            break;
    } });
}

export { CheckmarkIcon as C, defineCustomElement as d };
