// @nylas/.web-elements-GF2lXkj6/dist/components/close.js
import { proxyCustomElement, HTMLElement, h } from '@stencil/core/internal/client';

const iconCss = ".sc-close-icon-h{display:flex}";

const CloseIcon = proxyCustomElement(class CloseIcon extends HTMLElement {
    constructor() {
        super();
        this.__registerHost();
        this.width = '16';
        this.height = '16';
    }
    render() {
        return (h("svg", { key: '781556b9f4b7e7601555738d269d42c865e1d39a', width: "21", height: "21", fill: "none", xmlns: "http://www.w3.org/2000/svg" }, h("g", { key: '40e7787cbf60dc41c46e23571582d97b6ba95b6e', "clip-path": "url(#a)" }, h("path", { key: '991d9d4c71663288ebd4c5202c997624b8215678', d: "M5.73 4.574a.75.75 0 1 0-1.061 1.06l4.419 4.42-4.42 4.42a.75.75 0 1 0 1.061 1.06l4.42-4.42 4.419 4.42a.75.75 0 0 0 1.06-1.06l-4.419-4.42 4.42-4.42a.75.75 0 1 0-1.061-1.06l-4.42 4.42-4.419-4.42Z", fill: "currentColor" })), h("defs", { key: 'f8d0c70fc77f5be3791251e0f03208ece6af517f' }, h("clipPath", { key: '1c88efda98af5a21f7efbfa60d68ddad75e73893', id: "a" }, h("path", { key: '8d66c6f7b956df6d1db19ab1c7282ffaebc2406a', fill: "#fff", transform: "translate(.148 .054)", d: "M0 0h20v20H0z" })))));
    }
    static get style() { return iconCss; }
}, [2, "close-icon", {
        "width": [1],
        "height": [1]
    }]);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["close-icon"];
    components.forEach(tagName => { switch (tagName) {
        case "close-icon":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, CloseIcon);
            }
            break;
    } });
}

export { CloseIcon as C, defineCustomElement as d };
