// @nylas/.web-elements-GF2lXkj6/dist/components/envelope.js
import { proxyCustomElement, HTMLElement, h } from '@stencil/core/internal/client';

const iconCss = ".sc-envelope-icon-h{display:flex}";

const EnvelopeIcon = proxyCustomElement(class EnvelopeIcon extends HTMLElement {
    constructor() {
        super();
        this.__registerHost();
        this.width = '16';
        this.height = '13';
    }
    render() {
        return (h("svg", { key: '120db9adc96cbc0d1d59cc2dfb4f0db9aee9bdde', width: "16", height: "13", viewBox: "0 0 16 13", fill: "none", xmlns: "http://www.w3.org/2000/svg" }, h("path", { key: 'cdaf2c6b324f8dbc11b5888101ede0c0fc4e8602', d: "M13.5 0C14.8807 0 16 1.11929 16 2.5V10.5C16 11.8807 14.8807 13 13.5 13H2.5C1.11929 13 0 11.8807 0 10.5V2.5C0 1.11929 1.11929 0 2.5 0H13.5ZM15 3.961L8.25351 7.93097C8.12311 8.00767 7.96661 8.02045 7.82751 7.96932L7.74649 7.93097L1 3.963V10.5C1 11.3284 1.67157 12 2.5 12H13.5C14.3284 12 15 11.3284 15 10.5V3.961ZM13.5 1H2.5C1.67157 1 1 1.67157 1 2.5V2.802L8 6.91991L15 2.801V2.5C15 1.67157 14.3284 1 13.5 1Z", fill: "currentColor" })));
    }
    static get style() { return iconCss; }
}, [2, "envelope-icon", {
        "width": [1],
        "height": [1]
    }]);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["envelope-icon"];
    components.forEach(tagName => { switch (tagName) {
        case "envelope-icon":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, EnvelopeIcon);
            }
            break;
    } });
}

export { EnvelopeIcon as E, defineCustomElement as d };
