// @nylas/.web-elements-GF2lXkj6/dist/components/inbox.js
import { proxyCustomElement, HTMLElement, h } from '@stencil/core/internal/client';

const iconCss = ".sc-inbox-icon-h{display:flex}";

const InboxIcon = proxyCustomElement(class InboxIcon extends HTMLElement {
    constructor() {
        super();
        this.__registerHost();
        this.width = '15';
        this.height = '15';
    }
    render() {
        return (h("svg", { key: '3227c5a7258b329e27b0aefabacfb84cbda669eb', width: this.width, height: this.height, viewBox: "0 0 24 24", fill: "auto", xmlns: "http://www.w3.org/2000/svg" }, h("path", { key: '28e3db737991f8d8f73fbb3d2e86a47c596903eb', d: "M20 3H4C2.89543 3 2 3.89543 2 5V19C2 20.1046 2.89543 21 4 21H20C21.1046 21 22 20.1046 22 19V5C22 3.89543 21.1046 3 20 3Z", stroke: "auto", "stroke-width": "1.5" }), h("path", { key: '5f8c8bb09ef165ba72d57d860e09e88e0024d28c', d: "M2 5L12 12.5L22 5", stroke: "auto", "stroke-width": "1.5" })));
    }
    static get style() { return iconCss; }
}, [2, "inbox-icon", {
        "width": [1],
        "height": [1]
    }]);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["inbox-icon"];
    components.forEach(tagName => { switch (tagName) {
        case "inbox-icon":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, InboxIcon);
            }
            break;
    } });
}

export { InboxIcon as I, defineCustomElement as d };
