// @nylas/.web-elements-GF2lXkj6/dist/components/italic.js
import { proxyCustomElement, HTMLElement, h } from '@stencil/core/internal/client';

const iconCss = ".sc-italic-icon-h{display:flex}";

const ItalicIcon = proxyCustomElement(class ItalicIcon extends HTMLElement {
    constructor() {
        super();
        this.__registerHost();
        this.width = '15';
        this.height = '15';
    }
    render() {
        return (h("svg", { key: '42e08d459b503b00404ed6d853200439939d31c3', width: this.width, height: this.height, viewBox: "0 0 24 24", fill: "none", xmlns: "http://www.w3.org/2000/svg" }, h("path", { key: 'bf3ebd40f6ba15c02f132fcd0757b09d82ef031d', stroke: "currentColor", "stroke-linecap": "round", "stroke-linejoin": "round", "stroke-width": "2", d: "m10 20 4-16m2 0h-4m0 16H8" })));
    }
    static get style() { return iconCss; }
}, [2, "italic-icon", {
        "width": [1],
        "height": [1]
    }]);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["italic-icon"];
    components.forEach(tagName => { switch (tagName) {
        case "italic-icon":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, ItalicIcon);
            }
            break;
    } });
}

export { ItalicIcon as I, defineCustomElement as d };
