// @nylas/.web-elements-GF2lXkj6/dist/components/loading.js
import { proxyCustomElement, HTMLElement, h } from '@stencil/core/internal/client';

const loadingCss = ".wave{display:flex;align-items:center;justify-content:center}.dot{width:4px;height:4px;background-color:var(--dot-color, white);border-radius:50%;margin:0 2px;animation:wave 1.5s infinite ease-in-out}.dot:nth-child(2){animation-delay:0.2s}.dot:nth-child(3){animation-delay:0.4s}@keyframes wave{0%,60%,100%{transform:translateY(0)}30%{transform:translateY(-4px)}}";

const LoadingIcon = proxyCustomElement(class LoadingIcon extends HTMLElement {
    constructor() {
        super();
        this.__registerHost();
        this.__attachShadow();
        this.width = '15';
        this.height = '15';
    }
    render() {
        return (h("div", { key: '5ff9aa86f4d346ab894e59e1a4b96756a5f38231', class: "wave" }, h("div", { key: '35fa4038e2be7d2a18e65ada70f40f2d2e1001e8', class: "dot" }), h("div", { key: 'f16983cd2790b6a781cdb71e317046f7fa27c264', class: "dot" }), h("div", { key: 'c03e50ae47d23c1caa2c9c5c19f6d1ae274de1b7', class: "dot" })));
    }
    static get style() { return loadingCss; }
}, [1, "loading-icon", {
        "width": [1],
        "height": [1]
    }]);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["loading-icon"];
    components.forEach(tagName => { switch (tagName) {
        case "loading-icon":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, LoadingIcon);
            }
            break;
    } });
}

export { LoadingIcon as L, defineCustomElement as d };
