// @nylas/.web-elements-GF2lXkj6/dist/components/microsoft-logo.js
import { proxyCustomElement, HTMLElement, h } from '@stencil/core/internal/client';

const iconCss = ".sc-microsoft-logo-icon-h{display:flex}";

const MicrosoftLogoIcon = proxyCustomElement(class MicrosoftLogoIcon extends HTMLElement {
    constructor() {
        super();
        this.__registerHost();
        this.width = '22';
        this.height = '22';
    }
    render() {
        return (h("svg", { key: '20d8f297cf0cc69b7c398b6d0c24ca60a5c67109', width: this.width, height: this.height, viewBox: "0 0 22 22", fill: "none", xmlns: "http://www.w3.org/2000/svg" }, h("path", { key: 'dc420eb9c73909acfc6731606b24d015a600f34e', d: "M0.043457 0.043457H10.4782V10.4782H0.043457V0.043457Z", fill: "#F35325" }), h("path", { key: 'd22732be730bc90dcd3b415b9e5bba8de185ad87', d: "M11.5217 0.043457H21.9565V10.4782H11.5217V0.043457Z", fill: "#81BC06" }), h("path", { key: '8c8bff7ba92cad76967ccf67c7b3435d07193091', d: "M0.043457 11.5217H10.4782V21.9565H0.043457V11.5217Z", fill: "#05A6F0" }), h("path", { key: '51d0a3be2495c599aee6062c51fb4bd531fd4b5c', d: "M11.5217 11.5217H21.9565V21.9565H11.5217V11.5217Z", fill: "#FFBA08" })));
    }
    static get style() { return iconCss; }
}, [2, "microsoft-logo-icon", {
        "width": [1],
        "height": [1]
    }]);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["microsoft-logo-icon"];
    components.forEach(tagName => { switch (tagName) {
        case "microsoft-logo-icon":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, MicrosoftLogoIcon);
            }
            break;
    } });
}

export { MicrosoftLogoIcon as M, defineCustomElement as d };
