// @nylas/.web-elements-GF2lXkj6/dist/components/nylas-custom-booking-flow2.js
import { proxyCustomElement, HTMLElement, h, Host } from '@stencil/core/internal/client';
import { R as RegisterComponent } from './register-component.js';
import { a as debug, c as checkForMissingSlots } from './utils.js';
import { d as defineCustomElement$1 } from './nylas-form-card2.js';

const nylasCustomBookingFlowCss = ":host{display:block;margin:1rem;--nylas-primary:#2563eb;--nylas-error:#cc4841;--nylas-error-pressed:#992222;--nylas-wraning:#f06c00;--nylas-success:#16a392;--nylas-info:#2b8fc2;--nylas-base-0:#ffffff;--nylas-base-25:#fcfcfd;--nylas-base-50:#f8f9fc;--nylas-base-100:#eaecf5;--nylas-base-200:#d5d9eb;--nylas-base-300:#b3b8d8;--nylas-base-400:#717bbc;--nylas-base-500:#4e5ba6;--nylas-base-600:#3e4784;--nylas-base-700:#263f72;--nylas-base-800:#293056;--nylas-base-900:#101323;--nylas-base-950:#0e101b;--nylas-border-radius:0.25rem;--nylas-border-radius-2x:0.5rem;--nylas-border-radius-3x:0.75rem;--nylas-font-family:\"Inter\", sans-serif}.nylas-custom-booking-flow__title{color:var(--nylas-base-900);font-size:16px;font-style:normal;font-weight:600;line-height:20px;margin:0;text-align:left}.nylas-custom-booking-flow__subtitle{color:var(--nylas-base-600);font-size:14px;font-style:normal;font-weight:400;line-height:20px;margin:0.25rem 0px 0px;text-align:left}.nylas-custom-booking-flow__content{padding:1rem;display:flex;flex-direction:column;gap:1rem}.nylas-custom-booking-flow__block{display:flex;justify-content:space-between;gap:0.5rem}";

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (undefined && undefined.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
        return Reflect.metadata(k, v);
};
const NylasCustomBookingFlow = proxyCustomElement(class NylasCustomBookingFlow extends HTMLElement {
    constructor() {
        super();
        this.__registerHost();
        this.__attachShadow();
        this.internals = this.attachInternals();
    }
    connectedCallback() {
        debug('nylas-custom-booking-flow', 'connectedCallback');
    }
    disconnectedCallback() {
        debug('nylas-custom-booking-flow', 'disconnectedCallback');
    }
    componentWillLoad() {
        debug('nylas-custom-booking-flow', 'componentWillLoad');
    }
    componentDidLoad() {
        debug('nylas-custom-booking-flow', 'componentDidLoad');
        checkForMissingSlots(['inputs'], this.el);
    }
    render() {
        return (h(Host, { key: '9ac3169c2eff9023ac8b8059f14ae309af22d45a' }, h("nylas-form-card", { key: '02be9921b35d02afb300b272a2508871928514e3', exportparts: "nfc__header: ncbf__header" }, h("h1", { key: '0a7e4a9cc508c666cd0b5b3199b5e8fdbdcb9964', slot: "header-title", class: "nylas-custom-booking-flow__title", part: "ncbf__title" }, "Custom booking flow"), h("div", { key: '0416972885e9a3581b64677e297eb8cd10226bc8', slot: "content", class: "nylas-custom-booking-flow__content" }, h("slot", { key: '779f10585d3011df40021104b2ff8d9334544441', name: "inputs" })))));
    }
    static get formAssociated() { return true; }
    get el() { return this; }
    static get style() { return nylasCustomBookingFlowCss; }
}, [65, "nylas-custom-booking-flow"]);
__decorate([
    RegisterComponent({
        name: 'nylas-custom-booking-flow',
        eventToProps: {},
        fireRegisterEvent: true,
    }),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", []),
    __metadata("design:returntype", void 0)
], NylasCustomBookingFlow.prototype, "render", null);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["nylas-custom-booking-flow", "nylas-form-card"];
    components.forEach(tagName => { switch (tagName) {
        case "nylas-custom-booking-flow":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, NylasCustomBookingFlow);
            }
            break;
        case "nylas-form-card":
            if (!customElements.get(tagName)) {
                defineCustomElement$1();
            }
            break;
    } });
}

export { NylasCustomBookingFlow as N, defineCustomElement as d };
