// @nylas/.web-elements-GF2lXkj6/dist/components/nylas-event-info2.js
import { proxyCustomElement, HTMLElement, h, Host } from '@stencil/core/internal/client';
import { R as RegisterComponent } from './register-component.js';
import { a as debug, c as checkForMissingSlots } from './utils.js';
import { d as defineCustomElement$1 } from './nylas-form-card2.js';

const nylasEventInfoCss = ":host{display:block;margin:1rem;--nylas-primary:#2563eb;--nylas-error:#cc4841;--nylas-error-pressed:#992222;--nylas-wraning:#f06c00;--nylas-success:#16a392;--nylas-info:#2b8fc2;--nylas-base-0:#ffffff;--nylas-base-25:#fcfcfd;--nylas-base-50:#f8f9fc;--nylas-base-100:#eaecf5;--nylas-base-200:#d5d9eb;--nylas-base-300:#b3b8d8;--nylas-base-400:#717bbc;--nylas-base-500:#4e5ba6;--nylas-base-600:#3e4784;--nylas-base-700:#263f72;--nylas-base-800:#293056;--nylas-base-900:#101323;--nylas-base-950:#0e101b;--nylas-border-radius:0.25rem;--nylas-border-radius-2x:0.5rem;--nylas-border-radius-3x:0.75rem;--nylas-font-family:\"Inter\", sans-serif}.nylas-event-info__title{color:var(--nylas-base-900);font-size:16px;font-style:normal;font-weight:600;line-height:20px;margin:0;text-align:left}.nylas-event-info__subtitle{color:var(--nylas-base-600);font-size:14px;font-style:normal;font-weight:400;line-height:20px;margin:0.25rem 0px 0px;text-align:left}.nylas-event-info__content{padding:1rem;display:flex;flex-direction:column;gap:1rem}.nylas-event-info__block{display:flex;justify-content:space-between;gap:0.5rem}";

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (undefined && undefined.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
        return Reflect.metadata(k, v);
};
const NylasEventInfo = proxyCustomElement(class NylasEventInfo extends HTMLElement {
    constructor() {
        super();
        this.__registerHost();
        this.__attachShadow();
        this.internals = this.attachInternals();
    }
    connectedCallback() {
        debug('nylas-event-info', 'connectedCallback');
    }
    componentWillLoad() {
        debug('nylas-event-info', 'componentWillLoad');
    }
    componentDidLoad() {
        debug('nylas-event-info', 'componentDidLoad');
        checkForMissingSlots(['inputs'], this.el);
    }
    disconnectedCallback() {
        debug('nylas-event-info', 'disconnectedCallback');
    }
    render() {
        return (h(Host, { key: '99e646983326861e86df8fc8aaa731a22122e4ee' }, h("nylas-form-card", { key: 'd55c0c68e265458e464a077487cfcc4a34139467', exportparts: "nfc__header: nei__header" }, h("h1", { key: '2c934774492d57e6aa1e3a686a5712b23fa6ba6d', slot: "header-title", class: "nylas-event-info__title", part: "nei__title" }, "Event information"), h("div", { key: '89f21683e0ea7624eb8b9bcaa769cf4afd8e3337', slot: "content", class: "nylas-event-info__content" }, h("slot", { key: '8963ff3ffbb8309fb497ada00ddab86821726481', name: "inputs" })))));
    }
    static get formAssociated() { return true; }
    get el() { return this; }
    static get style() { return nylasEventInfoCss; }
}, [65, "nylas-event-info"]);
__decorate([
    RegisterComponent({
        name: 'nylas-event-info',
        eventToProps: {},
        fireRegisterEvent: true,
    }),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", []),
    __metadata("design:returntype", void 0)
], NylasEventInfo.prototype, "render", null);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["nylas-event-info", "nylas-form-card"];
    components.forEach(tagName => { switch (tagName) {
        case "nylas-event-info":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, NylasEventInfo);
            }
            break;
        case "nylas-form-card":
            if (!customElements.get(tagName)) {
                defineCustomElement$1();
            }
            break;
    } });
}

export { NylasEventInfo as N, defineCustomElement as d };
