// @nylas/.web-elements-GF2lXkj6/dist/components/play.js
import { proxyCustomElement, HTMLElement, h } from '@stencil/core/internal/client';

const iconCss = ".sc-play-icon-h{display:flex}";

const PlayIcon = proxyCustomElement(class PlayIcon extends HTMLElement {
    constructor() {
        super();
        this.__registerHost();
        this.width = '15';
        this.height = '15';
    }
    render() {
        return (h("svg", { key: '03aeffce3f9f6d60a15e334e51d4b8473724937a', width: this.width, height: this.height, viewBox: "0 0 24 24", fill: "auto", xmlns: "http://www.w3.org/2000/svg" }, h("path", { key: 'b150b7c852caa549be48caaf60e46bc442b15960', d: "M21 12L3 21V3L21 12Z", stroke: "auto", "stroke-width": "1.5", "stroke-linecap": "round", "stroke-linejoin": "round" })));
    }
    static get style() { return iconCss; }
}, [2, "play-icon", {
        "width": [1],
        "height": [1]
    }]);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["play-icon"];
    components.forEach(tagName => { switch (tagName) {
        case "play-icon":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, PlayIcon);
            }
            break;
    } });
}

export { PlayIcon as P, defineCustomElement as d };
