// @nylas/.web-elements-GF2lXkj6/dist/components/plus.js
import { proxyCustomElement, HTMLElement, h } from '@stencil/core/internal/client';

const iconCss = ".sc-plus-icon-h{display:flex}";

const PlusIcon = proxyCustomElement(class PlusIcon extends HTMLElement {
    constructor() {
        super();
        this.__registerHost();
        this.width = '20';
        this.height = '20';
    }
    render() {
        return (h("svg", { key: '7d8491efeecd10d51b6e7ddefe2a9889402daf13', width: this.width, height: this.height, fill: "none", xmlns: "http://www.w3.org/2000/svg" }, h("path", { key: '51086a93ef1651164187d97cf7f8841ffd2ff675', d: "M8.5 1C8.5 0.723858 8.27614 0.5 8 0.5C7.72386 0.5 7.5 0.723858 7.5 1V7.5H1C0.723858 7.5 0.5 7.72386 0.5 8C0.5 8.27614 0.723858 8.5 1 8.5H7.5V15C7.5 15.2761 7.72386 15.5 8 15.5C8.27614 15.5 8.5 15.2761 8.5 15V8.5H15C15.2761 8.5 15.5 8.27614 15.5 8C15.5 7.72386 15.2761 7.5 15 7.5H8.5V1Z", fill: "currentColor" })));
    }
    static get style() { return iconCss; }
}, [2, "plus-icon", {
        "width": [1],
        "height": [1]
    }]);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["plus-icon"];
    components.forEach(tagName => { switch (tagName) {
        case "plus-icon":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, PlusIcon);
            }
            break;
    } });
}

export { PlusIcon as P, defineCustomElement as d };
