// @nylas/.web-elements-GF2lXkj6/dist/components/sent.js
import { proxyCustomElement, HTMLElement, h } from '@stencil/core/internal/client';

const iconCss = ".sc-sent-icon-h{display:flex}";

const SentIcon = proxyCustomElement(class SentIcon extends HTMLElement {
    constructor() {
        super();
        this.__registerHost();
        this.width = '15';
        this.height = '15';
    }
    render() {
        return (h("svg", { key: '098aa96cca21e440ea4fe8d8363528fe9b1f277f', width: this.width, height: this.height, viewBox: "0 0 24 24", fill: "auto", xmlns: "http://www.w3.org/2000/svg" }, h("path", { key: 'b981887697dbb3c37cb32993c88324bea80cc098', d: "M21.5 12L3.5 3.5L3.5 20.5L21.5 12Z", stroke: "auto", "stroke-width": "1.5" }), h("path", { key: 'e7985ddf523274e49faee8da881e857a3aaca1ed', d: "M3.5 3.5L12 10.5", stroke: "auto", "stroke-width": "1.5" }), h("path", { key: '42a0ba5483a0cc9d79c1215bc5e0f31a406d256d', d: "M3.5 20.5L12 13.5", stroke: "auto", "stroke-width": "1.5" })));
    }
    static get style() { return iconCss; }
}, [2, "sent-icon", {
        "width": [1],
        "height": [1]
    }]);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["sent-icon"];
    components.forEach(tagName => { switch (tagName) {
        case "sent-icon":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, SentIcon);
            }
            break;
    } });
}

export { SentIcon as S, defineCustomElement as d };
