// @nylas/.web-elements-GF2lXkj6/dist/cjs/nylas-custom-booking-flow.cjs.entry.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-c14ea8f5.js');
const registerComponent = require('./register-component-2f532e44.js');
const utils = require('./utils-0e1c4d88.js');
require('./_commonjsHelpers-9463df90.js');

const nylasCustomBookingFlowCss = ":host{display:block;margin:1rem;--nylas-primary:#2563eb;--nylas-error:#cc4841;--nylas-error-pressed:#992222;--nylas-wraning:#f06c00;--nylas-success:#16a392;--nylas-info:#2b8fc2;--nylas-base-0:#ffffff;--nylas-base-25:#fcfcfd;--nylas-base-50:#f8f9fc;--nylas-base-100:#eaecf5;--nylas-base-200:#d5d9eb;--nylas-base-300:#b3b8d8;--nylas-base-400:#717bbc;--nylas-base-500:#4e5ba6;--nylas-base-600:#3e4784;--nylas-base-700:#263f72;--nylas-base-800:#293056;--nylas-base-900:#101323;--nylas-base-950:#0e101b;--nylas-border-radius:0.25rem;--nylas-border-radius-2x:0.5rem;--nylas-border-radius-3x:0.75rem;--nylas-font-family:\"Inter\", sans-serif}.nylas-custom-booking-flow__title{color:var(--nylas-base-900);font-size:16px;font-style:normal;font-weight:600;line-height:20px;margin:0;text-align:left}.nylas-custom-booking-flow__subtitle{color:var(--nylas-base-600);font-size:14px;font-style:normal;font-weight:400;line-height:20px;margin:0.25rem 0px 0px;text-align:left}.nylas-custom-booking-flow__content{padding:1rem;display:flex;flex-direction:column;gap:1rem}.nylas-custom-booking-flow__block{display:flex;justify-content:space-between;gap:0.5rem}";

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (undefined && undefined.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
        return Reflect.metadata(k, v);
};
const NylasCustomBookingFlow = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        if (hostRef.$hostElement$["s-ei"]) {
            this.internals = hostRef.$hostElement$["s-ei"];
        }
        else {
            this.internals = hostRef.$hostElement$.attachInternals();
            hostRef.$hostElement$["s-ei"] = this.internals;
        }
    }
    connectedCallback() {
        utils.debug('nylas-custom-booking-flow', 'connectedCallback');
    }
    disconnectedCallback() {
        utils.debug('nylas-custom-booking-flow', 'disconnectedCallback');
    }
    componentWillLoad() {
        utils.debug('nylas-custom-booking-flow', 'componentWillLoad');
    }
    componentDidLoad() {
        utils.debug('nylas-custom-booking-flow', 'componentDidLoad');
        utils.checkForMissingSlots(['inputs'], this.el);
    }
    render() {
        return (index.h(index.Host, { key: '9ac3169c2eff9023ac8b8059f14ae309af22d45a' }, index.h("nylas-form-card", { key: '02be9921b35d02afb300b272a2508871928514e3', exportparts: "nfc__header: ncbf__header" }, index.h("h1", { key: '0a7e4a9cc508c666cd0b5b3199b5e8fdbdcb9964', slot: "header-title", class: "nylas-custom-booking-flow__title", part: "ncbf__title" }, "Custom booking flow"), index.h("div", { key: '0416972885e9a3581b64677e297eb8cd10226bc8', slot: "content", class: "nylas-custom-booking-flow__content" }, index.h("slot", { key: '779f10585d3011df40021104b2ff8d9334544441', name: "inputs" })))));
    }
    static get formAssociated() { return true; }
    get el() { return index.getElement(this); }
};
__decorate([
    registerComponent.RegisterComponent({
        name: 'nylas-custom-booking-flow',
        eventToProps: {},
        fireRegisterEvent: true,
    }),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", []),
    __metadata("design:returntype", void 0)
], NylasCustomBookingFlow.prototype, "render", null);
NylasCustomBookingFlow.style = nylasCustomBookingFlowCss;

exports.nylas_custom_booking_flow = NylasCustomBookingFlow;
