// @nylas/.web-elements-GF2lXkj6/dist/cjs/nylas-booking-confirmation-redirect.cjs.entry.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-c14ea8f5.js');
const registerComponent = require('./register-component-2f532e44.js');
const utils = require('./utils-0e1c4d88.js');
require('./_commonjsHelpers-9463df90.js');

const nylasBookingConfirmationRedirectCss = ":host{display:block;--nylas-primary:#2563eb;--nylas-error:#cc4841;--nylas-error-pressed:#992222;--nylas-wraning:#f06c00;--nylas-success:#16a392;--nylas-info:#2b8fc2;--nylas-base-0:#ffffff;--nylas-base-25:#fcfcfd;--nylas-base-50:#f8f9fc;--nylas-base-100:#eaecf5;--nylas-base-200:#d5d9eb;--nylas-base-300:#b3b8d8;--nylas-base-400:#717bbc;--nylas-base-500:#4e5ba6;--nylas-base-600:#3e4784;--nylas-base-700:#263f72;--nylas-base-800:#293056;--nylas-base-900:#101323;--nylas-base-950:#0e101b;--nylas-border-radius:0.25rem;--nylas-border-radius-2x:0.5rem;--nylas-border-radius-3x:0.75rem;--nylas-font-family:\"Inter\", sans-serif;width:inherit}.nylas-booking-confirmation-redirect{width:inherit;display:flex;flex-direction:column}.nylas-booking-confirmation-redirect .nylas-booking-confirmation-redirect__dropdown span.label{display:flex;align-items:center;gap:4px;margin-bottom:4px;color:var(--nylas-base-800)}.nylas-booking-confirmation-redirect .nylas-booking-confirmation-redirect__dropdown span.label tooltip-component{display:flex}.nylas-booking-confirmation-redirect .nylas-booking-confirmation-redirect__dropdown select-dropdown::part(sd_dropdown){width:100%}.nylas-booking-confirmation-redirect .nylas-booking-confirmation-redirect__dropdown select-dropdown::part(sd_dropdown-button){width:100%;display:flex;justify-content:space-between;align-items:center;border:1px solid var(--nylas-base-200);border-radius:var(--nylas-border-radius-2x);padding:1rem}.nylas-booking-confirmation-redirect .nylas-booking-confirmation-redirect__dropdown select-dropdown::part(sd_dropdown-content){width:100%;max-width:unset}.nylas-booking-confirmation-redirect .nylas-booking-confirmation-redirect__dropdown select-dropdown::part(sd_dropdown-button-selected-label){max-width:calc(100% - 2rem);font-family:var(--nylas-font-family);font-size:16px;line-height:24px}.nylas-booking-confirmation-redirect .nylas-booking-confirmation-redirect__dropdown input-component::part(ic__input_wrapper){margin-top:1rem}.nylas-booking-confirmation-redirect .nylas-booking-confirmation-redirect__dropdown span.error-message{color:var(--nylas-error)}";

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (undefined && undefined.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
        return Reflect.metadata(k, v);
};
const NylasBookingConfirmationRedirect = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.valueChanged = index.createEvent(this, "valueChanged", 7);
        if (hostRef.$hostElement$["s-ei"]) {
            this.internals = hostRef.$hostElement$["s-ei"];
        }
        else {
            this.internals = hostRef.$hostElement$.attachInternals();
            hostRef.$hostElement$["s-ei"] = this.internals;
        }
        this.selectedConfiguration = undefined;
        this.name = 'confirmation-redirect';
        this.redirectUrl = '';
        this.redirectUrlErrorMessage = '';
        this.currentRedirectUrl = '';
        this.acceptUrl = undefined;
        this.componentLoaded = false;
    }
    elementNameChangedHandler(newValue) {
        utils.debug('nylas-booking-confirmation-redirect', 'elementNameChangedHandler', newValue);
        this.host.setAttribute('name', newValue);
    }
    selectedConfigurationChangedHandler(newValue) {
        utils.debug('nylas-booking-confirmation-redirect', 'selectedConfigurationChangedHandler', newValue);
        if (newValue?.scheduler?.confirmation_redirect_url) {
            this.currentRedirectUrl = newValue?.scheduler?.confirmation_redirect_url;
            this.acceptUrl = newValue?.scheduler?.confirmation_redirect_url !== '';
        }
    }
    connectedCallback() {
        utils.debug('nylas-booking-confirmation-redirect', 'connectedCallback');
    }
    disconnectedCallback() {
        utils.debug('nylas-booking-confirmation-redirect', 'disconnectedCallback');
    }
    componentWillLoad() {
        utils.debug('nylas-booking-confirmation-redirect', 'componentWillLoad');
        this.host.setAttribute('name', this.name);
    }
    componentDidLoad() {
        utils.debug('nylas-booking-confirmation-redirect', 'componentDidLoad');
        if (this.selectedConfiguration) {
            this.selectedConfigurationChangedHandler(this.selectedConfiguration);
        }
        else {
            this.currentRedirectUrl = this.redirectUrl;
            this.acceptUrl = this.redirectUrl !== '';
        }
        this.componentLoaded = true;
        if (typeof this.internals.setFormValue === 'function') {
            this.internals.setFormValue(this.currentRedirectUrl.toString(), this.name);
        }
    }
    componentWillUpdate() {
        utils.debug('nylas-booking-confirmation-redirect', 'componentWillUpdate');
    }
    componentDidUpdate() {
        utils.debug('nylas-booking-confirmation-redirect', 'componentDidUpdate');
    }
    componentWillRender() {
        utils.debug('nylas-booking-confirmation-redirect', 'componentWillRender');
    }
    componentDidRender() {
        utils.debug('nylas-booking-confirmation-redirect', 'componentDidRender');
    }
    nylasFormDropdownChangedHandler(event) {
        utils.debug('nylas-booking-confirmation-redirect', 'nylasFormDropdownChangedHandler', event.detail);
        const { name, value } = event.detail;
        if (name === 'confirmation-redirect-dropdown') {
            if (value === 'custom') {
                this.acceptUrl = true;
            }
            else {
                this.acceptUrl = false;
                this.internals.setFormValue('', this.name);
                this.valueChanged.emit({ name: this.name, value: '' });
            }
        }
    }
    nylasFormInputBlurHandler(event) {
        const { name, value } = event.detail;
        this.currentRedirectUrl = value;
        if (this.acceptUrl && name === 'confirmation-redirect') {
            this.redirectUrlErrorMessage = '';
            if (!utils.isValidUrl(this.currentRedirectUrl)) {
                this.internals.setValidity({ customError: true }, 'Please enter a valid URL.');
                this.redirectUrlErrorMessage = 'Please enter a valid URL.';
                return;
            }
            this.internals.setValidity({});
            const valueChanged = (event) => {
                const { value } = event.detail;
                this.internals.setFormValue(value.toString(), this.name);
            };
            this.valueChanged.emit({ ...event.detail, valueChanged });
        }
    }
    formSubmittedHandler(_event) {
        if (this.acceptUrl && !utils.isValidUrl(this.currentRedirectUrl)) {
            this.internals.setValidity({ customError: true }, 'Please enter a valid URL.');
            this.redirectUrlErrorMessage = 'Please enter a valid URL.';
        }
        else {
            this.internals.setValidity({});
        }
    }
    render() {
        const redirectOptions = [
            {
                label: 'Display default confirmation page',
                value: 'default',
            },
            {
                label: 'Redirect to custom URL',
                value: 'custom',
            },
        ];
        return (index.h(index.Host, { key: '664122ed45ac89e22d1638e1f89d152c6f3b2e64' }, index.h("div", { key: '86bad1489a91fc71ca708cf0832d7bf1ce5e4938', class: "nylas-booking-confirmation-redirect", part: "nbcr" }, redirectOptions.length > 0 && (index.h("div", { class: "nylas-booking-confirmation-redirect__dropdown" }, index.h("span", { class: "label" }, "When a booking is confirmed", index.h("tooltip-component", null, index.h("info-icon", { slot: "tooltip-icon" }), index.h("span", { slot: "tooltip-content" }, "You can either display the default confirmation page or redirect users to your own confirmation page."))), this.componentLoaded && (index.h("select-dropdown", { name: "confirmation-redirect-dropdown", options: redirectOptions, withSearch: false, defaultSelectedOption: redirectOptions.find(option => option.value === (this.acceptUrl ? 'custom' : 'default')), exportparts: "sd_dropdown: nbcr__dropdown, sd_dropdown-button: nbcr__dropdown-button, sd_dropdown-content: nbcr__dropdown-content" })), this.acceptUrl && (index.h("div", null, index.h("input-component", { name: "confirmation-redirect", id: "confirmation-redirect", type: "text", placeholder: "Enter custom redirect URL", required: false, part: "ic__input_wrapper: nbcr__input-textfield", defaultValue: this.currentRedirectUrl }), this.redirectUrlErrorMessage && index.h("span", { class: "error-message" }, this.redirectUrlErrorMessage))))))));
    }
    static get formAssociated() { return true; }
    get host() { return index.getElement(this); }
    static get watchers() { return {
        "name": ["elementNameChangedHandler"],
        "selectedConfiguration": ["selectedConfigurationChangedHandler"]
    }; }
};
__decorate([
    registerComponent.RegisterComponent({
        name: 'nylas-booking-confirmation-redirect',
        stateToProps: new Map([['schedulerConfig.selectedConfiguration', 'selectedConfiguration']]),
        eventToProps: {
            valueChanged: async (event, _nylasSchedulerConfigConnector) => {
                const { valueChanged } = event.detail;
                if (valueChanged) {
                    valueChanged(event);
                }
            },
        },
        fireRegisterEvent: true,
    }),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", []),
    __metadata("design:returntype", void 0)
], NylasBookingConfirmationRedirect.prototype, "render", null);
NylasBookingConfirmationRedirect.style = nylasBookingConfirmationRedirectCss;

exports.nylas_booking_confirmation_redirect = NylasBookingConfirmationRedirect;
