// @nylas/.web-elements-GF2lXkj6/dist/cjs/nylas-threads-search.cjs.entry.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-c14ea8f5.js');
const registerComponent = require('./register-component-2f532e44.js');
const utils = require('./utils-0e1c4d88.js');
require('./_commonjsHelpers-9463df90.js');

const nylasThreadsSearchCss = ":host{display:block}sp-search{width:100%;border:0}";

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (undefined && undefined.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
        return Reflect.metadata(k, v);
};
const NylasThreadsSearch = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.search = index.createEvent(this, "search", 7);
        this.onSearch = async (event) => {
            event.preventDefault();
            event.stopPropagation();
            const target = event.target;
            const value = target.value;
            if (value === this.internalQuery) {
                return;
            }
            this.internalQuery = value;
            this.search.emit(this.internalQuery);
        };
        this.query = '';
        this.internalQuery = '';
    }
    connectedCallback() { }
    disconnectedCallback() { }
    async componentWillLoad() {
        utils.debug(`[nylas-threads-search] Component will load`);
    }
    async componentDidLoad() {
        utils.debug(`[nylas-threads-search] Component did load`);
    }
    onQueryChange() {
        this.internalQuery = this.query;
    }
    render() {
        return (index.h(index.Host, { key: '6d49d740fb64cff220ddf6b74ca0393a3b6fd83a' }, index.h("sp-theme", { key: 'fda27523444c0c7350cdfdf6d1a29dd1d63158f5', scale: "medium", color: "dark" }, index.h("sp-search", { key: 'bb840884220a5008c144f90775ea10291c507b51', quiet: true, onSubmit: this.onSearch, onBlur: this.onSearch, value: this.internalQuery }))));
    }
    static get watchers() { return {
        "query": ["onQueryChange"]
    }; }
};
__decorate([
    registerComponent.RegisterComponent({
        name: 'nylas-pagination',
        stateToProps: new Map([['mailbox.searchQuery', 'query']]),
        eventToProps: {
            search: async (_event, nylasConnector) => {
                await nylasConnector.threads.getThreads({
                    search_query_native: _event.detail || undefined,
                    page_token: undefined,
                });
            },
        },
        fireRegisterEvent: true,
    }),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", []),
    __metadata("design:returntype", void 0)
], NylasThreadsSearch.prototype, "render", null);
NylasThreadsSearch.style = nylasThreadsSearchCss;

exports.nylas_threads_search = NylasThreadsSearch;
