// @nylas/.web-elements-GF2lXkj6/dist/cjs/nylas-cancelled-event-card.cjs.entry.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-c14ea8f5.js');
const registerComponent = require('./register-component-2f532e44.js');
const utils = require('./utils-0e1c4d88.js');
require('./_commonjsHelpers-9463df90.js');

const nylasCancelledEventCardCss = ":host{display:block}.nylas-cancelled-event-card{height:inherit;display:flex;flex-direction:column;justify-content:space-between;align-items:center;font-family:var(--nylas-font-family);width:400px;background-color:var(--nylas-base-0);color:var(--nylas-base-800);border-radius:var(--nylas-border-radius-2x);padding:1.5rem;position:relative;box-shadow:0px 1px 4px rgba(0, 0, 0, 0.1), 0px 3px 6px rgba(0, 0, 0, 0.06)}@media screen and (max-width: 768px){.nylas-cancelled-event-card{width:inherit}}.nylas-cancelled-event-card__title{font-size:18px;font-weight:600;line-height:24px;color:var(--nylas-base-900);text-align:center}.nylas-cancelled-event-description{font-size:16px;font-weight:500;color:var(--nylas-base-600)}.nylas-cancelled-event-card__calendar-icon{width:3rem;height:3rem;border-radius:50%;position:absolute;top:-1.25rem;left:50%;transform:translateX(-50%);background-color:var(--nylas-base-0);border:1px solid var(--nylas-base-200)}.nylas-cancelled-event-card__cta{width:100%;display:flex;flex-direction:row;align-items:flex-start;margin-top:1rem}calendar-cancel-icon{display:flex;align-items:center;justify-content:center;height:100%}button-component{width:100%}";

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (undefined && undefined.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
        return Reflect.metadata(k, v);
};
const NylasCancelledEventCard = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.closeCancelEventCardClicked = index.createEvent(this, "closeCancelEventCardClicked", 7);
        this.cancelledEventCardError = index.createEvent(this, "cancelledEventCardError", 7);
        this.handleCloseClicked = () => {
            this.isClosing = true;
            const errorHandler = (error) => {
                this.cancelledEventCardError.emit(error);
                this.isClosing = false;
            };
            this.closeCancelEventCardClicked.emit({ errorHandler });
        };
        this.cancelledEventInfo = undefined;
        this.isClosing = false;
        this.configSettings = undefined;
    }
    connectedCallback() {
        utils.debug(`[nylas-cancelled-event-card] Component connected`);
    }
    async componentWillLoad() {
        utils.debug(`[nylas-cancelled-event-card] Component will load`);
    }
    async componentDidLoad() {
        utils.debug(`[nylas-cancelled-event-card] Component did load`);
        if (!this.cancelledEventInfo) {
            console.warn(`[nylas-cancelled-event-card] No cancelled event info provided, "cancelledEventInfo" prop is required.`);
        }
    }
    disconnectedCallback() {
        utils.debug(`[nylas-cancelled-event-card] Component disconnected`);
    }
    render() {
        return (index.h(index.Host, { key: 'e890d06179d7e35d38abe0f30934645d0dd0866f', part: "ncec" }, index.h("div", { key: 'c2e455774c4d962dbd178c967d6ffcf0dad495e8', class: "nylas-cancelled-event-card", part: "ncec__card" }, index.h("div", { key: 'dae3d09a326b89e8d5c5ac662b663a87332b483f', class: "nylas-cancelled-event-card__calendar-icon", part: "ncec__icon" }, index.h("calendar-cancel-icon", { key: '908f8f9dce64ef8a02d12b8bea383a9e65662def' })), index.h("h3", { key: '61a2d624b51686a6a1d9e045a3efa99f204c9e21', class: "nylas-cancelled-event-card__title", part: "ncec__title" }, utils.instance.t('bookingCancelledTitle'), "!"), index.h("div", { key: '0cbf5bce08b5b623729e0741108ecd1a169c0cdb', class: "nylas-cancelled-event-card__description", part: "ncec__description" }, utils.instance.t('bookingCancelledMessage'), " ", this.configSettings?.organizer?.name || 'the participants', "."), index.h("div", { key: 'e43af5b395977d607b3c427bcb7d6062db8d6678', class: "nylas-cancelled-event-card__cta" }, index.h("button-component", { key: '36bb4ba3bf9fa46a52f5fd7b452bf8e001341d00', variant: "basic", disabled: this.isClosing, part: "ncec__button-outline", onClick: this.handleCloseClicked }, this.isClosing ? `${utils.instance.t('closingButton')}...` : `${utils.instance.t('closeButton')}`)))));
    }
};
__decorate([
    registerComponent.RegisterComponent({
        name: 'nylas-cancelled-event-card',
        stateToProps: new Map([['scheduler.configSettings', 'configSettings']]),
        eventToProps: {
            closeCancelEventCardClicked: async (_, nylasschedulerconnector) => {
                const result = await nylasschedulerconnector.scheduler.resetCancel();
                const { errorHandler } = _.detail;
                if (errorHandler && (!result || 'error' in result)) {
                    errorHandler(result);
                }
            },
            cancelledEventCardError: async (event, _nylasschedulerconnector) => {
                utils.debug('nylas-cancelled-event-card', 'cancelledEventCardError', event.detail);
            },
        },
        fireRegisterEvent: true,
    }),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", []),
    __metadata("design:returntype", void 0)
], NylasCancelledEventCard.prototype, "render", null);
NylasCancelledEventCard.style = nylasCancelledEventCardCss;

exports.nylas_cancelled_event_card = NylasCancelledEventCard;
