// @nylas/.web-elements-GF2lXkj6/dist/cjs/nylas-confirmed-event-card.cjs.entry.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-c14ea8f5.js');
const registerComponent = require('./register-component-2f532e44.js');
const utils = require('./utils-0e1c4d88.js');
require('./_commonjsHelpers-9463df90.js');

const nylasConfirmedEventCardCss = ":host{display:block;--nylas-primary:#2563eb;--nylas-error:#cc4841;--nylas-error-pressed:#992222;--nylas-wraning:#f06c00;--nylas-success:#16a392;--nylas-info:#2b8fc2;--nylas-base-0:#ffffff;--nylas-base-25:#fcfcfd;--nylas-base-50:#f8f9fc;--nylas-base-100:#eaecf5;--nylas-base-200:#d5d9eb;--nylas-base-300:#b3b8d8;--nylas-base-400:#717bbc;--nylas-base-500:#4e5ba6;--nylas-base-600:#3e4784;--nylas-base-700:#263f72;--nylas-base-800:#293056;--nylas-base-900:#101323;--nylas-base-950:#0e101b;--nylas-border-radius:0.25rem;--nylas-border-radius-2x:0.5rem;--nylas-border-radius-3x:0.75rem;--nylas-font-family:\"Inter\", sans-serif}.nylas-confirmed-event-card{height:inherit;display:flex;flex-direction:column;justify-content:space-between;align-items:center;font-family:var(--nylas-font-family);width:400px;background-color:var(--nylas-base-0);color:var(--nylas-base-800);border-radius:var(--nylas-border-radius-2x);padding:1.5rem;position:relative;box-shadow:0px 1px 4px rgba(0, 0, 0, 0.1), 0px 3px 6px rgba(0, 0, 0, 0.06)}@media screen and (max-width: 768px){.nylas-confirmed-event-card{width:inherit}}.nylas-confirmed-event-card__title{font-size:18px;font-weight:600;line-height:24px;color:var(--nylas-base-900);text-align:center}.nylas-confirmed-event-description{font-size:16px;font-weight:500;color:var(--nylas-base-600)}.nylas-confirmed-event-card__calendar-icon{width:3rem;height:3rem;border-radius:50%;position:absolute;top:-1.25rem;left:50%;transform:translateX(-50%);background-color:var(--nylas-base-0);border:1px solid var(--nylas-base-200)}.nylas-confirmed-event-card__cta{width:100%;display:flex;flex-direction:row;align-items:flex-start;margin-top:1rem}calendar-cancel-icon{display:flex;align-items:center;justify-content:center;height:100%}button-component{width:100%}";

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (undefined && undefined.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
        return Reflect.metadata(k, v);
};
const NylasConfirmedEventCard = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.closeConfirmEventCardClicked = index.createEvent(this, "closeConfirmEventCardClicked", 7);
        this.confirmedEventCardError = index.createEvent(this, "confirmedEventCardError", 7);
        this.handleCloseClicked = () => {
            this.isClosing = true;
            const errorHandler = (error) => {
                this.confirmedEventCardError.emit(error);
                this.isClosing = false;
            };
            this.closeConfirmEventCardClicked.emit({ errorHandler });
        };
        this.confirmedEventInfo = undefined;
        this.isClosing = false;
        this.configSettings = undefined;
    }
    connectedCallback() {
        utils.debug(`[nylas-confirmed-event-card] Component connected`);
    }
    async componentWillLoad() {
        utils.debug(`[nylas-confirmed-event-card] Component will load`);
    }
    async componentDidLoad() {
        utils.debug(`[nylas-confirmed-event-card] Component did load`);
        if (!this.confirmedEventInfo) {
            console.warn(`[nylas-confirmed-event-card] No confirmed event info provided, "confirmedEventInfo" prop is required.`);
        }
    }
    disconnectedCallback() {
        utils.debug(`[nylas-confirmed-event-card] Component disconnected`);
    }
    render() {
        return (index.h(index.Host, { key: 'f78755849f1e51daa16349080efbd1154ee5421f', part: "ncec" }, index.h("div", { key: '3bef7ce7a2365171566b29b096b1625d0724a322', class: "nylas-confirmed-event-card", part: "ncec__card" }, index.h("div", { key: '1a246865b9400017d1945eb079691ac8f94baf1b', class: "nylas-confirmed-event-card__calendar-icon", part: "ncec__icon" }, index.h("calendar-cancel-icon", { key: '6af3a586336dc9d89f9e20bd9fbb2d64641c883f' })), index.h("h3", { key: 'ccb933bf503bd9f316c04647dd99652ac4422c78', class: "nylas-confirmed-event-card__title", part: "ncec__title" }, utils.instance.t('confirmedEventCardTitle'), "!"), index.h("div", { key: '21d9fa9069fc947e05b1bfe5f9c5e9a6aa4bb091', class: "nylas-confirmed-event-card__description", part: "ncec__description" }, utils.instance.t('confirmedEventCardDescription'), " ", this.configSettings?.organizer?.name || 'the participants', "."), index.h("div", { key: '93ee8ddcb48b9ec5f30df4cce75ca1a13a38bfff', class: "nylas-confirmed-event-card__cta" }, index.h("button-component", { key: 'ca0960f860c3701821b492e318a47a7cf58d6c40', variant: "basic", disabled: this.isClosing, part: "ncec__button-outline", onClick: this.handleCloseClicked }, this.isClosing ? `${utils.instance.t('closingButton')}...` : `${utils.instance.t('closeButton')}`)))));
    }
};
__decorate([
    registerComponent.RegisterComponent({
        name: 'nylas-confirmed-event-card',
        stateToProps: new Map([['scheduler.configSettings', 'configSettings']]),
        eventToProps: {
            closeConfirmEventCardClicked: async (_, nylasschedulerconnector) => {
                const result = await nylasschedulerconnector.scheduler.resetConfirm();
                const { errorHandler } = _.detail;
                if (errorHandler && (!result || 'error' in result)) {
                    errorHandler(result);
                }
            },
        },
        fireRegisterEvent: true,
    }),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", []),
    __metadata("design:returntype", void 0)
], NylasConfirmedEventCard.prototype, "render", null);
NylasConfirmedEventCard.style = nylasConfirmedEventCardCss;

exports.nylas_confirmed_event_card = NylasConfirmedEventCard;
