// @nylas/.web-elements-GF2lXkj6/dist/collection/components/mailbox/nylas-composer/nylas-composer.js
import { debug } from "../../../utils/utils";
import { h, Host } from "@stencil/core";
export class NylasComposer {
    constructor() {
        this.closeComposer = (event) => {
            this.close.emit(event);
        };
        this.sendComposer = (event) => {
            if (!this.editorEl) {
                debug('no editor element');
                return;
            }
            const messageBody = this.editorEl.querySelector('.message-body');
            if (!messageBody) {
                debug('no message body');
                return;
            }
            if (this.message) {
                this.send.emit({
                    message: this.message,
                    replyBody: messageBody.querySelector('.message-body')?.innerHTML || '',
                });
                this.close.emit(event);
                this.editorEl.innerHTML = '';
            }
        };
        this.message = undefined;
        this.viewEmailElement = undefined;
    }
    connectedCallback() { }
    disconnectedCallback() {
        if (this.editorEl) {
            this.editorEl.innerHTML = '';
        }
    }
    componentWillLoad() { }
    async componentDidLoad() {
        if (this.message) {
            this.watchBody(this.message);
        }
        if (!this.editorEl) {
            debug('[connectedCallback] no editor element');
            return;
        }
        if (!this.toolbarEl) {
            debug('[connectedCallback] no toolbar element');
            return;
        }
        const editorEl = this.editorEl;
        const toolbarEl = this.toolbarEl;
        editorEl.addEventListener('mouseup', _ => {
            const selection = this.host.shadowRoot?.getSelection();
            if (!selection) {
                debug('no selection');
                return;
            }
            const range = selection.getRangeAt(0);
            const text = selection.toString();
            if (!text) {
                toolbarEl.style.display = 'none';
                return;
            }
            const messageBody = editorEl.querySelector('.message-body');
            if (!messageBody) {
                debug('no message body');
                return;
            }
            const foundNode = range.commonAncestorContainer;
            const foundElement = foundNode.parentElement;
            if (!foundElement) {
                return;
            }
            const messageBodyRect = messageBody.getBoundingClientRect();
            debug('messageBodyRect', { messageBodyRect });
            const foundRect = foundElement.getBoundingClientRect();
            debug('foundRect', { foundRect });
            const textNode = foundElement.childNodes[0];
            const textContent = textNode.textContent || '';
            const textPosition = textContent.indexOf(text);
            const fontSize = window.getComputedStyle(foundElement).fontSize;
            const fontSizeNumber = parseInt(fontSize.replace('px', ''), 10);
            const top = foundRect.top - messageBodyRect.top - toolbarEl.clientHeight;
            const left = textPosition * (1 / fontSizeNumber) * 100;
            toolbarEl.style.top = `${top}px`;
            toolbarEl.style.left = `${left}px`;
            toolbarEl.style.display = 'block';
        });
    }
    watchBody(newMessage) {
        if (!this.editorEl) {
            debug('no editor element');
            return;
        }
        this.editorEl.querySelectorAll('.message-body').forEach(el => el.remove());
        const messageBody = document.createElement('div');
        messageBody.classList.add('message-body');
        const emptyParagraph = document.createElement('p');
        emptyParagraph.innerHTML = '&nbsp;';
        messageBody.appendChild(emptyParagraph);
        const br = document.createElement('br');
        messageBody.appendChild(br);
        const replyText = document.createElement('div');
        replyText.classList.add('reply-text');
        replyText.innerText = `On ${new Date(newMessage.date * 1000).toLocaleDateString(undefined, { dateStyle: 'full' })} <${newMessage.from
            ?.map(from => from.name || from.email)
            .join(', ')}> wrote:`;
        messageBody.appendChild(replyText);
        if (newMessage.body) {
            const replyElement = document.createElement('html');
            replyElement.innerHTML = newMessage.body;
            const replyContainer = document.createElement('blockquote');
            replyContainer.classList.add('collapsed');
            replyContainer.classList.add('gmail_quote');
            replyContainer.innerHTML = replyElement.innerHTML;
            messageBody.appendChild(replyContainer);
            debug('replyContainer', { replyContainer });
        }
        this.editorEl.appendChild(messageBody);
        this.editorEl.querySelectorAll('blockquote.collapsed').forEach(blockquote => {
            blockquote.addEventListener('click', function (event) {
                event.preventDefault();
                const el = event.target;
                el.classList.remove('collapsed');
                el.classList.add('expanded');
            }, { once: true });
        });
        this.editorEl.focus();
    }
    render() {
        return (h(Host, { key: 'e24e688a3965e332b1d14284a0142bd82048449b' }, h("sp-theme", { key: '7bd78108aeab6bd84d51002dfecfdfd8658f82df', scale: "medium", color: "dark" }, h("div", { key: '0408b3da7f2fa20c80aa57167c32504fba8bd752', class: "editor", contentEditable: true, ref: r => (this.editorEl = r) }, h("div", { key: 'a0a78800e4a463cc47dbe74d4b95d633ce58cb0c', class: "toolbar", ref: r => (this.toolbarEl = r), contentEditable: false }, h("sp-button", { key: '80af95fbdadd9adfcc6b7cc7cf0c63f705eb8164', quiet: true, onClick: () => document.execCommand('bold') }, h("div", { key: 'f9db6a419ad21918d0abc3e45b50c9722b5655ec', slot: "icon" }, h("bold-icon", { key: 'a4e8df2a934c404ff435ef0920d3bd69204ecda2' })), h("div", { key: '609415df8fccca26ac59095400d7a8c83605156a', slot: "label" }, "Bold")), h("sp-button", { key: '4b90d3395738fda4242994134b7625535b21157b', quiet: true, onClick: () => document.execCommand('underline') }, h("div", { key: '2cf5470ae2a40175682cd2423035b59d8f2aabe2', slot: "icon" }, h("underline-icon", { key: '4d77573ab3628b4956c196be6db4c00cfcce9699' })), h("div", { key: '21362c41bcff11414325a38882586013cefef497', slot: "label" }, "Underline")), h("sp-button", { key: '8d1bdae9a2d782364c4848118bab933e8ca6c7f5', quiet: true, onClick: () => document.execCommand('italic') }, h("div", { key: '743c08e076b633500dd08b569d7496784447c73f', slot: "icon" }, h("italic-icon", { key: '0a1ca27562711edf8b7b7221d420fcfdd5110a95' })), h("div", { key: '3b1b29a8eb334a59147b9c46eb67064201a44bf8', slot: "label" }, "Italic")))), h("div", { key: '901be1e103c0f2b851669d8126d6258091530e81', class: 'buttons' }, h("sp-button", { key: 'b6bba4d7ae73f610bd3af6c08a0167ffab3dbc88', onClick: this.sendComposer, class: "send" }, "Send"), h("sp-button", { key: '7b68f79eb2fb7a73f3f69863a3cf6fafdcdfb9f1', onClick: this.closeComposer, class: "close" }, "Close")))));
    }
    static get is() { return "nylas-composer"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["nylas-composer.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["nylas-composer.css"]
        };
    }
    static get properties() {
        return {
            "message": {
                "type": "unknown",
                "mutable": false,
                "complexType": {
                    "original": "Message",
                    "resolved": "Message | undefined",
                    "references": {
                        "Message": {
                            "location": "import",
                            "path": "@nylas/core",
                            "id": "../nylas-js-core/dist/index.d.ts::Message"
                        }
                    }
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": "The message to reply to or forward"
                }
            },
            "viewEmailElement": {
                "type": "unknown",
                "mutable": false,
                "complexType": {
                    "original": "HTMLNylasViewEmailElement | null",
                    "resolved": "HTMLNylasViewEmailElement | null | undefined",
                    "references": {
                        "HTMLNylasViewEmailElement": {
                            "location": "global",
                            "id": "global::HTMLNylasViewEmailElement"
                        }
                    }
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": "The view email element"
                }
            }
        };
    }
    static get events() {
        return [{
                "method": "close",
                "name": "close",
                "bubbles": true,
                "cancelable": true,
                "composed": true,
                "docs": {
                    "tags": [],
                    "text": "Fired when the composer is closed"
                },
                "complexType": {
                    "original": "MouseEvent",
                    "resolved": "MouseEvent",
                    "references": {
                        "MouseEvent": {
                            "location": "global",
                            "id": "global::MouseEvent"
                        }
                    }
                }
            }, {
                "method": "send",
                "name": "send",
                "bubbles": true,
                "cancelable": true,
                "composed": true,
                "docs": {
                    "tags": [],
                    "text": "Fired when the send button is clicked"
                },
                "complexType": {
                    "original": "{\n    message: Message;\n    replyBody: string;\n  }",
                    "resolved": "{ message: Message; replyBody: string; }",
                    "references": {
                        "Message": {
                            "location": "import",
                            "path": "@nylas/core",
                            "id": "../nylas-js-core/dist/index.d.ts::Message"
                        }
                    }
                }
            }];
    }
    static get elementRef() { return "host"; }
    static get watchers() {
        return [{
                "propName": "message",
                "methodName": "watchBody"
            }];
    }
}
