// @nylas/.web-elements-GF2lXkj6/dist/collection/components/design-system/checkbox-component/checkbox-component.js
import { h } from "@stencil/core";
export class CheckboxComponent {
    constructor() {
        this.componentType = 'checkbox';
        this.name = 'checkbox';
        this.checked = undefined;
        this.label = '';
        this.required = false;
        this.isChecked = undefined;
        this.error = '';
    }
    componentWillLoad() {
        this.el.setAttribute('component-type', this.componentType);
    }
    componentDidLoad() {
        this.isChecked = this.checked || false;
    }
    handleBookingFormSubmitted(event) {
        if (this.required && !this.isChecked) {
            this.error = `${this.label} is required.`;
            return;
        }
        if (this.error) {
            event.preventDefault();
        }
    }
    toggleCheck() {
        this.error = '';
        this.isChecked = !this.isChecked;
        this.nylasFormCheckboxToggled.emit({
            checked: this.isChecked,
            name: this.name,
            label: this.label,
        });
    }
    render() {
        return (h("div", { key: '2e9457e9cc43e9361637d4e7d708d5ed7bcc81e7', class: "checkbox-container" }, h("label", { key: '3c67d580d1c99e55651d819d8412164f3f781853' }, h("input", { key: '01d2fd66e344ab9465cf1899327f5ee25deb6d8d', type: "checkbox", name: this.name, checked: this.isChecked, onChange: () => this.toggleCheck() }), h("span", { key: '67f17754686147fcfdaf20fdf779f4bb2a213e09', class: "checkmark" }), h("slot", { key: '4ee5c4d97e1911c0481a7983fd5893be14d27656', name: "label" }, h("span", { key: '387fd51b12ae65d1a68436146dd71fcad6a67e88', class: "label" }, this.label))), this.error && h("p", { class: "error" }, this.error)));
    }
    static get is() { return "checkbox-component"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["checkbox-component.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["checkbox-component.css"]
        };
    }
    static get properties() {
        return {
            "name": {
                "type": "string",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "The name of the checkbox. This is used to identify the checkbox when submitting a form."
                },
                "attribute": "name",
                "reflect": false,
                "defaultValue": "'checkbox'"
            },
            "checked": {
                "type": "boolean",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean | undefined",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": "The default value of the checkbox. This is the value that is displayed when the checkbox is rendered."
                },
                "attribute": "checked",
                "reflect": false
            },
            "label": {
                "type": "string",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "The label of the checkbox. This is displayed next to the checkbox."
                },
                "attribute": "label",
                "reflect": false,
                "defaultValue": "''"
            },
            "required": {
                "type": "boolean",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "Whether the checkbox is required. If true, the checkbox must be checked when submitting a form.\nDefault is false. If the checkbox is required and not checked, an error message is displayed."
                },
                "attribute": "required",
                "reflect": false,
                "defaultValue": "false"
            }
        };
    }
    static get states() {
        return {
            "isChecked": {},
            "error": {}
        };
    }
    static get events() {
        return [{
                "method": "nylasFormCheckboxToggled",
                "name": "nylasFormCheckboxToggled",
                "bubbles": true,
                "cancelable": true,
                "composed": true,
                "docs": {
                    "tags": [],
                    "text": "This event is fired when the checkbox is toggled."
                },
                "complexType": {
                    "original": "{\n    checked: boolean;\n    name: string;\n    label: string;\n  }",
                    "resolved": "{ checked: boolean; name: string; label: string; }",
                    "references": {}
                }
            }];
    }
    static get elementRef() { return "el"; }
    static get listeners() {
        return [{
                "name": "bookingFormSubmitted",
                "method": "handleBookingFormSubmitted",
                "target": "document",
                "capture": false,
                "passive": false
            }];
    }
}
