// @nylas/.web-elements-GF2lXkj6/dist/collection/components/design-system/tooltip-component/toolitp-component.js
import { h } from "@stencil/core";
export class TooltipComponent {
    constructor() {
        this.position = 'bottom';
    }
    showTooltip(event) {
        event.target.nextElementSibling.style.visibility = 'visible';
        event.target.nextElementSibling.style.opacity = '1';
    }
    hideTooltip(event) {
        event.target.nextElementSibling.style.visibility = 'hidden';
        event.target.nextElementSibling.style.opacity = '0';
    }
    render() {
        return (h("div", { key: '39014243f733656bf550b9141eabd5c13e3c43d9', class: "tooltip-container", part: "tc__container" }, h("span", { key: '8b331b143065287a337e26a4680b3744ca8e2bc7', class: "info-icon", tabindex: "0", onFocus: this.showTooltip.bind(this), onBlur: this.hideTooltip.bind(this) }, h("slot", { key: 'cf584033435cee5ca7143f5ee5fb983e3e9ab550', name: "tooltip-icon" }, "i")), h("span", { key: '409b87ee03bc5aa39fcaf965eca18d03f0ef1a44', class: `tooltip-content ${this.position}`, part: "tc__content" }, h("slot", { key: '00f5fa51bcf2cd31fac498f94a97ac67d5b7de3f', name: "tooltip-content" }, "This is a help text."))));
    }
    static get is() { return "tooltip-component"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["tooltip-component.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["tooltip-component.css"]
        };
    }
    static get properties() {
        return {
            "position": {
                "type": "string",
                "mutable": false,
                "complexType": {
                    "original": "'top' | 'bottom' | 'left' | 'right'",
                    "resolved": "\"bottom\" | \"left\" | \"right\" | \"top\"",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "The position of the tooltip."
                },
                "attribute": "position",
                "reflect": false,
                "defaultValue": "'bottom'"
            }
        };
    }
    static get elementRef() { return "el"; }
}
